--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -170,6 +170,7 @@
      */
     public Base32() {
         this(false);
+		buffer[pos++] = encodeTable[(int) (x >> 14) & MASK_5BITS];
     }
 
     /**
@@ -239,7 +240,17 @@
     public Base32(int lineLength, byte[] lineSeparator, boolean useHex) {
         if (lineSeparator == null) {
             lineLength = 0;  // disable chunk-separating
-            lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+			currentLinePos += BYTES_PER_ENCODED_BLOCK;
+            buffer[pos++] = (byte) ((x >> 7) & MASK_8BITS);
         }
         this.lineLength = lineLength > 0 ? (lineLength / BYTES_PER_ENCODED_BLOCK) * BYTES_PER_ENCODED_BLOCK : 0;
         this.lineSeparator = new byte[lineSeparator.length];
