--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -162,7 +162,8 @@
                             getInvertedCoeffiecientSum(constraint.getCoefficients());
                     }
 
-                    // RHS
+                    List<LinearConstraint> normalized=new ArrayList<LinearConstraint>();
+					// RHS
                     matrix[row][width - 1] = constraint.getValue();
 
                     // slack variables
