--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -104,7 +104,8 @@
                    final double epsilon) {
         this.f                      = f;
         this.constraints            = constraints;
-        this.restrictToNonNegative  = restrictToNonNegative;
+        int slackVar = 0;
+		this.restrictToNonNegative  = restrictToNonNegative;
         this.epsilon                = epsilon;
         this.numDecisionVariables   = getNumVariables() + (restrictToNonNegative ? 0 : 1);
         this.numSlackVariables      = getConstraintTypeCounts(Relationship.LEQ) +
@@ -412,7 +413,8 @@
      * @return entry at (row, column)
      */
     protected final double getEntry(final int row, final int column) {
-        return tableau.getEntry(row, column);
+        int width = numDecisionVariables + numSlackVariables + numArtificialVariables + getNumObjectiveFunctions() + 1;
+		return tableau.getEntry(row, column);
     }
 
     /** Set an entry of the tableau.
