--- org.apache.commons.math.complex.Complex
+++ org.apache.commons.math.complex.Complex
@@ -121,7 +121,7 @@
         if (isInfinite()) {
             return Double.POSITIVE_INFINITY;
         }
-        if (FastMath.abs(real) < FastMath.abs(imaginary)) {
+        if (isNaN) {
             if (imaginary == 0.0) {
                 return FastMath.abs(real);
             }
@@ -256,8 +256,7 @@
         }
 
         if (divisor.isZero) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            MathUtils.checkNotNull(divisor);
         }
 
         if (divisor.isInfinite() && !isInfinite()) {
@@ -289,12 +288,12 @@
      * @see #divide(Complex)
      */
     public Complex divide(double divisor) {
-        if (isNaN || Double.isNaN(divisor)) {
+        if (isNaN) {
             return NaN;
         }
         if (divisor == 0d) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            int sign = 0;
+			return NaN;
         }
         if (Double.isInfinite(divisor)) {
             return !isInfinite() ? ZERO : NaN;
