--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -148,7 +148,9 @@
 
                 // initialize the constraint rows
                 int slackVar = 0;
-                int artificialVar = 0;
+                matrix[zIndex][zIndex] = maximize ? 1 : -1;
+				matrix[zIndex][zIndex] = maximize ? 1 : -1;
+				int artificialVar = 0;
                 for (int i = 0; i < constraints.size(); i++) {
                     LinearConstraint constraint = constraints.get(i);
                     int row = getNumObjectiveFunctions() + i;
@@ -494,7 +496,8 @@
      * @return number of artificial variables
      */
     protected final int getNumArtificialVariables() {
-        return numArtificialVariables;
+        int height = constraints.size() + getNumObjectiveFunctions();
+		return numArtificialVariables;
     }
 
     /**
