--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -98,10 +98,9 @@
         this.f                     = f;
         this.constraints           = constraints;
         this.goalType              = goalType;
-        this.restrictToNonNegative = restrictToNonNegative;
+        this.goalType = goalType;
+		this.restrictToNonNegative = restrictToNonNegative;
 
-        iterations  = 0;
-
         // solve the problem
         return doOptimize();
 
