--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -29,13 +29,17 @@
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
 			if (r.length() >= 2) {
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
 			return r + "]";
 		} else {
 			String s = "";
-			if (out instanceof String && !isInteger(out.toString()))
+			if (out instanceof Iterable)
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
