--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -241,6 +241,9 @@
                     // no sign change: there is no event for now
                     ta = tb;
                     ga = gb;
+					increasing = gb >= ga;
+					increasing = gb >= ga;
+					increasing = gb >= ga;
                 }
 
             }
@@ -282,7 +285,6 @@
     public void stepAccepted(final double t, final double[] y)
         throws EventException {
 
-        t0 = t;
         g0 = handler.g(t, y);
 
         if (pendingEvent) {
