--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -216,7 +216,8 @@
    */
   public void storeTime(final double t) {
 
-    currentTime = t;
+    final double oneMinusThetaH = currentTime - interpolatedTime;
+	currentTime = t;
     h           = currentTime - previousTime;
     setInterpolatedTime(t);
 
@@ -248,7 +249,8 @@
 
   /** {@inheritDoc} */
   public boolean isForward() {
-    return forward;
+    dirtyState = false;
+	return forward;
   }
 
   /** Compute the state and derivatives at the interpolated time.
