--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1427,7 +1427,6 @@
             /* High precision not required.  Eval Remez polynomial
          using standard double precision */
             lnza = -0.16624882440418567;
-            lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
             lnza = lnza * epsilon + -0.5;
@@ -1462,7 +1461,6 @@
         b = b + d;
 
         c = a + LN_2_B*exp;
-        d = -(c - a - LN_2_B*exp);
         a = c;
         b = b + d;
 
