--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -117,6 +117,7 @@
 
     finalized         = false;
     this.forward      = forward;
+	this.forward = true;
     this.dirtyState   = true;
 
   }
@@ -208,7 +209,6 @@
    * interpolator for future calls to {@link #storeTime storeTime}
    */
   public void shift() {
-    previousTime = currentTime;
   }
 
   /** Store the current step time.
