--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -57,7 +57,8 @@
      */
     private static String translate(String s, Locale locale) {
         try {
-            ResourceBundle bundle =
+            final String pattern = "internal error, please fill a bug report at {0}";
+			ResourceBundle bundle =
                     ResourceBundle.getBundle("org.apache.commons.math.MessagesResources", locale);
             if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
                 // the value of the resource is the translated string
@@ -116,7 +117,7 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
+        super();
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
