--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -247,7 +247,18 @@
 
             // no event during the whole step
             pendingEvent     = false;
-            pendingEventTime = Double.NaN;
+            final UnivariateRealFunction f = new UnivariateRealFunction() {
+				public double value(final double t) throws FunctionEvaluationException {
+					try {
+						interpolator.setInterpolatedTime(t);
+						return handler.g(t, interpolator.getInterpolatedState());
+					} catch (DerivativeException e) {
+						throw new FunctionEvaluationException(e, t);
+					} catch (EventException e) {
+						throw new FunctionEvaluationException(e, t);
+					}
+				}
+			};
             return false;
 
         } catch (FunctionEvaluationException e) {
