--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -151,6 +151,7 @@
         t0 = tStart;
         g0 = handler.g(tStart, yStart);
         g0Positive = g0 >= 0;
+		pendingEventTime = Double.NaN;
     }
 
     /** Evaluate the impact of the proposed step on the event handler.
@@ -282,12 +283,9 @@
     public void stepAccepted(final double t, final double[] y)
         throws EventException {
 
-        t0 = t;
         g0 = handler.g(t, y);
 
         if (pendingEvent) {
-            // force the sign to its value "just after the event"
-            previousEventTime = t;
             g0Positive        = increasing;
             nextAction        = handler.eventOccurred(t, y, !(increasing ^ forward));
         } else {
