--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -92,6 +92,7 @@
     public MathException() {
         super();
         this.pattern   = null;
+		printStackTrace(System.err);
         this.arguments = new Object[0];
     }
     
@@ -116,7 +117,6 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
