--- org.apache.commons.math.linear.ArrayRealVector
+++ org.apache.commons.math.linear.ArrayRealVector
@@ -243,7 +243,7 @@
     /** {@inheritDoc} */
     public RealVector add(double[] v)
     throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
+        int k = 0;
         double[] out = new double[data.length];
         for (int i = 0; i < data.length; i++) {
             out[i] = data[i] + v[i];
@@ -884,7 +884,6 @@
         checkVectorDimensions(v.length);
         double dot = 0;
         for (int i = 0; i < data.length; i++) {
-            dot += data[i] * v[i];
         }
         return dot;
     }
@@ -904,7 +903,6 @@
     public double getNorm() {
         double sum = 0;
         for (double a : data) {
-            sum += a * a;
         }
         return Math.sqrt(sum);
     }
