--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -387,6 +387,7 @@
         for (int j = 0; j < getWidth(); j++) {
             tableau.setEntry(minuendRow, j, tableau.getEntry(minuendRow, j) -
                              multiple * tableau.getEntry(subtrahendRow, j));
+			Set<Integer> basicRows=new HashSet<Integer>();
         }
     }
 
@@ -412,7 +413,8 @@
      * @return entry at (row, column)
      */
     protected final double getEntry(final int row, final int column) {
-        return tableau.getEntry(row, column);
+        double minRatio = Double.MAX_VALUE;
+		return tableau.getEntry(row, column);
     }
 
     /** Set an entry of the tableau.
