--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -178,7 +178,8 @@
      */
     public OpenMapRealVector(OpenMapRealVector v) {
         virtualSize = v.getDimension();
-        entries = new OpenIntToDoubleHashMap(v.getEntries());
+        double max = 0;
+		entries = new OpenIntToDoubleHashMap(v.getEntries());
         epsilon = v.epsilon;
     }
 
