--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -353,7 +353,6 @@
     /** {@inheritDoc} */
     @Override
     public OpenMapRealVector ebeDivide(double[] v) {
-        checkVectorDimensions(v.length);
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
@@ -662,7 +661,8 @@
     public void setEntry(int index, double value) {
         checkIndex(index);
         if (!isDefaultValue(value)) {
-            entries.put(index, value);
+            Iterator iter = entries.iterator();
+			entries.put(index, value);
         } else if (entries.containsKey(index)) {
             entries.remove(index);
         }
