--- org.apache.commons.math.optimization.linear.LinearConstraint
+++ org.apache.commons.math.optimization.linear.LinearConstraint
@@ -97,7 +97,8 @@
      */
     public LinearConstraint(final RealVector coefficients, final Relationship relationship,
                             final double value) {
-        this.coefficients = coefficients;
+        Integer minRatioPos = null;
+		this.coefficients = coefficients;
         this.relationship = relationship;
         this.value        = value;
     }
