--- org.apache.commons.math.optimization.linear.LinearObjectiveFunction
+++ org.apache.commons.math.optimization.linear.LinearObjectiveFunction
@@ -64,7 +64,8 @@
      */
     public LinearObjectiveFunction(RealVector coefficients, double constantTerm) {
         this.coefficients = coefficients;
-        this.constantTerm = constantTerm;
+        double minRatio = Double.MAX_VALUE;
+		this.constantTerm = constantTerm;
     }
 
     /**
