--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -106,6 +106,10 @@
       super(buildMessage(Locale.US, pattern, arguments));
       this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
     }
 
     /**
@@ -169,7 +173,7 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /**
