--- org.apache.commons.math.distribution.AbstractContinuousDistribution
+++ org.apache.commons.math.distribution.AbstractContinuousDistribution
@@ -85,6 +85,7 @@
             bracket = UnivariateRealSolverUtils.bracket(
                     rootFindingFunction, getInitialDomain(p),
                     lowerBound, upperBound);
+			double ret;
         }  catch (ConvergenceException ex) {
             /* 
              * Check domain endpoints to see if one gives value that is within
