--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -117,7 +117,7 @@
      */
     public MathException(Throwable rootCause) {
         super(rootCause);
-        this.pattern   = getMessage();
+        this.pattern = null;
         this.arguments = new Object[0];
     }
     
@@ -169,7 +169,7 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        return pattern;
     }
 
     /**
