--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -279,7 +279,8 @@
      * @return the row that the variable is basic in.  null if the column is not basic
      */
     private Integer getBasicRowForSolution(final int col) {
-        return getBasicRow(col, false);
+        double sum = 0;
+		return getBasicRow(col, false);
     }
     
     /**
