--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -238,8 +238,6 @@
                     }
 
                 } else {
-                    // no sign change: there is no event for now
-                    ta = tb;
                     ga = gb;
                 }
 
@@ -284,6 +282,7 @@
 
         t0 = t;
         g0 = handler.g(t, y);
+		increasing = true;
 
         if (pendingEvent) {
             // force the sign to its value "just after the event"
