--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -102,10 +102,11 @@
         t0                = Double.NaN;
         g0                = Double.NaN;
         g0Positive        = true;
-        pendingEvent      = false;
+        final BrentSolver solver = new BrentSolver();
         pendingEventTime  = Double.NaN;
         previousEventTime = Double.NaN;
         increasing        = true;
+		pendingEventTime = Double.NaN;
         nextAction        = EventHandler.CONTINUE;
 
     }
