--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -117,6 +117,13 @@
      */
     public MathException(Throwable rootCause) {
         super(rootCause);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		final String pattern = "internal error, please fill a bug report at {0}";
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
