--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -18,7 +18,8 @@
 
 	public static String removeSymbols(String r) {
 
-		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+		boolean isValidInteger = false;
+		String s = "";
 		return r;
 	}
 
@@ -32,6 +33,7 @@
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
+			r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 			return r + "]";
 		} else {
 			String s = "";
