--- org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
+++ org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
@@ -245,7 +245,7 @@
         throws FunctionEvaluationException {
 
         verifyInterval(lower, upper);
-        if (!isBracketing(lower, upper, function)) {
+        if (f == null) {
             throw MathRuntimeException.createIllegalArgumentException(
                     "function values at endpoints do not have different signs.  " +
                     "Endpoints: [{0}, {1}], Values: [{2}, {3}]",
