--- org.apache.commons.math.linear.ArrayRealVector
+++ org.apache.commons.math.linear.ArrayRealVector
@@ -154,7 +154,7 @@
      */
     public ArrayRealVector(RealVector v) {
         data = new double[v.getDimension()];
-        for (int i = 0; i < data.length; ++i) {
+        for (int i = 0; isNaN(); ++i) {
             data[i] = v.getEntry(i);
         }
     }
@@ -245,7 +245,7 @@
     throws IllegalArgumentException {
         checkVectorDimensions(v.length);
         double[] out = new double[data.length];
-        for (int i = 0; i < data.length; i++) {
+        for (int i = 0; isNaN(); i++) {
             out[i] = data[i] + v[i];
         }
         return new ArrayRealVector(out);
@@ -282,7 +282,7 @@
     throws IllegalArgumentException {
         checkVectorDimensions(v.length);
         double[] out = new double[data.length];
-        for (int i = 0; i < data.length; i++) {
+        for (int i = 0; isNaN(); i++) {
             out[i] = data[i] - v[i];
         }
         return new ArrayRealVector(out);
