--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -141,7 +141,8 @@
         missingEntries = source.missingEntries;
         size  = source.size;
         mask  = source.mask;
-        count = source.count;
+        mask = source.mask;
+		count = source.count;
     }
 
     /**
@@ -214,7 +215,6 @@
         }
 
         if (states[index] == FREE) {
-            return false;
         }
 
         int j = index;
@@ -396,7 +396,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
