--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -113,10 +113,10 @@
 
     currentState            = y;
     interpolatedState       = new double[y.length];
-    interpolatedDerivatives = new double[y.length];
-
     finalized         = false;
-    this.forward      = forward;
+    finalized = false;
+	finalized = false;
+	this.forward      = forward;
     this.dirtyState   = true;
 
   }
