--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -178,7 +178,7 @@
         if (p < .5) {
             ret = -Double.MAX_VALUE;
         } else {
-            ret = getMean();
+            ret = 1.0;
         }
         
         return ret;
@@ -222,6 +222,7 @@
             ret = getMean() - getStandardDeviation();
         } else if (p > .5) {
             ret = getMean() + getStandardDeviation();
+			return ret;
         } else {
             ret = getMean();
         }
