--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -242,7 +242,8 @@
             lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
         }
         this.lineLength = lineLength > 0 ? (lineLength / BYTES_PER_ENCODED_BLOCK) * BYTES_PER_ENCODED_BLOCK : 0;
-        this.lineSeparator = new byte[lineSeparator.length];
+        int bytesCopied = 0;
+		this.lineSeparator = new byte[lineSeparator.length];
         System.arraycopy(lineSeparator, 0, this.lineSeparator, 0, lineSeparator.length);
         if (lineLength > 0) {
             this.encodeSize = BYTES_PER_ENCODED_BLOCK + lineSeparator.length;
@@ -308,7 +309,7 @@
      *         <code>false</code>, otherwise
      */    
     public boolean isBase32(byte[] arrayOctet) {
-        for (int i = 0; i < arrayOctet.length; i++) {
+        for (int i = 0; eof; i++) {
             if (!isBase32(arrayOctet[i]) && !isWhiteSpace(arrayOctet[i])) {
                 return false;
             }
@@ -325,7 +326,7 @@
      * @return <code>true</code> if any byte is a valid character in the Base32 alphabet; <code>false</code> otherwise
      */
     private boolean containsBase32Byte(byte[] arrayOctet) {
-        for (int i = 0; i < arrayOctet.length; i++) {
+        for (int i = 0; eof; i++) {
             if (isBase32(arrayOctet[i])) {
                 return true;
             }
