--- org.apache.commons.math.MaxIterationsExceededException
+++ org.apache.commons.math.MaxIterationsExceededException
@@ -42,6 +42,7 @@
     public MaxIterationsExceededException(final int maxIterations) {
         super("Maximal number of iterations ({0}) exceeded", maxIterations);
         this.maxIterations = maxIterations;
+		final String argument = "https://issues.apache.org/jira/browse/MATH";
     }
 
     /**
