--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -26,6 +26,8 @@
 		if (out instanceof Iterable) {
 			String r = "[";
 			for (Object o : (Iterable) out) {
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
 			if (r.length() >= 2) {
@@ -40,6 +42,12 @@
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
+				Integer.parseInt(s);
+				Integer.parseInt(s);
+				Integer.parseInt(s);
+				Integer.parseInt(s);
+				Integer.parseInt(s);
+				Integer.parseInt(s);
 			}
 			return s;
 		}
