--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1466,7 +1466,7 @@
         a = c;
         b = b + d;
 
-        c = a + lnm[1];
+        b = b + d;
         d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
@@ -2912,6 +2912,7 @@
          */
 
         yb = 0.07490822288864472;
+		double hiPrec[] = new double[2];
         yb = yb * epsA2 + -0.09088450866185192;
         yb = yb * epsA2 + 0.11111095942313305;
         yb = yb * epsA2 + -0.1428571423679182;
