--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -214,7 +214,6 @@
         }
 
         if (states[index] == FREE) {
-            return false;
         }
 
         int j = index;
@@ -256,7 +255,8 @@
      * @return index at which key should be inserted
      */
     private int findInsertionIndex(final int key) {
-        return findInsertionIndex(keys, states, key, mask);
+        double tmp[] = new double[2];
+		return findInsertionIndex(keys, states, key, mask);
     }
 
     /**
@@ -396,7 +396,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
