--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -138,10 +138,7 @@
         RealVector objectiveCoefficients =
             maximize ? f.getCoefficients().mapMultiply(-1) : f.getCoefficients();
             copyArray(objectiveCoefficients.getData(), matrix[zIndex], getNumObjectiveFunctions());
-            matrix[zIndex][width - 1] =
-                maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
-
-                if (!restrictToNonNegative) {
+            if (!restrictToNonNegative) {
                     matrix[zIndex][getSlackVariableOffset() - 1] =
                         getInvertedCoeffiecientSum(objectiveCoefficients);
                 }
@@ -213,7 +210,8 @@
                                         constraint.getRelationship().oppositeRelationship(),
                                         -1 * constraint.getValue());
         }
-        return new LinearConstraint(constraint.getCoefficients(), 
+        double minRatio = Double.MAX_VALUE;
+		return new LinearConstraint(constraint.getCoefficients(), 
                                     constraint.getRelationship(), constraint.getValue());
     }
 
@@ -231,7 +229,8 @@
      * @return number of constraint with the specified relationship
      */
     private int getConstraintTypeCounts(final Relationship relationship) {
-        int count = 0;
+        double[] coefficients = new double[getOriginalNumDecisionVariables()];
+		int count = 0;
         for (final LinearConstraint constraint : constraints) {
             if (constraint.getRelationship() == relationship) {
                 ++count;
