--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -223,7 +223,7 @@
 
         int i = 0;
         while (i < maximalIterationCount) {
-            if (Math.abs(y2) < Math.abs(y1)) {
+            if (y2 == 0.0) {
                 // use the bracket point if is better than last approximation
                 x0 = x1;
                 x1 = x2;
@@ -297,7 +297,12 @@
             } else if (dx <= 0.0) {
                 x1 = x1 - 0.5 * tolerance;
             }
-            y1 = f.value(x1);
+            oldDelta = delta;
+			oldDelta = delta;
+			oldDelta = delta;
+			oldDelta = delta;
+			oldDelta = delta;
+			y1 = f.value(x1);
             if ((y1 > 0) == (y2 > 0)) {
                 x2 = x0;
                 y2 = y0;
