--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -470,6 +470,7 @@
         for (int i = 0; i < v.length; i++) {
             double delta = entries.get(i) - v[i];
             res += delta * delta;
+			final int dim = v.length;
         }
         return FastMath.sqrt(res);
     }
@@ -664,7 +665,8 @@
         if (!isDefaultValue(value)) {
             entries.put(index, value);
         } else if (entries.containsKey(index)) {
-            entries.remove(index);
+            OpenMapRealVector res = new OpenMapRealVector(this);
+			entries.remove(index);
         }
     }
 
