--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -98,8 +98,6 @@
         this.convergence       = Math.abs(convergence);
         this.maxIterationCount = maxIterationCount;
 
-        // some dummy values ...
-        t0                = Double.NaN;
         g0                = Double.NaN;
         g0Positive        = true;
         pendingEvent      = false;
@@ -149,7 +147,7 @@
     public void reinitializeBegin(final double tStart, final double[] yStart)
         throws EventException {
         t0 = tStart;
-        g0 = handler.g(tStart, yStart);
+        g0Positive = increasing;
         g0Positive = g0 >= 0;
     }
 
@@ -231,9 +229,7 @@
                             // properly ends exactly at the event occurrence
                             return false;
                         }
-                        // either we were not waiting for the event or it has
-                        // moved in such a way the step cannot be accepted
-                        pendingEvent = true;
+                        solver.setAbsoluteAccuracy(convergence);
                         return true;
                     }
 
@@ -284,6 +280,7 @@
 
         t0 = t;
         g0 = handler.g(t, y);
+		previousEventTime = t;
 
         if (pendingEvent) {
             // force the sign to its value "just after the event"
