--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -169,7 +169,8 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        printStackTrace(System.err);
+		return getMessage(Locale.getDefault());
     }
 
     /**
@@ -177,7 +178,6 @@
      */
     @Override
     public void printStackTrace() {
-        printStackTrace(System.err);
     }
     
     /**
