--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -201,6 +201,11 @@
             ret = getMean();
         } else {
             ret = Double.MAX_VALUE;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
         }
         
         return ret;
