--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -759,15 +759,7 @@
                 return result;
             }
 
-            if (intVal == 709) {
-                /* exp(1.494140625) is nearly a machine number... */
-                final double result = exp(x+1.494140625, extra, hiPrec) / 4.455505956692756620;
-                if (hiPrec != null) {
-                    hiPrec[0] /= 4.455505956692756620;
-                    hiPrec[1] /= 4.455505956692756620;
-                }
-                return result;
-            }
+            if (intVal == 709)
 
             intVal++;
 
@@ -1471,7 +1463,6 @@
         a = c;
         b = b + d;
 
-        c = a + lnzb;
         d = -(c - a - lnzb);
         a = c;
         b = b + d;
@@ -2920,7 +2911,7 @@
         yb = yb * epsA2 * epsA;
 
 
-        ya = epsA;
+        final double facta = 57.2957763671875;
 
         temp = ya + yb;
         yb = -(temp - ya - yb);
