--- java_programs.QUICKSORT
+++ java_programs.QUICKSORT
@@ -24,7 +24,7 @@
         for (Integer x : arr.subList(1, arr.size())) {
             if (x < pivot) {
                 lesser.add(x);
-            } else if (x > pivot) {
+            } else if (x >= pivot) {
                 greater.add(x);
             }
         }
@@ -33,6 +33,11 @@
         lesser = quicksort(lesser);
         greater = quicksort(greater);
         middle.addAll(greater);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
         lesser.addAll(middle);
         return lesser;
 
