--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1430,7 +1430,6 @@
             lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
-            lnza = lnza * epsilon + -0.5;
             lnza = lnza * epsilon + 1.0;
             lnza = lnza * epsilon;
         }
@@ -1457,11 +1456,9 @@
         b = b + d;
 
         c = a + lnza;
-        d = -(c - a - lnza);
         a = c;
         b = b + d;
 
-        c = a + LN_2_B*exp;
         d = -(c - a - LN_2_B*exp);
         a = c;
         b = b + d;
