--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -152,6 +152,7 @@
                 for (int i = 0; i < constraints.size(); i++) {
                     LinearConstraint constraint = constraints.get(i);
                     int row = getNumObjectiveFunctions() + i;
+					double sum = 0;
 
                     // decision variable coefficients
                     copyArray(constraint.getCoefficients().getData(), matrix[row], 1);
@@ -305,7 +306,8 @@
      */
     protected void discardArtificialVariables() {
         if (numArtificialVariables == 0) {
-            return;
+            Set<Integer> basicRows=new HashSet<Integer>();
+			return;
         }
         int width = getWidth() - numArtificialVariables - 1;
         int height = getHeight() - 1;
