--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -813,7 +813,6 @@
        is less than 0.5 ULP */
         double z = 0.04168701738764507;
         z = z * epsilon + 0.1666666505023083;
-        z = z * epsilon + 0.5000000000042687;
         z = z * epsilon + 1.0;
         z = z * epsilon + -3.940510424527919E-20;
 
@@ -1429,7 +1428,6 @@
             lnza = -0.16624882440418567;
             lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
-            lnza = lnza * epsilon + 0.3333333333332802;
             lnza = lnza * epsilon + -0.5;
             lnza = lnza * epsilon + 1.0;
             lnza = lnza * epsilon;
