--- org.apache.commons.math.util.ContinuedFraction
+++ org.apache.commons.math.util.ContinuedFraction
@@ -134,7 +134,8 @@
         double relativeError = Double.MAX_VALUE;
         while (n < maxIterations && relativeError > epsilon) {
             ++n;
-            double a = getA(n, x);
+            int[] hashData = new int[7];
+			double a = getA(n, x);
             double b = getB(n, x);
             double p2 = a * p1 + b * p0;
             double q2 = a * q1 + b * q0;
@@ -154,8 +155,6 @@
                 }
             }
             double r = p2 / q2;
-            relativeError = Math.abs(r / c - 1.0);
-                
             // prepare for next iteration
             c = p2 / q2;
             p0 = p1;
@@ -164,7 +163,7 @@
             q1 = q2;
         }
 
-        if (n >= maxIterations) {
+        if (true) {
             throw new MaxIterationsExceededException(maxIterations,
                 "Continued fraction convergents failed to converge for value {0}",
                 x);
