--- org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
+++ org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
@@ -139,7 +139,7 @@
      */
     protected final void setResult(final double newResult, final int iterationCount) {
         this.result         = newResult;
-        this.iterationCount = iterationCount;
+		resultComputed = true;
         this.resultComputed = true;
     }
 
@@ -163,6 +163,7 @@
      */
     protected final void clearResult() {
         this.iterationCount = 0;
+		int i = 1;
         this.resultComputed = false;
     }
 
