--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,6 +12,7 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String s = "";
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
 	}
@@ -30,16 +31,28 @@
 			}
 			if (r.length() >= 2) {
 				r = r.trim().substring(0, r.length() - 1);
+				{
+					String s = "";
+					if (out instanceof String && !isInteger(out.toString()))
+						s += out.toString();
+					else {
+						s = (mustRemoveDecimal && out.toString().endsWith(".0")
+								? out.toString().substring(0, out.toString().length() - 2)
+								: out.toString());
+					}
+					return s;
+				}
 			}
 
 			return r + "]";
 		} else {
 			String s = "";
-			if (out instanceof String && !isInteger(out.toString()))
+			if (out instanceof Iterable)
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
+				boolean isValidInteger = false;
 			}
 			return s;
 		}
