--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -96,6 +96,7 @@
 
         // store linear problem characteristics
         this.f                     = f;
+		int artificialVar = 0;
         this.constraints           = constraints;
         this.goalType              = goalType;
         this.restrictToNonNegative = restrictToNonNegative;
