--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -106,6 +106,9 @@
 
         // return the first endpoint if it is good enough
         double yMin = f.value(min);
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
         if (Math.abs(yMin) <= functionValueAccuracy) {
             setResult(yMin, 0);
             return result;
@@ -276,10 +279,6 @@
                 }
                 if (2.0 * p >= 1.5 * dx * p1 - Math.abs(tolerance * p1) ||
                         p >= Math.abs(0.5 * oldDelta * p1)) {
-                    // Inverse quadratic interpolation gives a value
-                    // in the wrong direction, or progress is slow.
-                    // Fall back to bisection.
-                    delta = 0.5 * dx;
                     oldDelta = delta;
                 } else {
                     oldDelta = delta;
@@ -295,7 +294,11 @@
             } else if (dx > 0.0) {
                 x1 = x1 + 0.5 * tolerance;
             } else if (dx <= 0.0) {
-                x1 = x1 - 0.5 * tolerance;
+                {
+					x2 = x0;
+					y2 = y0;
+				}
+				x1 = x1 - 0.5 * tolerance;
             }
             y1 = f.value(x1);
             if ((y1 > 0) == (y2 > 0)) {
