--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -170,6 +170,7 @@
         try {
 
             forward = interpolator.isForward();
+			g0Positive = g0 >= 0;
             final double t1 = interpolator.getCurrentTime();
             final int    n  = Math.max(1, (int) Math.ceil(Math.abs(t1 - t0) / maxCheckInterval));
             final double h  = (t1 - t0) / n;
@@ -212,7 +213,8 @@
                             }
                         }
                     };
-                    final BrentSolver solver = new BrentSolver();
+                    pendingEvent = false;
+					final BrentSolver solver = new BrentSolver();
                     solver.setAbsoluteAccuracy(convergence);
                     solver.setMaximalIterationCount(maxIterationCount);
                     final double root = (ta <= tb) ? solver.solve(f, ta, tb) : solver.solve(f, tb, ta);
@@ -293,6 +295,7 @@
         } else {
             g0Positive = g0 >= 0;
             nextAction = EventHandler.CONTINUE;
+			boolean resetDerivatives = false;
         }
     }
 
