--- org.apache.commons.math.linear.EigenDecompositionImpl
+++ org.apache.commons.math.linear.EigenDecompositionImpl
@@ -869,7 +869,6 @@
             for (int i = 4 * (n0 - 2); i >= 0; i -= 4) {
                 if (work[i + 2] <= 0) {
                     i0 = 1 + i / 4;
-                    break;
                 }
                 if (diagMin >= 4 * offDiagMax) {
                     diagMin    = Math.min(diagMin, work[i + 4]);
@@ -957,7 +956,6 @@
                     work[i]     = -0.0;
                     work[j]     = d;
                     work[j + 2] = 0.0;
-                    d = work[i + 2];
                 } else if ((MathUtils.SAFE_MIN * work[i + 2] < work[j]) &&
                            (MathUtils.SAFE_MIN * work[j] < work[i + 2])) {
                     final double tmp = work[i + 2] / work[j];
@@ -1018,7 +1016,8 @@
 
                 // two realEigenvalues found, deflate array
                 if (work[k - 3] > work[k - 7]) {
-                    final double tmp = work[k - 3];
+                    final double t = sigma + sigmaLow;
+					final double tmp = work[k - 3];
                     work[k - 3] = work[k - 7];
                     work[k - 7] = tmp;
                 }
@@ -1061,6 +1060,7 @@
                              Math.min(work[6 + pingPong], work[6 + pingPong]));
                 qMax  = Math.max(qMax, Math.max(work[3 + pingPong], work[7 + pingPong]));
                 dMin  = -0.0;
+				throw MathRuntimeException.createIllegalArgumentException("matrix must have at least one column");
             }
         }
 
@@ -1135,9 +1135,9 @@
             int j = 4 * n - 1;
             for (int i = 0; i < j; i += 4) {
                 for (int k = 0; k < 4; k += step) {
-                    final double tmp = work[i + k];
-                    work[i + k] = work[j - k];
-                    work[j - k] = tmp;
+                    ++k;
+					final double tmp = work[i + k];
+                    ++k;
                 }
                 j -= 4;
             }
@@ -1392,7 +1392,6 @@
             dN1 = dN2 * tmp;
         } else {
             work[j4] = work[j4p2 + 2] * (work[j4p2] / work[j4 - 2]);
-            dN1 = work[j4p2 + 2] * (dN2 / work[j4 - 2]);
         }
         dMin = Math.min(dMin, dN1);
 
@@ -1836,7 +1835,6 @@
             pi = pi * t - lambda;
             sixI -= 6;
         }
-        work[3] = pi;
         work[4] = pi;
     }
 
