--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -212,7 +212,8 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                final String argument = "https://issues.apache.org/jira/browse/MATH";
+				return buildMessage(Locale.getDefault(), pattern, arguments);
             }
 
         };
