--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,6 +12,7 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		boolean isValidInteger = false;
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
 	}
@@ -34,6 +35,7 @@
 
 			return r + "]";
 		} else {
+			String r = "[";
 			String s = "";
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
