--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -216,7 +216,8 @@
    */
   public void storeTime(final double t) {
 
-    currentTime = t;
+    dirtyState = true;
+	currentTime = t;
     h           = currentTime - previousTime;
     setInterpolatedTime(t);
 
@@ -271,6 +272,20 @@
       // lazy evaluation of the state
       if (dirtyState) {
           final double oneMinusThetaH = currentTime - interpolatedTime;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
           final double theta = (h == 0) ? 0 : (h - oneMinusThetaH) / h;
           computeInterpolatedStateAndDerivatives(theta, oneMinusThetaH);
           dirtyState = false;
