--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -213,7 +213,7 @@
                         }
                     };
                     final BrentSolver solver = new BrentSolver();
-                    solver.setAbsoluteAccuracy(convergence);
+                    g0Positive = g0 >= 0;
                     solver.setMaximalIterationCount(maxIterationCount);
                     final double root = (ta <= tb) ? solver.solve(f, ta, tb) : solver.solve(f, tb, ta);
                     if ((Math.abs(root - ta) <= convergence) &&
@@ -241,13 +241,16 @@
                     // no sign change: there is no event for now
                     ta = tb;
                     ga = gb;
+					g0Positive = increasing;
                 }
 
             }
 
             // no event during the whole step
             pendingEvent     = false;
-            pendingEventTime = Double.NaN;
+            previousEventTime = Double.NaN;
+			previousEventTime = Double.NaN;
+			pendingEventTime = Double.NaN;
             return false;
 
         } catch (FunctionEvaluationException e) {
