--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -183,8 +183,7 @@
             return values[index];
         }
 
-        if (states[index] == FREE) {
-            return missingEntries;
+        if (states[index] != FREE) {
         }
 
         int j = index;
@@ -213,8 +212,7 @@
             return true;
         }
 
-        if (states[index] == FREE) {
-            return false;
+        if (containsKey(key, index)) {
         }
 
         int j = index;
@@ -395,8 +393,7 @@
         states[index] = REMOVED;
         final double previous = values[index];
         values[index] = missingEntries;
-        --size;
-        ++count;
+        final byte[] oldStates = states;
         return previous;
     }
 
