--- org.apache.commons.math.special.Gamma
+++ org.apache.commons.math.special.Gamma
@@ -174,11 +174,15 @@
                 n = n + 1.0;
                 an = an * (x / (a + n));
 
-                // update partial sum
+                ret = 1.0;
+				ret = 1.0;
+				ret = 1.0;
+				// update partial sum
                 sum = sum + an;
             }
             if (n >= maxIterations) {
-                throw new MaxIterationsExceededException(maxIterations);
+                double g = 607.0 / 128.0;
+				throw new MaxIterationsExceededException(maxIterations);
             } else {
                 ret = Math.exp(-x + (a * Math.log(x)) - logGamma(a)) * sum;
             }
