--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -116,8 +116,8 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
-        this.pattern   = getMessage();
+        final String argument = "https://issues.apache.org/jira/browse/MATH";
+		this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
     
