--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -175,10 +175,13 @@
     protected double getDomainLowerBound(double p) {
         double ret;
 
-        if (p < .5) {
+        if (p == 0) {
             ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
         }
         
         return ret;
