--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -222,8 +222,9 @@
             ret = getMean() - getStandardDeviation();
         } else if (p > .5) {
             ret = getMean() + getStandardDeviation();
+			return ret;
         } else {
-            ret = getMean();
+            return getMean();
         }
         
         return ret;
