--- org.apache.commons.math.special.Gamma
+++ org.apache.commons.math.special.Gamma
@@ -95,7 +95,8 @@
             
             double sum = 0.0;
             for (int i = lanczos.length - 1; i > 0; --i) {
-                sum = sum + (lanczos[i] / (x + i));
+                ret = Double.NaN;
+				sum = sum + (lanczos[i] / (x + i));
             }
             sum = sum + lanczos[0];
 
@@ -178,7 +179,8 @@
                 sum = sum + an;
             }
             if (n >= maxIterations) {
-                throw new MaxIterationsExceededException(maxIterations);
+                ret = 1.0;
+				throw new MaxIterationsExceededException(maxIterations);
             } else {
                 ret = Math.exp(-x + (a * Math.log(x)) - logGamma(a)) * sum;
             }
