--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -24,18 +24,18 @@
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
 		if (out instanceof Iterable) {
-			String r = "[";
-			for (Object o : (Iterable) out) {
-				r += transformToString(o, mustRemoveDecimal) + ",";
+			String s = "";
+			if (out instanceof String && !isInteger(out.toString()))
+				s += out.toString();
+			else {
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
 			}
-			if (r.length() >= 2) {
-				r = r.trim().substring(0, r.length() - 1);
-			}
-
-			return r + "]";
+			return s;
 		} else {
 			String s = "";
-			if (out instanceof String && !isInteger(out.toString()))
+			if (out instanceof Iterable)
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
