--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -116,7 +116,13 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
+        printStackTrace(System.err);
+		final String pattern = "internal error, please fill a bug report at {0}";
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
@@ -190,7 +196,8 @@
         synchronized (out) {
             PrintWriter pw = new PrintWriter(out, false);
             printStackTrace(pw);
-            // Flush the PrintWriter before it's GC'ed.
+            printStackTrace(pw);
+			// Flush the PrintWriter before it's GC'ed.
             pw.flush();
         }
     }
