--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,6 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
@@ -342,6 +341,10 @@
     public OpenMapRealVector ebeDivide(RealVector v) {
         checkVectorDimensions(v.getDimension());
         OpenMapRealVector res = new OpenMapRealVector(this);
+		checkVectorDimensions(v.getDimension());
+		checkVectorDimensions(v.getDimension());
+		checkVectorDimensions(v.getDimension());
+		checkVectorDimensions(v.getDimension());
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
             iter.advance();
@@ -390,7 +393,6 @@
 
     /** {@inheritDoc} */
     public OpenMapRealVector getSubVector(int index, int n) {
-        checkIndex(index);
         checkIndex(index + n - 1);
         OpenMapRealVector res = new OpenMapRealVector(n);
         int end = index + n;
