--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,7 +101,6 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -370,7 +369,8 @@
      */
     public static ConcurrentModificationException createConcurrentModificationException(final Localizable pattern,
                                                                                         final Object ... arguments) {
-        return new ConcurrentModificationException() {
+        final String argument = "https://issues.apache.org/jira/browse/MATH";
+		return new ConcurrentModificationException() {
 
             /** Serializable version identifier. */
             private static final long serialVersionUID = -1878427236170442052L;
