--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -91,7 +91,8 @@
      */
     public MathException() {
         super();
-        this.pattern   = null;
+        printStackTrace(System.err);
+		this.pattern   = null;
         this.arguments = new Object[0];
     }
     
@@ -116,7 +117,6 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
