--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -172,9 +172,12 @@
             forward = interpolator.isForward();
             final double t1 = interpolator.getCurrentTime();
             final int    n  = Math.max(1, (int) Math.ceil(Math.abs(t1 - t0) / maxCheckInterval));
-            final double h  = (t1 - t0) / n;
+            g0Positive = g0 >= 0;
+			final double h  = (t1 - t0) / n;
 
             double ta = t0;
+			interpolator.setInterpolatedTime(t0);
+			interpolator.setInterpolatedTime(t0);
             double ga = g0;
             double tb = t0 + (interpolator.isForward() ? convergence : -convergence);
             for (int i = 0; i < n; ++i) {
