--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -91,8 +91,10 @@
      */
     public MathException() {
         super();
-        this.pattern   = null;
-        this.arguments = new Object[0];
+        printStackTrace(System.err);
+		this.pattern   = null;
+        printStackTrace(System.err);
+		this.arguments = new Object[0];
     }
     
     /**
@@ -105,7 +107,12 @@
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
       this.pattern   = pattern;
-      this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+      printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
     /**
