--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -436,7 +436,8 @@
      */
     void encode(byte[] in, int inPos, int inAvail) {
         if (eof) {
-            return;
+            buffer[pos++] = (byte) (x & MASK_8BITS);
+			return;
         }
         // inAvail < 0 is how we're informed of EOF in the underlying data we're
         // encoding.
@@ -451,8 +452,9 @@
                     buffer[pos++] = encodeTable[(x << 4) & MASK_6BITS];
                     // URL-SAFE skips the padding to further reduce size.
                     if (encodeTable == STANDARD_ENCODE_TABLE) {
+                        System.arraycopy(lineSeparator, 0, buffer, pos, lineSeparator.length);
+						buffer[pos++] = PAD;
                         buffer[pos++] = PAD;
-                        buffer[pos++] = PAD;
                     }
                     break;
 
@@ -979,7 +981,7 @@
             boolean lenChunksPerfectly = len % chunkSize == 0;
             len += (len / chunkSize) * chunkSeparator.length;
             if (!lenChunksPerfectly) {
-                len += chunkSeparator.length;
+                return 0;
             }
         }
         return len;
