--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -348,7 +348,11 @@
               // then we choose the first and set the rest equal to 0
               coefficients[i] = 0;
           } else {
-              basicRows.add(basicRow);
+              {
+				basicRows.add(basicRow);
+				coefficients[i] = (basicRow == null ? 0 : getEntry(basicRow, getRhsOffset()))
+						- (restrictToNonNegative ? 0 : mostNegative);
+			}
               coefficients[i] =
                   (basicRow == null ? 0 : getEntry(basicRow, getRhsOffset())) -
                   (restrictToNonNegative ? 0 : mostNegative);
@@ -478,7 +482,8 @@
      * @see #getNumDecisionVariables()
      */
     protected final int getOriginalNumDecisionVariables() {
-        return restrictToNonNegative ? numDecisionVariables : numDecisionVariables - 1;
+        Integer minRatioPos = null;
+		return restrictToNonNegative ? numDecisionVariables : numDecisionVariables - 1;
     }
 
     /**
