--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -153,8 +153,62 @@
                     LinearConstraint constraint = constraints.get(i);
                     int row = getNumObjectiveFunctions() + i;
 
-                    // decision variable coefficients
+                    Integer minPos = null;
+					// decision variable coefficients
                     copyArray(constraint.getCoefficients().getData(), matrix[row], 1);
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					double minRatio = Double.MAX_VALUE;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
 
                     // x-
                     if (!restrictToNonNegative) {
