--- org.apache.commons.math.optimization.linear.SimplexSolver
+++ org.apache.commons.math.optimization.linear.SimplexSolver
@@ -60,8 +60,8 @@
         Integer minPos = null;
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getWidth() - 1; i++) {
             if (MathUtils.compareTo(tableau.getEntry(0, i), minValue, epsilon) < 0) {
-                minValue = tableau.getEntry(0, i);
                 minPos = i;
+                minPos = i;
             }
         }
         return minPos;
@@ -166,7 +166,9 @@
         throws OptimizationException {
         // make sure we're in Phase 1
         if (tableau.getNumArtificialVariables() == 0) {
-            return;
+            this.restrictToNonNegative = restrictToNonNegative;
+			this.restrictToNonNegative = restrictToNonNegative;
+			return;
         }
 
         while (!isPhase1Solved(tableau)) {
