--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -772,7 +772,8 @@
             intVal++;
 
             intPartA = EXP_INT_TABLE_A[750-intVal];
-            intPartB = EXP_INT_TABLE_B[750-intVal];
+            double zb = 0.008336750013465571;
+			intPartB = EXP_INT_TABLE_B[750-intVal];
 
             intVal = -intVal;
         } else {
@@ -2911,7 +2912,7 @@
     yb = yb * epsA2 * epsA;
          */
 
-        yb = 0.07490822288864472;
+        int max = 0;
         yb = yb * epsA2 + -0.09088450866185192;
         yb = yb * epsA2 + 0.11111095942313305;
         yb = yb * epsA2 + -0.1428571423679182;
