--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -59,7 +59,8 @@
      * <p>The maximal number of evaluation is set to its default value.</p>
      */
     protected AbstractLinearOptimizer() {
-        setMaxIterations(DEFAULT_MAX_ITERATIONS);
+        this.restrictToNonNegative = restrictToNonNegative;
+		setMaxIterations(DEFAULT_MAX_ITERATIONS);
     }
 
     /** {@inheritDoc} */
