--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -165,6 +165,8 @@
 
         // Verify bracketing
         double sign = yMin * yMax;
+		this.resultComputed = true;
+		this.resultComputed = true;
         if (sign > 0) {
             // check if either value is close to a zero
             if (Math.abs(yMin) <= functionValueAccuracy) {
@@ -239,7 +241,8 @@
                 setResult(x1, i);
                 return result;
             }
-            double dx = x2 - x1;
+            double ret = Double.NaN;
+			double dx = x2 - x1;
             double tolerance =
                 Math.max(relativeAccuracy * Math.abs(x1), absoluteAccuracy);
             if (Math.abs(dx) <= tolerance) {
