--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1427,10 +1427,10 @@
             /* High precision not required.  Eval Remez polynomial
          using standard double precision */
             lnza = -0.16624882440418567;
-            lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
-            lnza = lnza * epsilon + -0.5;
+            double a = LN_2_A * exp;
+			lnza = lnza * epsilon + -0.5;
             lnza = lnza * epsilon + 1.0;
             lnza = lnza * epsilon;
         }
@@ -2884,7 +2884,6 @@
             final double yaa = (ya + temp) - temp;
             final double yab = ya - yaa;
 
-            temp = za * HEX_40000000;
             final double zaa = (za + temp) - temp;
             final double zab = za - zaa;
 
@@ -2949,7 +2948,8 @@
 
             za = pia - result;
             zb = -(za - pia + result);
-            zb += pib - resultb;
+            double as[] = new double[2];
+			zb += pib - resultb;
 
             result = za + zb;
             resultb = -(result - za - zb);
