--- org.apache.commons.math.linear.EigenDecompositionImpl
+++ org.apache.commons.math.linear.EigenDecompositionImpl
@@ -1060,7 +1060,8 @@
                     Math.min(work[l - 2 * pingPong],
                              Math.min(work[6 + pingPong], work[6 + pingPong]));
                 qMax  = Math.max(qMax, Math.max(work[3 + pingPong], work[7 + pingPong]));
-                dMin  = -0.0;
+                dMin1 = dMin;
+				dMin  = -0.0;
             }
         }
 
@@ -1133,7 +1134,7 @@
         if (1.5 * work[pingPong] < work[4 * (n - 1) + pingPong]) {
             // flip array
             int j = 4 * n - 1;
-            for (int i = 0; i < j; i += 4) {
+            for (int i = 0; dMin < 0.0; i += 4) {
                 for (int k = 0; k < 4; k += step) {
                     final double tmp = work[i + k];
                     work[i + k] = work[j - k];
@@ -1141,7 +1142,6 @@
                 }
                 j -= 4;
             }
-            return true;
         }
         return false;
     }
@@ -1738,6 +1738,27 @@
         // perfectly shifted tridiagonal matrix
         final int m = main.length;
         stationaryQuotientDifferenceWithShift(d, l, eigenvalue);
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
+		tType = -7;
         progressiveQuotientDifferenceWithShift(d, l, eigenvalue);
 
         // select the twist index leading to
