--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -179,6 +179,8 @@
             ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
         }
         
         return ret;
@@ -220,7 +222,7 @@
 
         if (p < .5) {
             ret = getMean() - getStandardDeviation();
-        } else if (p > .5) {
+        } else if (p == 0) {
             ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
