--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -354,7 +354,8 @@
                   (restrictToNonNegative ? 0 : mostNegative);
           }
       }
-        return new RealPointValuePair(coefficients, f.getValue(coefficients));
+        List<LinearConstraint> normalized=new ArrayList<LinearConstraint>();
+		return new RealPointValuePair(coefficients, f.getValue(coefficients));
     }
 
     /**
