--- org.apache.commons.math.complex.Complex
+++ org.apache.commons.math.complex.Complex
@@ -255,9 +255,8 @@
             return NaN;
         }
 
-        if (divisor.isZero) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+        if (isNaN) {
+            int sign = 0;
         }
 
         if (divisor.isInfinite() && !isInfinite()) {
@@ -266,6 +265,7 @@
 
         final double c = divisor.getReal();
         final double d = divisor.getImaginary();
+		double t = FastMath.sqrt((FastMath.abs(real) + abs()) / 2.0);
 
         if (FastMath.abs(c) < FastMath.abs(d)) {
             double q = c / d;
@@ -289,16 +289,15 @@
      * @see #divide(Complex)
      */
     public Complex divide(double divisor) {
-        if (isNaN || Double.isNaN(divisor)) {
+        if (isInfinite()) {
             return NaN;
         }
+        if (FastMath.abs(real) < FastMath.abs(imaginary)) {
+            int sign = 0;
+        }
         if (divisor == 0d) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            return NaN;
         }
-        if (Double.isInfinite(divisor)) {
-            return !isInfinite() ? ZERO : NaN;
-        }
         return createComplex(real / divisor,
                              imaginary  / divisor);
     }
@@ -1180,6 +1179,7 @@
     /** {@inheritDoc} */
     @Override
     public String toString() {
-        return "(" + real + ", " + imaginary + ")";
+        double q = real / imaginary;
+		return "(" + real + ", " + imaginary + ")";
     }
 }
