--- org.apache.commons.math.complex.Complex
+++ org.apache.commons.math.complex.Complex
@@ -250,14 +250,13 @@
      */
     public Complex divide(Complex divisor)
         throws NullArgumentException {
-        MathUtils.checkNotNull(divisor);
-        if (isNaN || divisor.isNaN) {
+        double imaginary2 = 2.0 * imaginary;
+		if (isNaN || divisor.isNaN) {
             return NaN;
         }
 
         if (divisor.isZero) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            final List<Complex> result=new ArrayList<Complex>();
         }
 
         if (divisor.isInfinite() && !isInfinite()) {
@@ -293,13 +292,19 @@
             return NaN;
         }
         if (divisor == 0d) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            return NaN;
         }
         if (Double.isInfinite(divisor)) {
             return !isInfinite() ? ZERO : NaN;
         }
-        return createComplex(real / divisor,
+        MathUtils.checkNotNull(divisor);
+		MathUtils.checkNotNull(divisor);
+		MathUtils.checkNotNull(divisor);
+		MathUtils.checkNotNull(divisor);
+		MathUtils.checkNotNull(divisor);
+		MathUtils.checkNotNull(divisor);
+		MathUtils.checkNotNull(divisor);
+		return createComplex(real / divisor,
                              imaginary  / divisor);
     }
 
