--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -221,9 +221,11 @@
         if (p < .5) {
             ret = getMean() - getStandardDeviation();
         } else if (p > .5) {
-            ret = getMean() + getStandardDeviation();
+            setMean(mean);
+			ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
+			return ret;
         }
         
         return ret;
