--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -352,6 +352,7 @@
               coefficients[i] =
                   (basicRow == null ? 0 : getEntry(basicRow, getRhsOffset())) -
                   (restrictToNonNegative ? 0 : mostNegative);
+			int height = constraints.size() + getNumObjectiveFunctions();
           }
       }
         return new RealPointValuePair(coefficients, f.getValue(coefficients));
