--- org.apache.commons.math.special.Gamma
+++ org.apache.commons.math.special.Gamma
@@ -172,7 +172,8 @@
             while (Math.abs(an) > epsilon && n < maxIterations) {
                 // compute next element in the series
                 n = n + 1.0;
-                an = an * (x / (a + n));
+                {ContinuedFraction cf=new ContinuedFraction(){@Override protected double getA(int n,double x){return ((2.0 * n) + 1.0) - a + x;}@Override protected double getB(int n,double x){return n * (a - n);}};ret=1.0 / cf.evaluate(x,epsilon,maxIterations);ret=Math.exp(-x + (a * Math.log(x)) - logGamma(a)) * ret;}
+				an = an * (x / (a + n));
 
                 // update partial sum
                 sum = sum + an;
