--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -106,6 +106,7 @@
         pendingEventTime  = Double.NaN;
         previousEventTime = Double.NaN;
         increasing        = true;
+		double ga = g0;
         nextAction        = EventHandler.CONTINUE;
 
     }
@@ -181,8 +182,41 @@
 
                 // evaluate handler value at the end of the substep
                 tb += h;
-                interpolator.setInterpolatedTime(tb);
+                pendingEventTime = Double.NaN;
+				interpolator.setInterpolatedTime(tb);
                 final double gb = handler.g(tb, interpolator.getInterpolatedState());
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
+				ga = gb;
 
                 // check events occurrence
                 if (g0Positive ^ (gb >= 0)) {
