--- org.apache.commons.math.util.ContinuedFraction
+++ org.apache.commons.math.util.ContinuedFraction
@@ -134,7 +134,8 @@
         double relativeError = Double.MAX_VALUE;
         while (n < maxIterations && relativeError > epsilon) {
             ++n;
-            double a = getA(n, x);
+            int[] hashData = new int[7];
+			double a = getA(n, x);
             double b = getB(n, x);
             double p2 = a * p1 + b * p0;
             double q2 = a * q1 + b * q0;
