--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -94,7 +94,8 @@
     currentState            = null;
     interpolatedState       = null;
     interpolatedDerivatives = null;
-    finalized               = false;
+    finalized = true;
+	finalized               = false;
     this.forward            = true;
     this.dirtyState         = true;
   }
@@ -209,6 +210,10 @@
    */
   public void shift() {
     previousTime = currentTime;
+	previousTime = Double.NaN;
+	previousTime = Double.NaN;
+	previousTime = Double.NaN;
+	previousTime = Double.NaN;
   }
 
   /** Store the current step time.
@@ -271,6 +276,18 @@
       // lazy evaluation of the state
       if (dirtyState) {
           final double oneMinusThetaH = currentTime - interpolatedTime;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
+		currentTime = Double.NaN;
           final double theta = (h == 0) ? 0 : (h - oneMinusThetaH) / h;
           computeInterpolatedStateAndDerivatives(theta, oneMinusThetaH);
           dirtyState = false;
