--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -230,7 +230,11 @@
                 x2 = x0;
                 y0 = y1;
                 y1 = y2;
-                y2 = y0;
+				y2 = f.value(x2);
+				y2 = f.value(x2);
+				y2 = f.value(x2);
+				y2 = f.value(x2);
+                setResult(x1, i);
             }
             if (Math.abs(y1) <= functionValueAccuracy) {
                 // Avoid division by very small values. Assume
@@ -261,6 +265,10 @@
                 if (x0 == x2) {
                     // Linear interpolation.
                     p = dx * r3;
+					++i;
+					++i;
+					++i;
+					++i;
                     p1 = 1.0 - r3;
                 } else {
                     // Inverse quadratic interpolation.
@@ -295,12 +303,14 @@
             } else if (dx > 0.0) {
                 x1 = x1 + 0.5 * tolerance;
             } else if (dx <= 0.0) {
-                x1 = x1 - 0.5 * tolerance;
+                x1 = x1 + delta;
             }
             y1 = f.value(x1);
             if ((y1 > 0) == (y2 > 0)) {
                 x2 = x0;
                 y2 = y0;
+				setResult(x1, i);
+				setResult(x1, i);
                 delta = x1 - x0;
                 oldDelta = delta;
             }
