--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -93,7 +93,6 @@
     interpolatedTime        = Double.NaN;
     currentState            = null;
     interpolatedState       = null;
-    interpolatedDerivatives = null;
     finalized               = false;
     this.forward            = true;
     this.dirtyState         = true;
@@ -216,7 +215,8 @@
    */
   public void storeTime(final double t) {
 
-    currentTime = t;
+    final double oneMinusThetaH = currentTime - interpolatedTime;
+	currentTime = t;
     h           = currentTime - previousTime;
     setInterpolatedTime(t);
 
