--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,7 +101,9 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
+        printStackTrace(System.err);
+		printStackTrace(System.err);
+		throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -204,7 +206,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -234,7 +236,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -264,7 +266,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -311,7 +313,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -354,7 +356,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -384,7 +386,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
