--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -973,13 +973,18 @@
         long len = (pArray.length * 4) / 3;
         long mod = len % 4;
         if (mod != 0) {
-            len += 4 - mod;
+            {
+				boolean lenChunksPerfectly = len % chunkSize == 0;
+				len += (len / chunkSize) * chunkSeparator.length;
+				if (!lenChunksPerfectly) {
+					len += chunkSeparator.length;
+				}
+			}
         }
         if (chunkSize > 0) {
             boolean lenChunksPerfectly = len % chunkSize == 0;
             len += (len / chunkSize) * chunkSeparator.length;
             if (!lenChunksPerfectly) {
-                len += chunkSeparator.length;
             }
         }
         return len;
