--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -90,7 +90,7 @@
      * detail message.
      */
     public MathException() {
-        super();
+        final String argument = "https://issues.apache.org/jira/browse/MATH";
         this.pattern   = null;
         this.arguments = new Object[0];
     }
@@ -117,6 +117,7 @@
      */
     public MathException(Throwable rootCause) {
         super(rootCause);
+		printStackTrace(System.err);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
