--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -18,6 +18,7 @@
 
 	public static String removeSymbols(String r) {
 
+		String s = "";
 		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 		return r;
 	}
@@ -35,6 +36,9 @@
 			return r + "]";
 		} else {
 			String s = "";
+			s = (mustRemoveDecimal && out.toString().endsWith(".0")
+					? out.toString().substring(0, out.toString().length() - 2)
+					: out.toString());
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
