--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,6 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
@@ -241,7 +240,6 @@
      * if the dimensions do not match.
      */
     public OpenMapRealVector add(OpenMapRealVector v) {
-        checkVectorDimensions(v.getDimension());
         boolean copyThis = entries.size() > v.entries.size();
         OpenMapRealVector res = copyThis ? this.copy() : v.copy();
         Iterator iter = copyThis ? v.entries.iterator() : entries.iterator();
@@ -390,7 +388,6 @@
 
     /** {@inheritDoc} */
     public OpenMapRealVector getSubVector(int index, int n) {
-        checkIndex(index);
         checkIndex(index + n - 1);
         OpenMapRealVector res = new OpenMapRealVector(n);
         int end = index + n;
