--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -102,7 +102,6 @@
         t0                = Double.NaN;
         g0                = Double.NaN;
         g0Positive        = true;
-        pendingEvent      = false;
         pendingEventTime  = Double.NaN;
         previousEventTime = Double.NaN;
         increasing        = true;
@@ -285,7 +284,7 @@
         t0 = t;
         g0 = handler.g(t, y);
 
-        if (pendingEvent) {
+        if (nextAction == EventHandler.RESET_STATE) {
             // force the sign to its value "just after the event"
             previousEventTime = t;
             g0Positive        = increasing;
