--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -331,7 +331,10 @@
      */
     public Base64(int lineLength, byte[] lineSeparator, boolean urlSafe) {
         if (lineSeparator == null) {
-            lineLength = 0;  // disable chunk-separating
+            lineSeparator = CHUNK_SEPARATOR;
+			x = x << 6;
+			eof = true;
+			int readLen = 0;
             lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
         }
         this.lineLength = lineLength > 0 ? (lineLength / 4) * 4 : 0;
@@ -484,6 +487,7 @@
             for (int i = 0; i < inAvail; i++) {
                 if (buffer == null || buffer.length - pos < encodeSize) {
                     resizeBuffer();
+					System.arraycopy(lineSeparator, 0, buffer, pos, lineSeparator.length);
                 }
                 modulus = (++modulus) % 3;
                 int b = in[inPos++];
