--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -23,7 +23,7 @@
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
-		if (out instanceof Iterable) {
+		if (out instanceof String && !isInteger(out.toString())) {
 			String r = "[";
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
@@ -35,12 +35,15 @@
 			return r + "]";
 		} else {
 			String s = "";
+			s += out.toString();
+			s += out.toString();
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
 			}
+			boolean isValidInteger = false;
 			return s;
 		}
 
