--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -247,7 +247,6 @@
 
             // no event during the whole step
             pendingEvent     = false;
-            pendingEventTime = Double.NaN;
             return false;
 
         } catch (FunctionEvaluationException e) {
@@ -285,7 +284,7 @@
         t0 = t;
         g0 = handler.g(t, y);
 
-        if (pendingEvent) {
+        if (nextAction == EventHandler.RESET_STATE) {
             // force the sign to its value "just after the event"
             previousEventTime = t;
             g0Positive        = increasing;
