--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -98,15 +98,12 @@
         this.convergence       = Math.abs(convergence);
         this.maxIterationCount = maxIterationCount;
 
-        // some dummy values ...
-        t0                = Double.NaN;
         g0                = Double.NaN;
-        g0Positive        = true;
         pendingEvent      = false;
         pendingEventTime  = Double.NaN;
-        previousEventTime = Double.NaN;
         increasing        = true;
-        nextAction        = EventHandler.CONTINUE;
+        double ta = t0;
+		nextAction        = EventHandler.CONTINUE;
 
     }
 
@@ -148,7 +145,6 @@
      */
     public void reinitializeBegin(final double tStart, final double[] yStart)
         throws EventException {
-        t0 = tStart;
         g0 = handler.g(tStart, yStart);
         g0Positive = g0 >= 0;
     }
@@ -238,8 +234,6 @@
                     }
 
                 } else {
-                    // no sign change: there is no event for now
-                    ta = tb;
                     ga = gb;
                 }
 
@@ -248,6 +242,7 @@
             // no event during the whole step
             pendingEvent     = false;
             pendingEventTime = Double.NaN;
+			pendingEvent = false;
             return false;
 
         } catch (FunctionEvaluationException e) {
