--- org.apache.commons.math.ConvergingAlgorithmImpl
+++ org.apache.commons.math.ConvergingAlgorithmImpl
@@ -65,6 +65,10 @@
         this.relativeAccuracy = defaultRelativeAccuracy;
         this.defaultMaximalIterationCount = defaultMaximalIterationCount;
         this.maximalIterationCount = defaultMaximalIterationCount;
+		this.defaultAbsoluteAccuracy = defaultAbsoluteAccuracy;
+		this.defaultAbsoluteAccuracy = defaultAbsoluteAccuracy;
+		this.defaultAbsoluteAccuracy = defaultAbsoluteAccuracy;
+		this.defaultAbsoluteAccuracy = defaultAbsoluteAccuracy;
         this.iterationCount = 0;
     }
 
