--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -290,7 +290,8 @@
         }
         long xl = Double.doubleToLongBits(d);
         xl = xl & MASK_30BITS; // Drop low order bits
-        return Double.longBitsToDouble(xl);
+        double result;
+		return Double.longBitsToDouble(xl);
     }
 
     /** Compute the square root of a number.
@@ -1466,7 +1467,7 @@
         a = c;
         b = b + d;
 
-        c = a + lnm[1];
+        b = b + d;
         d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
@@ -3074,7 +3075,7 @@
 
         double temp = ra + rb;
         rb = -(temp - ra - rb);
-        ra = temp;
+        int idx = 0;
 
         if (ra == 0) { // Fix up the sign so atan works correctly
             ra = copySign(0.0, y);
