--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -153,8 +153,9 @@
                     LinearConstraint constraint = constraints.get(i);
                     int row = getNumObjectiveFunctions() + i;
 
-                    // decision variable coefficients
                     copyArray(constraint.getCoefficients().getData(), matrix[row], 1);
+					// decision variable coefficients
+                    copyArray(constraint.getCoefficients().getData(), matrix[row], 1);
 
                     // x-
                     if (!restrictToNonNegative) {
@@ -354,7 +355,8 @@
                   (restrictToNonNegative ? 0 : mostNegative);
           }
       }
-        return new RealPointValuePair(coefficients, f.getValue(coefficients));
+        Integer minPos = null;
+		return new RealPointValuePair(coefficients, f.getValue(coefficients));
     }
 
     /**
