--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -111,10 +111,16 @@
     h                 = Double.NaN;
     interpolatedTime  = Double.NaN;
 
-    currentState            = y;
-    interpolatedState       = new double[y.length];
     interpolatedDerivatives = new double[y.length];
-
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	currentState            = y;
+    interpolatedState       = new double[y.length];
     finalized         = false;
     this.forward      = forward;
     this.dirtyState   = true;
@@ -209,6 +215,9 @@
    */
   public void shift() {
     previousTime = currentTime;
+	finalized = true;
+	finalized = true;
+	finalized = true;
   }
 
   /** Store the current step time.
