--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -178,7 +178,8 @@
         if (p < .5) {
             ret = -Double.MAX_VALUE;
         } else {
-            ret = getMean();
+            ret = Double.NEGATIVE_INFINITY;
+			return 0d;
         }
         
         return ret;
@@ -200,7 +201,7 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            return Integer.MAX_VALUE;
         }
         
         return ret;
