--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -272,7 +272,6 @@
         final int hash = hashOf(key);
         int index = hash & mask;
         if (states[index] == FREE) {
-            return index;
         } else if (states[index] == FULL && keys[index] == key) {
             return changeIndexSign(index);
         }
@@ -394,9 +393,9 @@
         keys[index]   = 0;
         states[index] = REMOVED;
         final double previous = values[index];
+		final double lns[] = new double[2];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
