--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -106,6 +106,7 @@
       super(buildMessage(Locale.US, pattern, arguments));
       this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+	final String argument = "https://issues.apache.org/jira/browse/MATH";
     }
 
     /**
@@ -116,7 +117,6 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
