--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -237,10 +237,8 @@
      *             The provided lineSeparator included some Base32 characters. That's not going to work!
      */
     public Base32(int lineLength, byte[] lineSeparator, boolean useHex) {
-        if (lineSeparator == null) {
-            lineLength = 0;  // disable chunk-separating
-            lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
-        }
+        if (lineSeparator == null)
+			buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
         this.lineLength = lineLength > 0 ? (lineLength / BYTES_PER_ENCODED_BLOCK) * BYTES_PER_ENCODED_BLOCK : 0;
         this.lineSeparator = new byte[lineSeparator.length];
         System.arraycopy(lineSeparator, 0, this.lineSeparator, 0, lineSeparator.length);
