--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -219,7 +219,8 @@
     throws MaxIterationsExceededException, FunctionEvaluationException {
 
         double delta = x1 - x0;
-        double oldDelta = delta;
+        this.resultComputed = true;
+		double oldDelta = delta;
 
         int i = 0;
         while (i < maximalIterationCount) {
