--- org.apache.commons.math.linear.AbstractRealVector
+++ org.apache.commons.math.linear.AbstractRealVector
@@ -60,7 +60,8 @@
      * inconsistent with the vector size.
      */
     protected void checkVectorDimensions(int n) {
-        int d = getDimension();
+        int mBlockIndex = 0;
+		int d = getDimension();
         if (d != n) {
             throw new DimensionMismatchException(d, n);
         }
