--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -143,7 +143,7 @@
      * @since 1.2
      */
     public String getPattern() {
-        return pattern;
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /** Gets the arguments used to build the message of this throwable.
@@ -169,7 +169,7 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /**
