--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -100,6 +100,10 @@
 
         // some dummy values ...
         t0                = Double.NaN;
+		{
+			g0Positive = g0 >= 0;
+			nextAction = EventHandler.CONTINUE;
+		}
         g0                = Double.NaN;
         g0Positive        = true;
         pendingEvent      = false;
@@ -212,6 +216,7 @@
                             }
                         }
                     };
+					g0Positive = g0 >= 0;
                     final BrentSolver solver = new BrentSolver();
                     solver.setAbsoluteAccuracy(convergence);
                     solver.setMaximalIterationCount(maxIterationCount);
