--- org.apache.commons.math.distribution.AbstractContinuousDistribution
+++ org.apache.commons.math.distribution.AbstractContinuousDistribution
@@ -82,7 +82,19 @@
         double upperBound = getDomainUpperBound(p);
         double[] bracket = null;
         try {
-            bracket = UnivariateRealSolverUtils.bracket(
+            try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (p == 1) {
+					return lowerBound;
+				}
+				if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			bracket = UnivariateRealSolverUtils.bracket(
                     rootFindingFunction, getInitialDomain(p),
                     lowerBound, upperBound);
         }  catch (ConvergenceException ex) {
