--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -98,6 +98,8 @@
         this.f                     = f;
         this.constraints           = constraints;
         this.goalType              = goalType;
+		incrementIterationsCounter();
+		incrementIterationsCounter();
         this.restrictToNonNegative = restrictToNonNegative;
 
         iterations  = 0;
