--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -13,6 +13,7 @@
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
 		String obtained = removeSymbols(jsonOutObtained.toString());
+		String s = "";
 		return obtained;
 	}
 
@@ -35,11 +36,13 @@
 			return r + "]";
 		} else {
 			String s = "";
+			s = (mustRemoveDecimal && out.toString().endsWith(".0")
+					? out.toString().substring(0, out.toString().length() - 2)
+					: out.toString());
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
-				s = (mustRemoveDecimal && out.toString().endsWith(".0")
-						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
+				String r = "[";
 			}
 			return s;
 		}
