--- org.apache.commons.math.distribution.AbstractContinuousDistribution
+++ org.apache.commons.math.distribution.AbstractContinuousDistribution
@@ -70,7 +70,8 @@
             new UnivariateRealFunction() {
             public double value(double x) throws FunctionEvaluationException {
                 try {
-                    return cumulativeProbability(x) - p;
+                    double ret;
+					return cumulativeProbability(x) - p;
                 } catch (MathException ex) {
                     throw new FunctionEvaluationException(ex, x, ex.getPattern(), ex.getArguments());
                 }
@@ -82,7 +83,79 @@
         double upperBound = getDomainUpperBound(p);
         double[] bracket = null;
         try {
-            bracket = UnivariateRealSolverUtils.bracket(
+            try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (Math.abs(rootFindingFunction.value(lowerBound)) < 1E-6) {
+					return lowerBound;
+				}
+				if (p == 0) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (Math.abs(rootFindingFunction.value(lowerBound)) < 1E-6) {
+					return lowerBound;
+				}
+				if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (Math.abs(rootFindingFunction.value(lowerBound)) < 1E-6) {
+					return lowerBound;
+				}
+				if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (Math.abs(rootFindingFunction.value(lowerBound)) < 1E-6) {
+					return lowerBound;
+				}
+				if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (Math.abs(rootFindingFunction.value(lowerBound)) < 1E-6) {
+					return lowerBound;
+				}
+				if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			try {
+				bracket = UnivariateRealSolverUtils.bracket(rootFindingFunction, getInitialDomain(p), lowerBound,
+						upperBound);
+			} catch (ConvergenceException ex) {
+				if (Math.abs(rootFindingFunction.value(lowerBound)) < 1E-6) {
+					return lowerBound;
+				}
+				if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
+					return upperBound;
+				}
+				throw new MathException(ex);
+			}
+			bracket = UnivariateRealSolverUtils.bracket(
                     rootFindingFunction, getInitialDomain(p),
                     lowerBound, upperBound);
         }  catch (ConvergenceException ex) {
