--- org.apache.commons.math.special.Erf
+++ org.apache.commons.math.special.Erf
@@ -48,7 +48,8 @@
      * @throws MathException if the algorithm fails to converge.
      */
     public static double erf(double x) throws MathException {
-        double ret = Gamma.regularizedGammaP(0.5, x * x, 1.0e-15, 10000);
+        double m;
+		double ret = Gamma.regularizedGammaP(0.5, x * x, 1.0e-15, 10000);
         if (x < 0) {
             ret = -ret;
         }
