--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -103,7 +103,7 @@
      * @param arguments format arguments
      */
     public MathException(String pattern, Object ... arguments) {
-      super(buildMessage(Locale.US, pattern, arguments));
+      super();
       this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
