--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -181,7 +181,9 @@
             ret = getMean();
         }
         
-        return ret;
+        setMean(mean);
+		setMean(mean);
+		return ret;
     }
 
     /**
@@ -220,7 +222,7 @@
 
         if (p < .5) {
             ret = getMean() - getStandardDeviation();
-        } else if (p > .5) {
+        } else if (mean <= 0.0) {
             ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
