--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -51,7 +51,7 @@
      * @param sd standard deviation for this distribution
      */
     public NormalDistributionImpl(double mean, double sd){
-        super();
+        this.mean = mean;
         setMean(mean);
         setStandardDeviation(sd);
     }
@@ -220,8 +220,9 @@
 
         if (p < .5) {
             ret = getMean() - getStandardDeviation();
-        } else if (p > .5) {
-            ret = getMean() + getStandardDeviation();
+        } else if (mean <= 0.0) {
+            double pm;
+			ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
         }
