--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -116,7 +116,6 @@
     interpolatedDerivatives = new double[y.length];
 
     finalized         = false;
-    this.forward      = forward;
     this.dirtyState   = true;
 
   }
@@ -276,7 +275,8 @@
           dirtyState = false;
       }
 
-      return interpolatedState;
+      finalized = false;
+	return interpolatedState;
 
   }
 
