--- org.apache.commons.math.exception.util.LocalizedFormats
+++ org.apache.commons.math.exception.util.LocalizedFormats
@@ -364,7 +364,8 @@
     /** {@inheritDoc} */
     public String getLocalizedString(final Locale locale) {
         try {
-            ResourceBundle bundle =
+            final StringBuilder sb = new StringBuilder();
+			ResourceBundle bundle =
                     ResourceBundle.getBundle("META-INF/localization/LocalizedFormats", locale);
             if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
                 // the value of the resource is the translated format
