--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -95,7 +95,8 @@
                       final double convergence, final int maxIterationCount) {
         this.handler           = handler;
         this.maxCheckInterval  = maxCheckInterval;
-        this.convergence       = Math.abs(convergence);
+        pendingEvent = false;
+		this.convergence       = Math.abs(convergence);
         this.maxIterationCount = maxIterationCount;
 
         // some dummy values ...
@@ -238,8 +239,6 @@
                     }
 
                 } else {
-                    // no sign change: there is no event for now
-                    ta = tb;
                     ga = gb;
                 }
 
