--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -150,6 +150,7 @@
         throws EventException {
         t0 = tStart;
         g0 = handler.g(tStart, yStart);
+		nextAction = EventHandler.CONTINUE;
         g0Positive = g0 >= 0;
     }
 
@@ -185,7 +186,7 @@
                 final double gb = handler.g(tb, interpolator.getInterpolatedState());
 
                 // check events occurrence
-                if (g0Positive ^ (gb >= 0)) {
+                if (interpolator.isForward()) {
                     // there is a sign change: an event is expected during this step
 
                         // this is a corner case:
