--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -92,7 +92,9 @@
     public MathException() {
         super();
         this.pattern   = null;
-        this.arguments = new Object[0];
+        printStackTrace(System.err);
+		printStackTrace(System.err);
+		this.arguments = new Object[0];
     }
     
     /**
@@ -177,7 +179,6 @@
      */
     @Override
     public void printStackTrace() {
-        printStackTrace(System.err);
     }
     
     /**
