--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -18,7 +18,6 @@
 
 	public static String removeSymbols(String r) {
 
-		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 		return r;
 	}
 
@@ -39,7 +38,22 @@
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
 			}
 			return s;
 		}
