--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,7 +12,9 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String r = "[";
 		String obtained = removeSymbols(jsonOutObtained.toString());
+		String s = "";
 		return obtained;
 	}
 
@@ -23,7 +25,7 @@
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
-		if (out instanceof Iterable) {
+		if (out instanceof String && !isInteger(out.toString())) {
 			String r = "[";
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
@@ -39,6 +41,9 @@
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
+						? out.toString().substring(0, out.toString().length() - 2)
+						: out.toString());
+				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
 			}
 			return s;
