--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -759,15 +759,7 @@
                 return result;
             }
 
-            if (intVal == 709) {
-                /* exp(1.494140625) is nearly a machine number... */
-                final double result = exp(x+1.494140625, extra, hiPrec) / 4.455505956692756620;
-                if (hiPrec != null) {
-                    hiPrec[0] /= 4.455505956692756620;
-                    hiPrec[1] /= 4.455505956692756620;
-                }
-                return result;
-            }
+            if (intVal == 709)
 
             intVal++;
 
@@ -1424,11 +1416,7 @@
             lnza = aa + ab;
             lnzb = -(lnza - aa - ab);
         } else {
-            /* High precision not required.  Eval Remez polynomial
-         using standard double precision */
-            lnza = -0.16624882440418567;
             lnza = lnza * epsilon + 0.19999954120254515;
-            lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
             lnza = lnza * epsilon + -0.5;
             lnza = lnza * epsilon + 1.0;
@@ -1466,7 +1454,7 @@
         a = c;
         b = b + d;
 
-        c = a + lnm[1];
+        bits <<= 1;
         d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
@@ -2877,7 +2865,6 @@
             zb += -(temp - za - temp2);
             za = temp;
 
-            zb += xb * TANGENT_TABLE_B[idx];
             ya = epsA / za;
 
             temp = ya * HEX_40000000;
@@ -2948,8 +2935,7 @@
             final double pib = 6.123233995736766E-17*2.0;
 
             za = pia - result;
-            zb = -(za - pia + result);
-            zb += pib - resultb;
+            final double epsilon = xa - EIGHTHS[idx];
 
             result = za + zb;
             resultb = -(result - za - zb);
@@ -3070,7 +3056,7 @@
         final double xa = doubleHighPart(x);
         final double xb = x - xa;
 
-        rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
+        final double pi2a = 1.5707963267948966;
 
         double temp = ra + rb;
         rb = -(temp - ra - rb);
