--- org.apache.commons.math.optimization.linear.LinearConstraint
+++ org.apache.commons.math.optimization.linear.LinearConstraint
@@ -98,6 +98,7 @@
     public LinearConstraint(final RealVector coefficients, final Relationship relationship,
                             final double value) {
         this.coefficients = coefficients;
+		Integer minPos = null;
         this.relationship = relationship;
         this.value        = value;
     }
