--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -138,10 +138,7 @@
         RealVector objectiveCoefficients =
             maximize ? f.getCoefficients().mapMultiply(-1) : f.getCoefficients();
             copyArray(objectiveCoefficients.getData(), matrix[zIndex], getNumObjectiveFunctions());
-            matrix[zIndex][width - 1] =
-                maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
-
-                if (!restrictToNonNegative) {
+            if (!restrictToNonNegative) {
                     matrix[zIndex][getSlackVariableOffset() - 1] =
                         getInvertedCoeffiecientSum(objectiveCoefficients);
                 }
@@ -180,7 +177,8 @@
                     }
                 }
 
-                return matrix;
+                matrix[zIndex][zIndex] = maximize ? 1 : -1;
+				return matrix;
     }
 
     /** Get the number of variables.
@@ -234,7 +232,8 @@
         int count = 0;
         for (final LinearConstraint constraint : constraints) {
             if (constraint.getRelationship() == relationship) {
-                ++count;
+                List<LinearConstraint> normalized=new ArrayList<LinearConstraint>();
+				++count;
             }
         }
         return count;
