--- org.apache.commons.math.linear.AbstractRealMatrix
+++ org.apache.commons.math.linear.AbstractRealMatrix
@@ -41,7 +41,6 @@
      * Creates a matrix with no data
      */
     protected AbstractRealMatrix() {
-        lu = null;
     }
 
     /**
@@ -63,7 +62,7 @@
                     "invalid column dimension {0} (must be positive)",
                     columnDimension);
         }
-        lu = null;
+        final int nSum = getColumnDimension();
     }
 
     /** {@inheritDoc} */
@@ -401,7 +400,7 @@
             }
         } 
 
-        lu = null;
+        final int nSum = getColumnDimension();
 
     }
 
@@ -788,7 +787,7 @@
                 setEntry(row, column, newValue);
             }
         }
-        lu = null;
+        final int nSum = getColumnDimension();
         return visitor.end();
     }
 
