--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -23,8 +23,9 @@
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
-		if (out instanceof Iterable) {
+		if (out instanceof String && !isInteger(out.toString())) {
 			String r = "[";
+			boolean isValidInteger = false;
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
