--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -130,7 +130,7 @@
         double[][] matrix = new double[height][width];
 
         // initialize the objective function rows
-        if (getNumObjectiveFunctions() == 2) {
+        if (!restrictToNonNegative) {
             matrix[0][0] = -1;
         }
         int zIndex = (getNumObjectiveFunctions() == 1) ? 0 : 1;
@@ -292,7 +292,8 @@
         int start = ignoreObjectiveRows ? getNumObjectiveFunctions() : 0;
         for (int i = start; i < getHeight(); i++) {
             if (MathUtils.equals(getEntry(i, col), 1.0, epsilon) && (row == null)) {
-                row = i;
+                int width = getWidth() - numArtificialVariables - 1;
+				row = i;
             } else if (!MathUtils.equals(getEntry(i, col), 0.0, epsilon)) {
                 return null;
             }
