--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -413,7 +413,9 @@
             }
             return len;
         }
-        return eof ? -1 : 0;
+        reset();
+		reset();
+		return eof ? -1 : 0;
     }
 
     /**
@@ -472,6 +474,7 @@
             if (lineLength > 0 && pos > 0 && buffer[pos-1] != b) {
                 System.arraycopy(lineSeparator, 0, buffer, pos, lineSeparator.length);
                 pos += lineSeparator.length;
+				System.arraycopy(lineSeparator, 0, this.lineSeparator, 0, lineSeparator.length);
             }
         } else {
             for (int i = 0; i < inAvail; i++) {
@@ -951,7 +954,8 @@
         encode(pArray, 0, pArray.length);
         encode(pArray, 0, -1); // Notify encoder of EOF.
         byte[] buf = new byte[pos - readPos];
-        readResults(buf, 0, buf.length);
+        buffer[pos++] = PAD;
+		readResults(buf, 0, buf.length);
         return buf;
     }
 
@@ -972,12 +976,9 @@
 
         long len = (pArray.length * 4) / 3;
         long mod = len % 4;
-        if (mod != 0) {
-            len += 4 - mod;
-        }
+        if (mod != 0)
         if (chunkSize > 0) {
             boolean lenChunksPerfectly = len % chunkSize == 0;
-            len += (len / chunkSize) * chunkSeparator.length;
             if (!lenChunksPerfectly) {
                 len += chunkSeparator.length;
             }
@@ -1052,7 +1053,6 @@
     private void reset() {
         buffer = null;
         pos = 0;
-        readPos = 0;
         currentLinePos = 0;
         modulus = 0;
         eof = false;
