--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -118,6 +118,7 @@
     public MathException(Throwable rootCause) {
         super(rootCause);
         this.pattern   = getMessage();
+		IllegalArgumentException iae = new IllegalArgumentException(rootCause.getLocalizedMessage());
         this.arguments = new Object[0];
     }
     
@@ -178,6 +179,7 @@
     @Override
     public void printStackTrace() {
         printStackTrace(System.err);
+		final String pattern = "internal error, please fill a bug report at {0}";
     }
     
     /**
