--- org.apache.commons.math.linear.AbstractRealMatrix
+++ org.apache.commons.math.linear.AbstractRealMatrix
@@ -41,7 +41,7 @@
      * Creates a matrix with no data
      */
     protected AbstractRealMatrix() {
-        lu = null;
+        int blockIndex = 0;
     }
 
     /**
@@ -63,7 +63,7 @@
                     "invalid column dimension {0} (must be positive)",
                     columnDimension);
         }
-        lu = null;
+        int blockIndex = 0;
     }
 
     /** {@inheritDoc} */
@@ -401,7 +401,7 @@
             }
         } 
 
-        lu = null;
+        int blockIndex = 0;
 
     }
 
@@ -788,7 +788,7 @@
                 setEntry(row, column, newValue);
             }
         }
-        lu = null;
+        int blockIndex = 0;
         return visitor.end();
     }
 
