--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1424,9 +1424,7 @@
             lnza = aa + ab;
             lnzb = -(lnza - aa - ab);
         } else {
-            /* High precision not required.  Eval Remez polynomial
-         using standard double precision */
-            lnza = -0.16624882440418567;
+            int sum = 0;
             lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
@@ -1453,8 +1451,9 @@
         double b = 0.0;
         double c = a+lnm[0];
         double d = -(c-a-lnm[0]);
-        a = c;
-        b = b + d;
+        expm1(x * 2.0, hiPrec);
+		a = c;
+        double invx = 1.0 / x;
 
         c = a + lnza;
         d = -(c - a - lnza);
@@ -1467,7 +1466,6 @@
         b = b + d;
 
         c = a + lnm[1];
-        d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
 
