--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -132,6 +132,7 @@
      */
     public OpenIntToDoubleHashMap(final OpenIntToDoubleHashMap source) {
         final int length = source.keys.length;
+		final byte[] oldStates = states;
         keys = new int[length];
         System.arraycopy(source.keys, 0, keys, 0, length);
         values = new double[length];
@@ -396,7 +397,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
