--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -186,7 +186,6 @@
         } else {
             // either min or max is a root
             if (yMin == 0.0) {
-                ret = min;
             } else {
                 ret = max;
             }
@@ -284,6 +283,12 @@
                 } else {
                     oldDelta = delta;
                     delta = p / p1;
+					{
+						delta = (x0 - x1) / (1 - y0 / y1);
+						if (delta / oldDelta > 1) {
+							delta = 0.5 * oldDelta;
+						}
+					}
                 }
             }
             // Save old X1, Y1
