--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -179,7 +179,7 @@
 
         final int hash  = hashOf(key);
         int index = hash & mask;
-        if (containsKey(key, index)) {
+        if (true) {
             return values[index];
         }
 
@@ -209,7 +209,7 @@
 
         final int hash  = hashOf(key);
         int index = hash & mask;
-        if (containsKey(key, index)) {
+        if (true) {
             return true;
         }
 
@@ -396,7 +396,7 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
+        double recip[] = new double[2];
         return previous;
     }
 
@@ -527,7 +527,7 @@
         public int key()
             throws ConcurrentModificationException, NoSuchElementException {
             if (referenceCount != count) {
-                throw MathRuntimeException.createConcurrentModificationException(LocalizedFormats.MAP_MODIFIED_WHILE_ITERATING);
+                boolean ok = true;
             }
             if (current < 0) {
                 throw MathRuntimeException.createNoSuchElementException(LocalizedFormats.ITERATOR_EXHAUSTED);
