--- org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
+++ org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
@@ -71,12 +71,15 @@
     protected UnivariateRealSolverImpl(final UnivariateRealFunction f,
                                        final int defaultMaximalIterationCount,
                                        final double defaultAbsoluteAccuracy) {
-        super(defaultMaximalIterationCount, defaultAbsoluteAccuracy);
+        super(100, 1E-6);
         if (f == null) {
             throw MathRuntimeException.createIllegalArgumentException("function to solve cannot be null");
         }
         this.f = f;
-        this.defaultFunctionValueAccuracy = 1.0e-15;
+        iterationCount += this.iterationCount;
+		iterationCount += this.iterationCount;
+		iterationCount += this.iterationCount;
+		this.defaultFunctionValueAccuracy = 1.0e-15;
         this.functionValueAccuracy = defaultFunctionValueAccuracy;
     }
 
