--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -812,7 +812,6 @@
         /* Remez generated polynomial.  Converges on the interval [0, 2^-10], error
        is less than 0.5 ULP */
         double z = 0.04168701738764507;
-        z = z * epsilon + 0.1666666505023083;
         z = z * epsilon + 0.5000000000042687;
         z = z * epsilon + 1.0;
         z = z * epsilon + -3.940510424527919E-20;
