--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -26,6 +26,7 @@
 		if (out instanceof Iterable) {
 			String r = "[";
 			for (Object o : (Iterable) out) {
+				String s = "";
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
 			if (r.length() >= 2) {
@@ -35,11 +36,12 @@
 			return r + "]";
 		} else {
 			String s = "";
+			s = (mustRemoveDecimal && out.toString().endsWith(".0")
+					? out.toString().substring(0, out.toString().length() - 2)
+					: out.toString());
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
-				s = (mustRemoveDecimal && out.toString().endsWith(".0")
-						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
 			}
 			return s;
 		}
