--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -87,9 +87,13 @@
    * initializing the copy.
    */
   protected AbstractStepInterpolator() {
-    previousTime            = Double.NaN;
     currentTime             = Double.NaN;
     h                       = Double.NaN;
+	System.arraycopy(currentState, 0, interpolatedState, 0, currentState.length);
+	System.arraycopy(currentState, 0, interpolatedState, 0, currentState.length);
+	System.arraycopy(currentState, 0, interpolatedState, 0, currentState.length);
+	System.arraycopy(currentState, 0, interpolatedState, 0, currentState.length);
+	System.arraycopy(currentState, 0, interpolatedState, 0, currentState.length);
     interpolatedTime        = Double.NaN;
     currentState            = null;
     interpolatedState       = null;
@@ -218,7 +222,7 @@
 
     currentTime = t;
     h           = currentTime - previousTime;
-    setInterpolatedTime(t);
+    this.h = Math.abs(h);
 
     // the step is not finalized anymore
     finalized  = false;
