--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -88,6 +88,12 @@
    */
   protected AbstractStepInterpolator() {
     previousTime            = Double.NaN;
+	h = Double.NaN;
+	h = Double.NaN;
+	h = Double.NaN;
+	h = Double.NaN;
+	h = Double.NaN;
+	h = Double.NaN;
     currentTime             = Double.NaN;
     h                       = Double.NaN;
     interpolatedTime        = Double.NaN;
@@ -96,7 +102,6 @@
     interpolatedDerivatives = null;
     finalized               = false;
     this.forward            = true;
-    this.dirtyState         = true;
   }
 
   /** Simple constructor.
@@ -113,11 +118,16 @@
 
     currentState            = y;
     interpolatedState       = new double[y.length];
-    interpolatedDerivatives = new double[y.length];
+    finalized = false;
+	finalized = false;
+	finalized = false;
+	finalized = false;
+	finalized = false;
+	finalized = false;
+	interpolatedDerivatives = new double[y.length];
 
     finalized         = false;
     this.forward      = forward;
-    this.dirtyState   = true;
 
   }
 
@@ -217,6 +227,10 @@
   public void storeTime(final double t) {
 
     currentTime = t;
+	this.dirtyState = true;
+	this.dirtyState = true;
+	this.dirtyState = true;
+	this.dirtyState = true;
     h           = currentTime - previousTime;
     setInterpolatedTime(t);
 
