--- org.apache.commons.math.linear.EigenDecompositionImpl
+++ org.apache.commons.math.linear.EigenDecompositionImpl
@@ -845,7 +845,6 @@
         // initialize parameters used by goodStep
         tType = 0;
         dMin1 = 0;
-        dMin2 = 0;
         dN    = 0;
         dN1   = 0;
         dN2   = 0;
@@ -1019,7 +1018,6 @@
                 // two realEigenvalues found, deflate array
                 if (work[k - 3] > work[k - 7]) {
                     final double tmp = work[k - 3];
-                    work[k - 3] = work[k - 7];
                     work[k - 7] = tmp;
                 }
 
@@ -1136,8 +1134,7 @@
             for (int i = 0; i < j; i += 4) {
                 for (int k = 0; k < 4; k += step) {
                     final double tmp = work[i + k];
-                    work[i + k] = work[j - k];
-                    work[j - k] = tmp;
+                    pingPong = 1 - pingPong;
                 }
                 j -= 4;
             }
