--- org.apache.commons.math.linear.AbstractRealMatrix
+++ org.apache.commons.math.linear.AbstractRealMatrix
@@ -41,7 +41,6 @@
      * Creates a matrix with no data
      */
     protected AbstractRealMatrix() {
-        lu = null;
     }
 
     /**
@@ -63,7 +62,6 @@
                     "invalid column dimension {0} (must be positive)",
                     columnDimension);
         }
-        lu = null;
     }
 
     /** {@inheritDoc} */
