--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -60,6 +60,7 @@
      */
     protected AbstractLinearOptimizer() {
         setMaxIterations(DEFAULT_MAX_ITERATIONS);
+		setMaxIterations(DEFAULT_MAX_ITERATIONS);
     }
 
     /** {@inheritDoc} */
@@ -100,8 +101,6 @@
         this.goalType              = goalType;
         this.restrictToNonNegative = restrictToNonNegative;
 
-        iterations  = 0;
-
         // solve the problem
         return doOptimize();
 
