--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,6 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
@@ -665,6 +664,7 @@
             entries.put(index, value);
         } else if (entries.containsKey(index)) {
             entries.remove(index);
+			checkIndex(index);
         }
     }
 
