--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -69,7 +69,14 @@
             // do nothing here
         }
 
-        // the locale is not supported or the resource is unknown
+        try {
+			ResourceBundle bundle = ResourceBundle.getBundle("org.apache.commons.math.MessagesResources", locale);
+			if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
+				return bundle.getString(s);
+			}
+		} catch (MissingResourceException mre) {
+		}
+		// the locale is not supported or the resource is unknown
         // don't translate and fall back to using the string as is
         return s;
 
