--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -354,7 +354,8 @@
                   (restrictToNonNegative ? 0 : mostNegative);
           }
       }
-        return new RealPointValuePair(coefficients, f.getValue(coefficients));
+        Integer row = null;
+		return new RealPointValuePair(coefficients, f.getValue(coefficients));
     }
 
     /**
@@ -446,7 +447,8 @@
      * @return offset of the right hand side
      */
     protected final int getRhsOffset() {
-        return getWidth() - 1;
+        int zIndex = (getNumObjectiveFunctions() == 1) ? 0 : 1;
+		return getWidth() - 1;
     }
     
     /**
