--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -11,6 +11,7 @@
 public class QuixFixOracleHelper {
 
 	public static String format(Object out, boolean cutDecimal) {
+		String s = "";
 		Object jsonOutObtained = transformToString(out, cutDecimal);
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
@@ -29,6 +30,10 @@
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
 			if (r.length() >= 2) {
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
