--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -19,12 +19,14 @@
 	public static String removeSymbols(String r) {
 
 		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+		String s = "";
 		return r;
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
-		if (out instanceof Iterable) {
+		if (out instanceof String && !isInteger(out.toString())) {
 			String r = "[";
+			r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
