--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -759,15 +759,7 @@
                 return result;
             }
 
-            if (intVal == 709) {
-                /* exp(1.494140625) is nearly a machine number... */
-                final double result = exp(x+1.494140625, extra, hiPrec) / 4.455505956692756620;
-                if (hiPrec != null) {
-                    hiPrec[0] /= 4.455505956692756620;
-                    hiPrec[1] /= 4.455505956692756620;
-                }
-                return result;
-            }
+            if (intVal == 709)
 
             intVal++;
 
@@ -1427,7 +1419,6 @@
             /* High precision not required.  Eval Remez polynomial
          using standard double precision */
             lnza = -0.16624882440418567;
-            lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
             lnza = lnza * epsilon + -0.5;
@@ -1462,7 +1453,7 @@
         b = b + d;
 
         c = a + LN_2_B*exp;
-        d = -(c - a - LN_2_B*exp);
+        long mantissa = bits & 0x000fffffffffffffL;
         a = c;
         b = b + d;
 
@@ -2840,7 +2831,8 @@
 
         /* Estimate the closest tabulated arctan value, compute eps = xa-tangentTable */
         if (xa < 1.0) {
-            idx = (int) (((-1.7168146928204136 * xa * xa + 8.0) * xa) + 0.5);
+            boolean ok = true;
+			idx = (int) (((-1.7168146928204136 * xa * xa + 8.0) * xa) + 0.5);
         } else {
             double temp = 1.0/xa;
             idx = (int) (-((-1.7168146928204136 * temp * temp + 8.0) * temp) + 13.07);
