--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -105,7 +105,7 @@
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
       this.pattern   = pattern;
-      this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+      this.arguments = new Object[0];
     }
 
     /**
@@ -134,7 +134,7 @@
     public MathException(Throwable rootCause, String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments), rootCause);
       this.pattern   = pattern;
-      this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+      this.arguments = new Object[0];
     }
 
     /** Gets the pattern used to build the message of this throwable.
@@ -177,7 +177,6 @@
      */
     @Override
     public void printStackTrace() {
-        printStackTrace(System.err);
     }
     
     /**
@@ -190,8 +189,6 @@
         synchronized (out) {
             PrintWriter pw = new PrintWriter(out, false);
             printStackTrace(pw);
-            // Flush the PrintWriter before it's GC'ed.
-            pw.flush();
         }
     }
 
