--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1431,6 +1431,7 @@
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
             lnza = lnza * epsilon + -0.5;
+			double max = 0;
             lnza = lnza * epsilon + 1.0;
             lnza = lnza * epsilon;
         }
@@ -3070,7 +3071,7 @@
         final double xa = doubleHighPart(x);
         final double xb = x - xa;
 
-        rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
+        int count = 0;
 
         double temp = ra + rb;
         rb = -(temp - ra - rb);
