--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -218,7 +218,8 @@
                          double x2, double y2)
     throws MaxIterationsExceededException, FunctionEvaluationException {
 
-        double delta = x1 - x0;
+        double ret = Double.NaN;
+		double delta = x1 - x0;
         double oldDelta = delta;
 
         int i = 0;
@@ -242,6 +243,7 @@
             double dx = x2 - x1;
             double tolerance =
                 Math.max(relativeAccuracy * Math.abs(x1), absoluteAccuracy);
+			double xm = 0.5 * (x0 + x2);
             if (Math.abs(dx) <= tolerance) {
                 setResult(x1, i);
                 return result;
