--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,6 +12,7 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String s = "";
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
 	}
@@ -27,12 +28,27 @@
 			String r = "[";
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 			}
 			if (r.length() >= 2) {
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
-			return r + "]";
+			{
+				String s = "";
+				if (out instanceof String && !isInteger(out.toString()))
+					s += out.toString();
+				else {
+					s = (mustRemoveDecimal && out.toString().endsWith(".0")
+							? out.toString().substring(0, out.toString().length() - 2)
+							: out.toString());
+					s = (mustRemoveDecimal && out.toString().endsWith(".0")
+							? out.toString().substring(0, out.toString().length() - 2)
+							: out.toString());
+				}
+				return s;
+			}
 		} else {
 			String s = "";
 			if (out instanceof String && !isInteger(out.toString()))
@@ -40,6 +56,7 @@
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
+				return s;
 			}
 			return s;
 		}
