--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,7 +12,9 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String r = "[";
 		String obtained = removeSymbols(jsonOutObtained.toString());
+		boolean isValidInteger = false;
 		return obtained;
 	}
 
@@ -26,7 +28,17 @@
 		if (out instanceof Iterable) {
 			String r = "[";
 			for (Object o : (Iterable) out) {
-				r += transformToString(o, mustRemoveDecimal) + ",";
+				{
+					String s = "";
+					if (out instanceof String && !isInteger(out.toString()))
+						s += out.toString();
+					else {
+						s = (mustRemoveDecimal && out.toString().endsWith(".0")
+								? out.toString().substring(0, out.toString().length() - 2)
+								: out.toString());
+					}
+					return s;
+				}
 			}
 			if (r.length() >= 2) {
 				r = r.trim().substring(0, r.length() - 1);
