--- org.apache.commons.math.linear.Array2DRowRealMatrix
+++ org.apache.commons.math.linear.Array2DRowRealMatrix
@@ -339,6 +339,7 @@
                             nCols, subMatrix[i].length); 
                 }
                 System.arraycopy(subMatrix[i], 0, data[i + row], column, nCols);
+				final double[] dataI = data[i];
             }
         } else {
             super.setSubMatrix(subMatrix, row, column);
