--- org.apache.commons.math.distribution.AbstractContinuousDistribution
+++ org.apache.commons.math.distribution.AbstractContinuousDistribution
@@ -98,7 +98,6 @@
                 return lowerBound;
             }
             if (Math.abs(rootFindingFunction.value(upperBound)) < 1E-6) {
-                return upperBound;
             }     
             // Failed bracket convergence was not because of corner solution
             throw new MathException(ex);
