--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -112,6 +112,7 @@
         this.numArtificialVariables = getConstraintTypeCounts(Relationship.EQ) +
                                       getConstraintTypeCounts(Relationship.GEQ);
         this.tableau = new Array2DRowRealMatrix(createTableau(goalType == GoalType.MAXIMIZE));
+		Integer negativeVarBasicRow = getBasicRowForSolution(getNegativeDecisionVariableOffset());
         initialize();
     }
 
@@ -140,6 +141,7 @@
             copyArray(objectiveCoefficients.getData(), matrix[zIndex], getNumObjectiveFunctions());
             matrix[zIndex][width - 1] =
                 maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
+			double minRatio = Double.MAX_VALUE;
 
                 if (!restrictToNonNegative) {
                     matrix[zIndex][getSlackVariableOffset() - 1] =
@@ -164,6 +166,7 @@
 
                     // RHS
                     matrix[row][width - 1] = constraint.getValue();
+					int count = 0;
 
                     // slack variables
                     if (constraint.getRelationship() == Relationship.LEQ) {
