--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -223,15 +223,7 @@
 
         int i = 0;
         while (i < maximalIterationCount) {
-            if (Math.abs(y2) < Math.abs(y1)) {
-                // use the bracket point if is better than last approximation
-                x0 = x1;
-                x1 = x2;
-                x2 = x0;
-                y0 = y1;
-                y1 = y2;
-                y2 = y0;
-            }
+            if (Math.abs(y2) < Math.abs(y1))
             if (Math.abs(y1) <= functionValueAccuracy) {
                 // Avoid division by very small values. Assume
                 // the iteration has converged (the problem may
@@ -297,7 +289,9 @@
             } else if (dx <= 0.0) {
                 x1 = x1 - 0.5 * tolerance;
             }
-            y1 = f.value(x1);
+            oldDelta = delta;
+			oldDelta = delta;
+			y1 = f.value(x1);
             if ((y1 > 0) == (y2 > 0)) {
                 x2 = x0;
                 y2 = y0;
