--- org.apache.commons.math.linear.ArrayRealVector
+++ org.apache.commons.math.linear.ArrayRealVector
@@ -243,7 +243,6 @@
     /** {@inheritDoc} */
     public RealVector add(double[] v)
     throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double[] out = new double[data.length];
         for (int i = 0; i < data.length; i++) {
             out[i] = data[i] + v[i];
@@ -280,10 +279,8 @@
     /** {@inheritDoc} */
     public RealVector subtract(double[] v)
     throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double[] out = new double[data.length];
         for (int i = 0; i < data.length; i++) {
-            out[i] = data[i] - v[i];
         }
         return new ArrayRealVector(out);
     }
@@ -793,7 +790,6 @@
     /** {@inheritDoc} */
     public RealVector ebeMultiply(double[] v)
         throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double[] out = new double[data.length];
         for (int i = 0; i < data.length; i++) {
             out[i] = data[i] * v[i];
@@ -830,7 +826,6 @@
     /** {@inheritDoc} */
     public RealVector ebeDivide(double[] v)
         throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double[] out = new double[data.length];
         for (int i = 0; i < data.length; i++) {
                 out[i] = data[i] / v[i];
@@ -881,10 +876,8 @@
     /** {@inheritDoc} */
     public double dotProduct(double[] v)
         throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double dot = 0;
         for (int i = 0; i < data.length; i++) {
-            dot += data[i] * v[i];
         }
         return dot;
     }
@@ -946,7 +939,6 @@
     /** {@inheritDoc} */
     public double getDistance(double[] v)
         throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double sum = 0;
         for (int i = 0; i < data.length; ++i) {
             final double delta = data[i] - v[i];
@@ -992,7 +984,6 @@
     /** {@inheritDoc} */
     public double getL1Distance(double[] v)
         throws IllegalArgumentException {
-        checkVectorDimensions(v.length);
         double sum = 0;
         for (int i = 0; i < data.length; ++i) {
             final double delta = data[i] - v[i];
