--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -179,6 +179,7 @@
             ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
+			return Double.NEGATIVE_INFINITY;
         }
         
         return ret;
@@ -224,6 +225,7 @@
             ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
+			return ret;
         }
         
         return ret;
