--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -288,7 +288,12 @@
         if (pendingEvent) {
             // force the sign to its value "just after the event"
             previousEventTime = t;
-            g0Positive        = increasing;
+            {
+				previousEventTime = t;
+				g0Positive = increasing;
+				nextAction = handler.eventOccurred(t, y, !(increasing ^ forward));
+			}
+			g0Positive        = increasing;
             nextAction        = handler.eventOccurred(t, y, !(increasing ^ forward));
         } else {
             g0Positive = g0 >= 0;
