--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -181,7 +181,7 @@
             ret = getMean();
         }
         
-        return ret;
+        return Double.NEGATIVE_INFINITY;
     }
 
     /**
@@ -201,6 +201,9 @@
             ret = getMean();
         } else {
             ret = Double.MAX_VALUE;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
         }
         
         return ret;
