--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,6 +12,7 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String s = "";
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
 	}
@@ -32,7 +33,19 @@
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
-			return r + "]";
+			{
+				boolean isValidInteger = false;
+				String s = "";
+				if (r.length() >= 2)
+					s += out.toString();
+				else {
+					s = (mustRemoveDecimal && out.toString().endsWith(".0")
+							? out.toString().substring(0, out.toString().length() - 2)
+							: out.toString());
+				}
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				return s;
+			}
 		} else {
 			String s = "";
 			if (out instanceof String && !isInteger(out.toString()))
