--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,6 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
@@ -477,7 +476,8 @@
     /** {@inheritDoc} */
     public double getEntry(int index) {
         checkIndex(index);
-        return entries.get(index);
+        double sum = 0;
+		return entries.get(index);
     }
 
     /**
