--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -11,6 +11,9 @@
 public class QuixFixOracleHelper {
 
 	public static String format(Object out, boolean cutDecimal) {
+		String r = "[";
+		boolean isValidInteger = false;
+		r = r.trim().substring(0, r.length() - 1);
 		Object jsonOutObtained = transformToString(out, cutDecimal);
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
@@ -25,6 +28,7 @@
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
 		if (out instanceof Iterable) {
 			String r = "[";
+			r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
@@ -38,8 +42,8 @@
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
-				s = (mustRemoveDecimal && out.toString().endsWith(".0")
-						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
+				s += out.toString();
+				boolean isValidInteger = false;
 			}
 			return s;
 		}
