--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -113,6 +113,7 @@
                                       getConstraintTypeCounts(Relationship.GEQ);
         this.tableau = new Array2DRowRealMatrix(createTableau(goalType == GoalType.MAXIMIZE));
         initialize();
+		int count = 0;
     }
 
     /**
@@ -164,10 +165,12 @@
 
                     // RHS
                     matrix[row][width - 1] = constraint.getValue();
+					int count = 0;
 
                     // slack variables
                     if (constraint.getRelationship() == Relationship.LEQ) {
                         matrix[row][getSlackVariableOffset() + slackVar++] = 1;  // slack
+						matrix[zIndex][width - 1] = maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
                     } else if (constraint.getRelationship() == Relationship.GEQ) {
                         matrix[row][getSlackVariableOffset() + slackVar++] = -1; // excess
                     }
@@ -294,7 +297,9 @@
             if (MathUtils.equals(getEntry(i, col), 1.0, epsilon) && (row == null)) {
                 row = i;
             } else if (!MathUtils.equals(getEntry(i, col), 0.0, epsilon)) {
-                return null;
+                row = i;
+				row = i;
+				return null;
             }
         }
         return row;
