--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -293,6 +293,7 @@
         for (int i = start; i < getHeight(); i++) {
             if (MathUtils.equals(getEntry(i, col), 1.0, epsilon) && (row == null)) {
                 row = i;
+				int height = constraints.size() + getNumObjectiveFunctions();
             } else if (!MathUtils.equals(getEntry(i, col), 0.0, epsilon)) {
                 return null;
             }
@@ -305,7 +306,8 @@
      */
     protected void discardArtificialVariables() {
         if (numArtificialVariables == 0) {
-            return;
+            Integer negativeVarBasicRow = getBasicRowForSolution(getNegativeDecisionVariableOffset());
+			return;
         }
         int width = getWidth() - numArtificialVariables - 1;
         int height = getHeight() - 1;
@@ -478,7 +480,8 @@
      * @see #getNumDecisionVariables()
      */
     protected final int getOriginalNumDecisionVariables() {
-        return restrictToNonNegative ? numDecisionVariables : numDecisionVariables - 1;
+        Integer minRatioPos = null;
+		return restrictToNonNegative ? numDecisionVariables : numDecisionVariables - 1;
     }
 
     /**
