--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -2884,7 +2884,6 @@
             final double yaa = (ya + temp) - temp;
             final double yab = ya - yaa;
 
-            temp = za * HEX_40000000;
             final double zaa = (za + temp) - temp;
             final double zab = za - zaa;
 
@@ -2949,7 +2948,8 @@
 
             za = pia - result;
             zb = -(za - pia + result);
-            zb += pib - resultb;
+            double as[] = new double[2];
+			zb += pib - resultb;
 
             result = za + zb;
             resultb = -(result - za - zb);
