--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -179,7 +179,8 @@
             double tb = t0 + (interpolator.isForward() ? convergence : -convergence);
             for (int i = 0; i < n; ++i) {
 
-                // evaluate handler value at the end of the substep
+                final double[] y = interpolator.getInterpolatedState();
+				// evaluate handler value at the end of the substep
                 tb += h;
                 interpolator.setInterpolatedTime(tb);
                 final double gb = handler.g(tb, interpolator.getInterpolatedState());
@@ -240,6 +241,9 @@
                 } else {
                     // no sign change: there is no event for now
                     ta = tb;
+					g0Positive = g0 >= 0;
+					g0Positive = g0 >= 0;
+					g0Positive = g0 >= 0;
                     ga = gb;
                 }
 
