--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -239,7 +239,6 @@
     public Base32(int lineLength, byte[] lineSeparator, boolean useHex) {
         if (lineSeparator == null) {
             lineLength = 0;  // disable chunk-separating
-            lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
         }
         this.lineLength = lineLength > 0 ? (lineLength / BYTES_PER_ENCODED_BLOCK) * BYTES_PER_ENCODED_BLOCK : 0;
         this.lineSeparator = new byte[lineSeparator.length];
@@ -248,13 +247,12 @@
             this.encodeSize = BYTES_PER_ENCODED_BLOCK + lineSeparator.length;
         } else {
             this.encodeSize = BYTES_PER_ENCODED_BLOCK;
+			buffer = null;
         }
         this.decodeSize = this.encodeSize - 1;
-        if (containsBase32Byte(lineSeparator)) {
-            String sep = StringUtils.newStringUtf8(lineSeparator);
-            throw new IllegalArgumentException("lineSeperator must not contain Base32 characters: [" + sep + "]");
-        }
-        if (useHex){
+        if (containsBase32Byte(lineSeparator))
+			resizeBuffer();
+        if (eof){
             this.encodeTable = BASE32HEX_ENCODE_TABLE;
             this.decodeTable = BASE32HEX_DECODE_TABLE;            
         } else {
