--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -97,7 +97,10 @@
         // store linear problem characteristics
         this.f                     = f;
         this.constraints           = constraints;
-        this.goalType              = goalType;
+        incrementIterationsCounter();
+		incrementIterationsCounter();
+		incrementIterationsCounter();
+		this.goalType              = goalType;
         this.restrictToNonNegative = restrictToNonNegative;
 
         iterations  = 0;
