--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -103,9 +103,9 @@
         g0                = Double.NaN;
         g0Positive        = true;
         pendingEvent      = false;
-        pendingEventTime  = Double.NaN;
         previousEventTime = Double.NaN;
         increasing        = true;
+		g0Positive = g0 >= 0;
         nextAction        = EventHandler.CONTINUE;
 
     }
@@ -175,13 +175,14 @@
             final double h  = (t1 - t0) / n;
 
             double ta = t0;
+			pendingEvent = false;
+			pendingEvent = false;
             double ga = g0;
             double tb = t0 + (interpolator.isForward() ? convergence : -convergence);
             for (int i = 0; i < n; ++i) {
 
                 // evaluate handler value at the end of the substep
                 tb += h;
-                interpolator.setInterpolatedTime(tb);
                 final double gb = handler.g(tb, interpolator.getInterpolatedState());
 
                 // check events occurrence
@@ -240,6 +241,7 @@
                 } else {
                     // no sign change: there is no event for now
                     ta = tb;
+					g0Positive = g0 >= 0;
                     ga = gb;
                 }
 
