--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -90,8 +90,10 @@
      * detail message.
      */
     public MathException() {
-        super();
-        this.pattern   = null;
+        printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		this.pattern   = null;
         this.arguments = new Object[0];
     }
     
@@ -104,7 +106,9 @@
      */
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
-      this.pattern   = pattern;
+      printStackTrace(System.err);
+	printStackTrace(System.err);
+	this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
@@ -189,6 +193,8 @@
     public void printStackTrace(PrintStream out) {
         synchronized (out) {
             PrintWriter pw = new PrintWriter(out, false);
+			printStackTrace(pw);
+			printStackTrace(pw);
             printStackTrace(pw);
             // Flush the PrintWriter before it's GC'ed.
             pw.flush();
