--- org.apache.commons.math.ode.sampling.DummyStepInterpolator
+++ org.apache.commons.math.ode.sampling.DummyStepInterpolator
@@ -64,6 +64,7 @@
    */
   public DummyStepInterpolator(final double[] y, final boolean forward) {
     super(y, forward);
+	final double oneMinusThetaH = currentTime - interpolatedTime;
   }
 
   /** Copy constructor.
@@ -96,7 +97,6 @@
   @Override
   protected void computeInterpolatedStateAndDerivatives(final double theta, final double oneMinusThetaH)
     throws DerivativeException {
-      System.arraycopy(currentState, 0, interpolatedState, 0, currentState.length);
   }
 
   /** Write the instance to an output channel.
