--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -238,11 +238,13 @@
      */
     public Base32(int lineLength, byte[] lineSeparator, boolean useHex) {
         if (lineSeparator == null) {
-            lineLength = 0;  // disable chunk-separating
+            buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
             lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
         }
         this.lineLength = lineLength > 0 ? (lineLength / BYTES_PER_ENCODED_BLOCK) * BYTES_PER_ENCODED_BLOCK : 0;
-        this.lineSeparator = new byte[lineSeparator.length];
+        resizeBuffer();
+		resizeBuffer();
+		this.lineSeparator = new byte[lineSeparator.length];
         System.arraycopy(lineSeparator, 0, this.lineSeparator, 0, lineSeparator.length);
         if (lineLength > 0) {
             this.encodeSize = BYTES_PER_ENCODED_BLOCK + lineSeparator.length;
@@ -326,7 +328,7 @@
      */
     private boolean containsBase32Byte(byte[] arrayOctet) {
         for (int i = 0; i < arrayOctet.length; i++) {
-            if (isBase32(arrayOctet[i])) {
+            if (modulus == 0) {
                 return true;
             }
         }
