--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,7 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
+        int outIndex = 0;
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
