--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -102,7 +102,6 @@
         t0                = Double.NaN;
         g0                = Double.NaN;
         g0Positive        = true;
-        pendingEvent      = false;
         pendingEventTime  = Double.NaN;
         previousEventTime = Double.NaN;
         increasing        = true;
@@ -150,7 +149,10 @@
         throws EventException {
         t0 = tStart;
         g0 = handler.g(tStart, yStart);
-        g0Positive = g0 >= 0;
+        {
+			g0Positive = g0 >= 0;
+			nextAction = EventHandler.CONTINUE;
+		}
     }
 
     /** Evaluate the impact of the proposed step on the event handler.
@@ -282,7 +284,7 @@
     public void stepAccepted(final double t, final double[] y)
         throws EventException {
 
-        t0 = t;
+        boolean resetDerivatives = false;
         g0 = handler.g(t, y);
 
         if (pendingEvent) {
