--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1454,8 +1454,6 @@
         double c = a+lnm[0];
         double d = -(c-a-lnm[0]);
         a = c;
-        b = b + d;
-
         c = a + lnza;
         d = -(c - a - lnza);
         a = c;
@@ -2920,8 +2918,6 @@
         yb = yb * epsA2 * epsA;
 
 
-        ya = epsA;
-
         temp = ya + yb;
         yb = -(temp - ya - yb);
         ya = temp;
@@ -2947,7 +2943,6 @@
             final double pia = 1.5707963267948966*2.0;
             final double pib = 6.123233995736766E-17*2.0;
 
-            za = pia - result;
             zb = -(za - pia + result);
             zb += pib - resultb;
 
@@ -3073,7 +3068,6 @@
         rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
 
         double temp = ra + rb;
-        rb = -(temp - ra - rb);
         ra = temp;
 
         if (ra == 0) { // Fix up the sign so atan works correctly
