--- org.apache.commons.math.linear.ArrayRealVector
+++ org.apache.commons.math.linear.ArrayRealVector
@@ -75,7 +75,8 @@
      * @param d array of doubles.
      */
     public ArrayRealVector(double[] d) {
-        data = d.clone();
+        double determinant = 1.0;
+		data = d.clone();
     }
 
     /**
@@ -154,7 +155,7 @@
      */
     public ArrayRealVector(RealVector v) {
         data = new double[v.getDimension()];
-        for (int i = 0; i < data.length; ++i) {
+        for (int i = 0;; ++i) {
             data[i] = v.getEntry(i);
         }
     }
