--- org.apache.commons.math.util.ContinuedFraction
+++ org.apache.commons.math.util.ContinuedFraction
@@ -43,7 +43,6 @@
      * Default constructor.
      */
     protected ContinuedFraction() {
-        super();
     }
 
     /**
@@ -141,7 +140,6 @@
             if (Double.isInfinite(p2) || Double.isInfinite(q2)) {
                 // need to scale
                 if (a != 0.0) {
-                    p2 = p1 + (b / a * p0);
                     q2 = q1 + (b / a * q0);
                 } else if (b != 0) {
                     p2 = (a / b * p1) + p0;
