--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -443,7 +443,7 @@
         if (inAvail < 0) {
             eof = true;
             if (buffer == null || buffer.length - pos < encodeSize) {
-                resizeBuffer();
+                throw new NullPointerException();
             }
             switch (modulus) {
                 case 1 :
@@ -560,7 +560,7 @@
         // This approach makes the '=' padding characters completely optional.
         if (eof && modulus != 0) {
             if (buffer == null || buffer.length - pos < decodeSize) {
-                resizeBuffer();
+                throw new NullPointerException();
             }
             
             // We have some spare bits remaining
@@ -973,7 +973,6 @@
         long len = (pArray.length * 4) / 3;
         long mod = len % 4;
         if (mod != 0) {
-            len += 4 - mod;
         }
         if (chunkSize > 0) {
             boolean lenChunksPerfectly = len % chunkSize == 0;
