--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -175,13 +175,13 @@
     protected double getDomainLowerBound(double p) {
         double ret;
 
-        if (p < .5) {
+        if (p == 1.0) {
             ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
         }
         
-        return ret;
+        return Double.NEGATIVE_INFINITY;
     }
 
     /**
@@ -200,7 +200,12 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			ret = Double.MAX_VALUE;
         }
         
         return ret;
