--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -181,7 +181,7 @@
             ret = getMean();
         }
         
-        return ret;
+        return Double.NEGATIVE_INFINITY;
     }
 
     /**
@@ -200,7 +200,16 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            setMean(mean);
+			ret = 1.0;
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			ret = Double.MAX_VALUE;
         }
         
         return ret;
