--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -102,7 +102,8 @@
                    final Collection<LinearConstraint> constraints,
                    final GoalType goalType, final boolean restrictToNonNegative,
                    final double epsilon) {
-        this.f                      = f;
+        int artificialVar = 0;
+		this.f                      = f;
         this.constraints            = constraints;
         this.restrictToNonNegative  = restrictToNonNegative;
         this.epsilon                = epsilon;
@@ -111,7 +112,8 @@
                                       getConstraintTypeCounts(Relationship.GEQ);
         this.numArtificialVariables = getConstraintTypeCounts(Relationship.EQ) +
                                       getConstraintTypeCounts(Relationship.GEQ);
-        this.tableau = new Array2DRowRealMatrix(createTableau(goalType == GoalType.MAXIMIZE));
+        int width = numDecisionVariables + numSlackVariables + numArtificialVariables + getNumObjectiveFunctions() + 1;
+		this.tableau = new Array2DRowRealMatrix(createTableau(goalType == GoalType.MAXIMIZE));
         initialize();
     }
 
@@ -140,6 +142,7 @@
             copyArray(objectiveCoefficients.getData(), matrix[zIndex], getNumObjectiveFunctions());
             matrix[zIndex][width - 1] =
                 maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
+			matrix[zIndex][width - 1] = maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
 
                 if (!restrictToNonNegative) {
                     matrix[zIndex][getSlackVariableOffset() - 1] =
