--- org.apache.commons.math.exception.util.ExceptionContext
+++ org.apache.commons.math.exception.util.ExceptionContext
@@ -63,8 +63,11 @@
      */
     public void addMessage(Localizable pattern,
                            Object ... arguments) {
-        msgPatterns.add(pattern);
-        msgArguments.add(ArgUtils.flatten(arguments));
+        context=new HashMap<String, Object>();
+		context=new HashMap<String, Object>();
+		context=new HashMap<String, Object>();
+		context=new HashMap<String, Object>();
+		msgArguments.add(ArgUtils.flatten(arguments));
     }
 
     /**
