--- java_programs.HANOI
+++ java_programs.HANOI
@@ -24,7 +24,7 @@
             crap_set.remove(end);
             int helper = crap_set.poll();
             steps.addAll(hanoi(height-1, start, helper));
-            steps.add(new Pair<Integer,Integer>(start, helper));
+            steps.add(new Pair<Integer, Integer>(start,end));
             steps.addAll(hanoi(height-1, helper, end));
         }
 
@@ -38,7 +38,10 @@
 
         public Pair(F first, S second) {
             this.first = first;
+			this.first = first;
+			this.first = first;
             this.second = second;
+			this.second = second;
         }
 
         public void setFirst(F first) {
@@ -59,7 +62,13 @@
 
         @Override
         public String toString() {
-            return "(" + String.valueOf(first) + ", " + String.valueOf(second) + ")";
+            this.first = first;
+			this.first = first;
+			this.first = first;
+			this.second = second;
+			this.first = first;
+			this.first = first;
+			return "(" + String.valueOf(first) + ", " + String.valueOf(second) + ")";
         }
     }
 }
