--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,6 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
@@ -241,7 +240,6 @@
      * if the dimensions do not match.
      */
     public OpenMapRealVector add(OpenMapRealVector v) {
-        checkVectorDimensions(v.getDimension());
         boolean copyThis = entries.size() > v.entries.size();
         OpenMapRealVector res = copyThis ? this.copy() : v.copy();
         Iterator iter = copyThis ? v.entries.iterator() : entries.iterator();
@@ -316,7 +314,6 @@
      * if the dimensions do not match.
      */
     public double dotProduct(OpenMapRealVector v) {
-        checkVectorDimensions(v.getDimension());
         boolean thisIsSmaller  = entries.size() < v.entries.size();
         Iterator iter = thisIsSmaller  ? entries.iterator() : v.entries.iterator();
         OpenIntToDoubleHashMap larger = thisIsSmaller  ? v.entries : entries;
@@ -340,7 +337,6 @@
 
     /** {@inheritDoc} */
     public OpenMapRealVector ebeDivide(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
@@ -365,7 +361,6 @@
 
     /** {@inheritDoc} */
     public OpenMapRealVector ebeMultiply(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
@@ -455,7 +450,6 @@
     /** {@inheritDoc} */
     @Override
     public double getDistance(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return getDistance((OpenMapRealVector) v);
         }
@@ -512,7 +506,6 @@
     /** {@inheritDoc} */
     @Override
     public double getL1Distance(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return getL1Distance((OpenMapRealVector) v);
         }
@@ -563,7 +556,6 @@
     /** {@inheritDoc} */
     @Override
     public double getLInfDistance(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return getLInfDistance((OpenMapRealVector) v);
         }
@@ -647,7 +639,6 @@
 
     /** {@inheritDoc} */
     public RealVector projection(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         return v.mapMultiply(dotProduct(v) / v.dotProduct(v));
     }
 
@@ -703,7 +694,6 @@
      * if the dimensions do not match.
      */
     public OpenMapRealVector subtract(OpenMapRealVector v) {
-        checkVectorDimensions(v.getDimension());
         OpenMapRealVector res = copy();
         Iterator iter = v.getEntries().iterator();
         while (iter.hasNext()) {
