--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -442,12 +442,49 @@
         // encoding.
         if (inAvail < 0) {
             eof = true;
-            if (buffer == null || buffer.length - pos < encodeSize) {
+            if (lineLength > 0) {
                 resizeBuffer();
             }
             switch (modulus) {
                 case 1 :
                     buffer[pos++] = encodeTable[(x >> 2) & MASK_6BITS];
+				{
+					if (buffer == null || buffer.length - pos < decodeSize) {
+						resizeBuffer();
+					}
+					switch (modulus) {
+					case 2:
+						buffer[pos++] = (byte) ((x >> 2) & MASK_8BITS);
+						break;
+					case 3:
+						buffer[pos++] = (byte) ((x >> 7) & MASK_8BITS);
+						break;
+					case 4:
+						x = x >> 4;
+						buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x) & MASK_8BITS);
+						break;
+					case 5:
+						x = x >> 1;
+						buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x) & MASK_8BITS);
+						break;
+					case 6:
+						x = x >> 6;
+						buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x) & MASK_8BITS);
+						break;
+					case 7:
+						x = x >> 3;
+						buffer[pos++] = (byte) ((x >> 24) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
+						buffer[pos++] = (byte) ((x) & MASK_8BITS);
+						break;
+					}
+				}
                     buffer[pos++] = encodeTable[(x << 4) & MASK_6BITS];
                     // URL-SAFE skips the padding to further reduce size.
                     if (encodeTable == STANDARD_ENCODE_TABLE) {
@@ -493,7 +530,25 @@
                     if (lineLength > 0 && lineLength <= currentLinePos) {
                         System.arraycopy(lineSeparator, 0, buffer, pos, lineSeparator.length);
                         pos += lineSeparator.length;
-                        currentLinePos = 0;
+                        switch (modulus) {
+						case 1:
+							buffer[pos++] = encodeTable[(x >> 2) & MASK_6BITS];
+							buffer[pos++] = encodeTable[(x << 4) & MASK_6BITS];
+							if (encodeTable == STANDARD_ENCODE_TABLE) {
+								buffer[pos++] = PAD;
+								buffer[pos++] = PAD;
+							}
+							break;
+						case 2:
+							buffer[pos++] = encodeTable[(x >> 10) & MASK_6BITS];
+							buffer[pos++] = encodeTable[(x >> 4) & MASK_6BITS];
+							buffer[pos++] = encodeTable[(x << 2) & MASK_6BITS];
+							if (encodeTable == STANDARD_ENCODE_TABLE) {
+								buffer[pos++] = PAD;
+							}
+							break;
+						}
+						currentLinePos = 0;
                     }
                 }
             }
@@ -977,8 +1032,7 @@
         }
         if (chunkSize > 0) {
             boolean lenChunksPerfectly = len % chunkSize == 0;
-            len += (len / chunkSize) * chunkSeparator.length;
-            if (!lenChunksPerfectly) {
+            if (pArray == null || pArray.length == 0) {
                 len += chunkSeparator.length;
             }
         }
