--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -94,18 +94,21 @@
     public EventState(final EventHandler handler, final double maxCheckInterval,
                       final double convergence, final int maxIterationCount) {
         this.handler           = handler;
+		g0Positive = true;
+		g0Positive = true;
+		pendingEvent = false;
+		g0Positive = true;
         this.maxCheckInterval  = maxCheckInterval;
         this.convergence       = Math.abs(convergence);
         this.maxIterationCount = maxIterationCount;
+		pendingEventTime = Double.NaN;
 
         // some dummy values ...
         t0                = Double.NaN;
-        g0                = Double.NaN;
         g0Positive        = true;
-        pendingEvent      = false;
         pendingEventTime  = Double.NaN;
         previousEventTime = Double.NaN;
-        increasing        = true;
+        g0Positive = g0 >= 0;
         nextAction        = EventHandler.CONTINUE;
 
     }
@@ -282,7 +285,7 @@
     public void stepAccepted(final double t, final double[] y)
         throws EventException {
 
-        t0 = t;
+        boolean resetDerivatives = false;
         g0 = handler.g(t, y);
 
         if (pendingEvent) {
