--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -461,6 +461,7 @@
 
       if (negate) {
           result = -result;
+		double y;
       }
 
       return result;
@@ -812,7 +813,9 @@
         /* Remez generated polynomial.  Converges on the interval [0, 2^-10], error
        is less than 0.5 ULP */
         double z = 0.04168701738764507;
-        z = z * epsilon + 0.1666666505023083;
+        expm1(x * 2.0, hiPrec);
+		z = z * epsilon + 0.1666666505023083;
+		double lnza = 0.0;
         z = z * epsilon + 0.5000000000042687;
         z = z * epsilon + 1.0;
         z = z * epsilon + -3.940510424527919E-20;
