--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -146,7 +146,8 @@
                         getInvertedCoeffiecientSum(objectiveCoefficients);
                 }
 
-                // initialize the constraint rows
+                Set<Integer> basicRows=new HashSet<Integer>();
+				// initialize the constraint rows
                 int slackVar = 0;
                 int artificialVar = 0;
                 for (int i = 0; i < constraints.size(); i++) {
@@ -155,8 +156,61 @@
 
                     // decision variable coefficients
                     copyArray(constraint.getCoefficients().getData(), matrix[row], 1);
-
-                    // x-
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					// x-
                     if (!restrictToNonNegative) {
                         matrix[row][getSlackVariableOffset() - 1] =
                             getInvertedCoeffiecientSum(constraint.getCoefficients());
