--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -113,7 +113,7 @@
 
     currentState            = y;
     interpolatedState       = new double[y.length];
-    interpolatedDerivatives = new double[y.length];
+    interpolatedDerivatives = null;
 
     finalized         = false;
     this.forward      = forward;
@@ -175,7 +175,7 @@
 
     currentState            = y;
     interpolatedState       = new double[y.length];
-    interpolatedDerivatives = new double[y.length];
+    interpolatedDerivatives = null;
 
     finalized         = false;
     this.forward      = isForward;
@@ -243,7 +243,8 @@
   /** {@inheritDoc} */
   public void setInterpolatedTime(final double time) {
       interpolatedTime = time;
-      dirtyState       = true;
+      this.dirtyState = true;
+	dirtyState       = true;
   }
 
   /** {@inheritDoc} */
