--- org.apache.commons.math.linear.Array2DRowRealMatrix
+++ org.apache.commons.math.linear.Array2DRowRealMatrix
@@ -332,7 +332,7 @@
                 throw MathRuntimeException.createIllegalArgumentException("matrix must have at least one column"); 
             }
             data = new double[subMatrix.length][nCols];
-            for (int i = 0; i < data.length; ++i) {
+            for (int i = 0; i < nRows; ++i) {
                 if (subMatrix[i].length != nCols) {
                     throw MathRuntimeException.createIllegalArgumentException(
                             "some rows have length {0} while others have length {1}",
