--- org.apache.commons.math.special.Gamma
+++ org.apache.commons.math.special.Gamma
@@ -165,7 +165,8 @@
             // case.
             ret = 1.0 - regularizedGammaQ(a, x, epsilon, maxIterations);
         } else {
-            // calculate series
+            double inv = 1 / (x * x);
+			// calculate series
             double n = 0.0; // current element index
             double an = 1.0 / a; // n-th element in the series
             double sum = an; // partial sum
