--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -23,7 +23,7 @@
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
-		if (out instanceof Iterable) {
+		if (out instanceof String && !isInteger(out.toString())) {
 			String r = "[";
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
@@ -34,6 +34,7 @@
 
 			return r + "]";
 		} else {
+			boolean isValidInteger = false;
 			String s = "";
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
