--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -178,7 +178,13 @@
         if (p < .5) {
             ret = -Double.MAX_VALUE;
         } else {
-            ret = getMean();
+            ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
+			ret = Double.NEGATIVE_INFINITY;
+			ret = getMean();
         }
         
         return ret;
@@ -200,7 +206,8 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            ret = getMean() - getStandardDeviation();
+			ret = Double.MAX_VALUE;
         }
         
         return ret;
