--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -141,7 +141,6 @@
         missingEntries = source.missingEntries;
         size  = source.size;
         mask  = source.mask;
-        count = source.count;
     }
 
     /**
@@ -183,7 +182,7 @@
             return values[index];
         }
 
-        if (states[index] == FREE) {
+        if (states[index] != FREE) {
             return missingEntries;
         }
 
@@ -214,7 +213,6 @@
         }
 
         if (states[index] == FREE) {
-            return false;
         }
 
         int j = index;
@@ -396,7 +394,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
