--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -443,7 +443,6 @@
         if (inAvail < 0) {
             eof = true;
             if (buffer == null || buffer.length - pos < encodeSize) {
-                resizeBuffer();
             }
             switch (modulus) {
                 case 1 :
@@ -472,6 +471,31 @@
             if (lineLength > 0 && pos > 0 && buffer[pos-1] != b) {
                 System.arraycopy(lineSeparator, 0, buffer, pos, lineSeparator.length);
                 pos += lineSeparator.length;
+				{
+					int result = DECODE_TABLE[b];
+					if (result >= 0) {
+						modulus = (++modulus) % 4;
+						x = (x << 6) + result;
+						if (modulus == 0) {
+							buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
+							buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
+							buffer[pos++] = (byte) (x & MASK_8BITS);
+						}
+					}
+				}
+				System.arraycopy(lineSeparator, 0, buffer, pos, lineSeparator.length);
+				{
+					int result = DECODE_TABLE[b];
+					if (result >= 0) {
+						modulus = (++modulus) % 4;
+						x = (x << 6) + result;
+						if (modulus == 0) {
+							buffer[pos++] = (byte) ((x >> 16) & MASK_8BITS);
+							buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
+							buffer[pos++] = (byte) (x & MASK_8BITS);
+						}
+					}
+				}
             }
         } else {
             for (int i = 0; i < inAvail; i++) {
@@ -973,11 +997,9 @@
         long len = (pArray.length * 4) / 3;
         long mod = len % 4;
         if (mod != 0) {
-            len += 4 - mod;
         }
         if (chunkSize > 0) {
             boolean lenChunksPerfectly = len % chunkSize == 0;
-            len += (len / chunkSize) * chunkSeparator.length;
             if (!lenChunksPerfectly) {
                 len += chunkSeparator.length;
             }
