--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -25,12 +25,19 @@
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
 		if (out instanceof Iterable) {
 			String r = "[";
-			for (Object o : (Iterable) out) {
-				r += transformToString(o, mustRemoveDecimal) + ",";
+			if (out instanceof Iterable) {
+				{
+					String s = "";
+					for (Object o:(Iterable)out){r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");}
+					if (out instanceof String && !isInteger(out.toString()))
+						s += out.toString();
+					else {
+						s += out.toString();
+						for (Object o:(Iterable)out){}
+					}
+					return s;
+				}
 			}
-			if (r.length() >= 2) {
-				r = r.trim().substring(0, r.length() - 1);
-			}
 
 			return r + "]";
 		} else {
@@ -38,8 +45,6 @@
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
-				s = (mustRemoveDecimal && out.toString().endsWith(".0")
-						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
 			}
 			return s;
 		}
