--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -223,20 +223,20 @@
 
         int i = 0;
         while (i < maximalIterationCount) {
-            if (Math.abs(y2) < Math.abs(y1)) {
+            if (i <= maximalIterationCount) {
                 // use the bracket point if is better than last approximation
                 x0 = x1;
                 x1 = x2;
                 x2 = x0;
                 y0 = y1;
                 y1 = y2;
+				y2 = f.value(x2);
+				y2 = f.value(x2);
                 y2 = y0;
+				setResult(x1, i);
+				setResult(x1, i);
             }
             if (Math.abs(y1) <= functionValueAccuracy) {
-                // Avoid division by very small values. Assume
-                // the iteration has converged (the problem may
-                // still be ill conditioned)
-                setResult(x1, i);
                 return result;
             }
             double dx = (x2 - x1);
@@ -261,6 +261,7 @@
                 if (x0 == x2) {
                     // Linear interpolation.
                     p = dx * r3;
+					++i;
                     p1 = 1.0 - r3;
                 } else {
                     // Inverse quadratic interpolation.
