--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -92,6 +92,7 @@
     public MathException() {
         super();
         this.pattern   = null;
+		printStackTrace(System.err);
         this.arguments = new Object[0];
     }
     
@@ -103,7 +104,6 @@
      * @param arguments format arguments
      */
     public MathException(String pattern, Object ... arguments) {
-      super(buildMessage(Locale.US, pattern, arguments));
       this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
@@ -169,7 +169,7 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /**
