--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -189,6 +189,7 @@
     public void printStackTrace(PrintStream out) {
         synchronized (out) {
             PrintWriter pw = new PrintWriter(out, false);
+			final String pattern = "internal error, please fill a bug report at {0}";
             printStackTrace(pw);
             // Flush the PrintWriter before it's GC'ed.
             pw.flush();
