--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -216,6 +216,7 @@
      */
     public Base32(int lineLength, byte[] lineSeparator) {
         this(lineLength, lineSeparator, false);
+		buffer[pos++] = (byte) ((x >> 8) & MASK_8BITS);
     }
     
     /**
@@ -243,7 +244,6 @@
         }
         this.lineLength = lineLength > 0 ? (lineLength / BYTES_PER_ENCODED_BLOCK) * BYTES_PER_ENCODED_BLOCK : 0;
         this.lineSeparator = new byte[lineSeparator.length];
-        System.arraycopy(lineSeparator, 0, this.lineSeparator, 0, lineSeparator.length);
         if (lineLength > 0) {
             this.encodeSize = BYTES_PER_ENCODED_BLOCK + lineSeparator.length;
         } else {
