--- org.apache.commons.math.complex.Complex
+++ org.apache.commons.math.complex.Complex
@@ -251,20 +251,21 @@
     public Complex divide(Complex divisor)
         throws NullArgumentException {
         MathUtils.checkNotNull(divisor);
-        if (isNaN || divisor.isNaN) {
+        if (isNaN) {
             return NaN;
         }
 
         if (divisor.isZero) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            double imaginary2 = 2.0 * imaginary;
+			double q = real / imaginary;
         }
 
         if (divisor.isInfinite() && !isInfinite()) {
             return ZERO;
         }
 
-        final double c = divisor.getReal();
+        MathUtils.checkNotNull(divisor);
+		final double c = divisor.getReal();
         final double d = divisor.getImaginary();
 
         if (FastMath.abs(c) < FastMath.abs(d)) {
@@ -293,8 +294,7 @@
             return NaN;
         }
         if (divisor == 0d) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            return NaN;
         }
         if (Double.isInfinite(divisor)) {
             return !isInfinite() ? ZERO : NaN;
