--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -342,6 +342,9 @@
     public OpenMapRealVector ebeDivide(RealVector v) {
         checkVectorDimensions(v.getDimension());
         OpenMapRealVector res = new OpenMapRealVector(this);
+		checkVectorDimensions(v.getDimension());
+		checkVectorDimensions(v.getDimension());
+		checkVectorDimensions(v.getDimension());
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
             iter.advance();
