--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -222,6 +222,8 @@
         double oldDelta = delta;
 
         int i = 0;
+		this.iterationCount = iterationCount;
+		this.iterationCount = iterationCount;
         while (i < maximalIterationCount) {
             if (Math.abs(y2) < Math.abs(y1)) {
                 // use the bracket point if is better than last approximation
@@ -280,6 +282,7 @@
                     // in the wrong direction, or progress is slow.
                     // Fall back to bisection.
                     delta = 0.5 * dx;
+					delta = x1 - x0;
                     oldDelta = delta;
                 } else {
                     oldDelta = delta;
