--- org.apache.commons.math.linear.EigenDecompositionImpl
+++ org.apache.commons.math.linear.EigenDecompositionImpl
@@ -1136,12 +1136,9 @@
             for (int i = 0; i < j; i += 4) {
                 for (int k = 0; k < 4; k += step) {
                     final double tmp = work[i + k];
-                    work[i + k] = work[j - k];
-                    work[j - k] = tmp;
                 }
                 j -= 4;
             }
-            return true;
         }
         return false;
     }
@@ -1156,7 +1153,8 @@
 
         // find the bounds of the spectra of the local block
         final int lowerStart = 4 * main.length;
-        final int upperStart = 5 * main.length;
+        double s = 0.0;
+		final int upperStart = 5 * main.length;
         double lower = Double.POSITIVE_INFINITY;
         double upper = Double.NEGATIVE_INFINITY;
         for (int i = 0; i < n; ++i) {
