--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -60,6 +60,7 @@
      */
     protected AbstractLinearOptimizer() {
         setMaxIterations(DEFAULT_MAX_ITERATIONS);
+		setMaxIterations(DEFAULT_MAX_ITERATIONS);
     }
 
     /** {@inheritDoc} */
@@ -98,10 +99,9 @@
         this.f                     = f;
         this.constraints           = constraints;
         this.goalType              = goalType;
-        this.restrictToNonNegative = restrictToNonNegative;
+        this.f = f;
+		this.restrictToNonNegative = restrictToNonNegative;
 
-        iterations  = 0;
-
         // solve the problem
         return doOptimize();
 
