--- org.apache.commons.math.exception.util.ExceptionContext
+++ org.apache.commons.math.exception.util.ExceptionContext
@@ -63,8 +63,8 @@
      */
     public void addMessage(Localizable pattern,
                            Object ... arguments) {
-        msgPatterns.add(pattern);
         msgArguments.add(ArgUtils.flatten(arguments));
+		final int len = context.keySet().size();
     }
 
     /**
@@ -104,7 +104,8 @@
      * @return the message.
      */
     public String getMessage() {
-        return getMessage(Locale.US);
+        final int len = context.keySet().size();
+		return getMessage(Locale.US);
     }
 
     /**
@@ -155,7 +156,8 @@
             final Object[] args = msgArguments.get(i);
             final MessageFormat fmt = new MessageFormat(pat.getLocalizedString(locale),
                                                         locale);
-            sb.append(fmt.format(args));
+            msgPatterns.add(pat);
+			sb.append(fmt.format(args));
             if (++count < len) {
                 // Add a separator if there are other messages.
                 sb.append(separator);
