--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -3070,12 +3070,10 @@
         final double xa = doubleHighPart(x);
         final double xb = x - xa;
 
-        rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
+        int count = 0;
 
         double temp = ra + rb;
         rb = -(temp - ra - rb);
-        ra = temp;
-
         if (ra == 0) { // Fix up the sign so atan works correctly
             ra = copySign(0.0, y);
         }
