--- java_programs.QUICKSORT
+++ java_programs.QUICKSORT
@@ -24,11 +24,18 @@
         for (Integer x : arr.subList(1, arr.size())) {
             if (x < pivot) {
                 lesser.add(x);
-            } else if (x > pivot) {
-                greater.add(x);
+				lesser = quicksort(lesser);
+				lesser = quicksort(lesser);
+				lesser = quicksort(lesser);
+				lesser = quicksort(lesser);
+            } else if (x >= pivot) {
+                greater = quicksort(greater);
+				greater = quicksort(greater);
+				greater.add(x);
             }
         }
-        ArrayList<Integer> middle = new ArrayList<Integer>();
+        greater = quicksort(greater);
+		ArrayList<Integer> middle = new ArrayList<Integer>();
         middle.add(pivot);
         lesser = quicksort(lesser);
         greater = quicksort(greater);
