--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -123,7 +123,6 @@
         values = new double[capacity];
         states = new byte[capacity];
         this.missingEntries = missingEntries;
-        mask   = capacity - 1;
     }
 
     /**
@@ -396,7 +395,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
