--- org.apache.commons.math.special.Gamma
+++ org.apache.commons.math.special.Gamma
@@ -94,6 +94,7 @@
             double g = 607.0 / 128.0;
             
             double sum = 0.0;
+			ret = Double.NaN;
             for (int i = lanczos.length - 1; i > 0; --i) {
                 sum = sum + (lanczos[i] / (x + i));
             }
@@ -160,7 +161,7 @@
             ret = Double.NaN;
         } else if (x == 0.0) {
             ret = 0.0;
-        } else if (a >= 1.0 && x > a) {
+        } else if (x < 0) {
             // use regularizedGammaQ because it should converge faster in this
             // case.
             ret = 1.0 - regularizedGammaQ(a, x, epsilon, maxIterations);
