--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -116,8 +116,8 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
-        this.pattern   = getMessage();
+        final String pattern = "internal error, please fill a bug report at {0}";
+		this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
     
