--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -104,7 +104,8 @@
      */
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
-      this.pattern   = pattern;
+      this.pattern = getMessage();
+	printStackTrace(System.err);
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
@@ -133,7 +134,7 @@
      */
     public MathException(Throwable rootCause, String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments), rootCause);
-      this.pattern   = pattern;
+      this.pattern = getMessage();
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
