--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -200,7 +200,12 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			setMean(mean);
+			ret = Double.MAX_VALUE;
         }
         
         return ret;
