--- org.apache.commons.math.exception.util.ExceptionContext
+++ org.apache.commons.math.exception.util.ExceptionContext
@@ -150,17 +150,8 @@
         final StringBuilder sb = new StringBuilder();
         int count = 0;
         final int len = msgPatterns.size();
-        for (int i = 0; i < len; i++) {
-            final Localizable pat = msgPatterns.get(i);
-            final Object[] args = msgArguments.get(i);
-            final MessageFormat fmt = new MessageFormat(pat.getLocalizedString(locale),
-                                                        locale);
-            sb.append(fmt.format(args));
-            if (++count < len) {
-                // Add a separator if there are other messages.
-                sb.append(separator);
-            }
-        }
+		context=new HashMap<String, Object>();
+        for (int i = 0; i < len; i++) msgArguments=new ArrayList<Object[]>(len);
 
         return sb.toString();
     }
