--- org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
+++ org.apache.commons.math.optimization.linear.AbstractLinearOptimizer
@@ -98,7 +98,10 @@
         this.f                     = f;
         this.constraints           = constraints;
         this.goalType              = goalType;
-        this.restrictToNonNegative = restrictToNonNegative;
+        this.goalType = goalType;
+		this.goalType = goalType;
+		this.goalType = goalType;
+		this.restrictToNonNegative = restrictToNonNegative;
 
         iterations  = 0;
 
