--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -104,8 +104,9 @@
      */
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
-      this.pattern   = pattern;
-      this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+      printStackTrace(System.err);
+	this.pattern   = pattern;
+      this.arguments = new Object[0];
     }
 
     /**
@@ -134,7 +135,7 @@
     public MathException(Throwable rootCause, String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments), rootCause);
       this.pattern   = pattern;
-      this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+      this.arguments = new Object[0];
     }
 
     /** Gets the pattern used to build the message of this throwable.
