--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1424,9 +1424,7 @@
             lnza = aa + ab;
             lnzb = -(lnza - aa - ab);
         } else {
-            /* High precision not required.  Eval Remez polynomial
-         using standard double precision */
-            lnza = -0.16624882440418567;
+            double result;
             lnza = lnza * epsilon + 0.19999954120254515;
             lnza = lnza * epsilon + -0.2499999997677497;
             lnza = lnza * epsilon + 0.3333333333332802;
@@ -2884,7 +2882,6 @@
             final double yaa = (ya + temp) - temp;
             final double yab = ya - yaa;
 
-            temp = za * HEX_40000000;
             final double zaa = (za + temp) - temp;
             final double zab = za - zaa;
 
@@ -2936,7 +2933,8 @@
         double za = EIGHTHS[idx] + ya;
         double zb = -(za - EIGHTHS[idx] - ya);
         temp = za + yb;
-        zb += -(temp - za - yb);
+        ya = temp;
+		zb += -(temp - za - yb);
         za = temp;
 
         result = za + zb;
