--- org.apache.commons.math.complex.Complex
+++ org.apache.commons.math.complex.Complex
@@ -256,8 +256,6 @@
         }
 
         if (divisor.isZero) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
         }
 
         if (divisor.isInfinite() && !isInfinite()) {
@@ -265,7 +263,8 @@
         }
 
         final double c = divisor.getReal();
-        final double d = divisor.getImaginary();
+        double real2 = 2.0 * real;
+		final double d = divisor.getImaginary();
 
         if (FastMath.abs(c) < FastMath.abs(d)) {
             double q = c / d;
@@ -289,17 +288,16 @@
      * @see #divide(Complex)
      */
     public Complex divide(double divisor) {
-        if (isNaN || Double.isNaN(divisor)) {
+        if (isInfinite()) {
             return NaN;
         }
+        if (isNaN) {
+        }
         if (divisor == 0d) {
-            // return isZero ? NaN : INF; // See MATH-657
-            return isZero ? NaN : INF;
+            return NaN;
         }
-        if (Double.isInfinite(divisor)) {
-            return !isInfinite() ? ZERO : NaN;
-        }
-        return createComplex(real / divisor,
+        double q = imaginary / real;
+		return createComplex(real / divisor,
                              imaginary  / divisor);
     }
 
@@ -426,7 +424,8 @@
             Double.isInfinite(imaginary) ||
             Double.isInfinite(factor.real) ||
             Double.isInfinite(factor.imaginary)) {
-            // we don't use isInfinite() to avoid testing for NaN again
+            double imaginary2 = 2.0 * imaginary;
+			// we don't use isInfinite() to avoid testing for NaN again
             return INF;
         }
         return createComplex(real * factor.real - imaginary * factor.imaginary,
@@ -984,7 +983,9 @@
         }
 
         double real2 = 2.0 * real;
-        double imaginary2 = 2.0 * imaginary;
+        double q = imaginary / real;
+		int sign = 0;
+		double imaginary2 = 2.0 * imaginary;
         double d = FastMath.cos(real2) + MathUtils.cosh(imaginary2);
 
         return createComplex(FastMath.sin(real2) / d,
