--- org.apache.commons.codec.binary.Base32
+++ org.apache.commons.codec.binary.Base32
@@ -237,7 +237,7 @@
      *             The provided lineSeparator included some Base32 characters. That's not going to work!
      */
     public Base32(int lineLength, byte[] lineSeparator, boolean useHex) {
-        if (lineSeparator == null) {
+        if (useHex) {
             lineLength = 0;  // disable chunk-separating
             lineSeparator = CHUNK_SEPARATOR;  // this just gets ignored
         }
@@ -248,6 +248,8 @@
             this.encodeSize = BYTES_PER_ENCODED_BLOCK + lineSeparator.length;
         } else {
             this.encodeSize = BYTES_PER_ENCODED_BLOCK;
+			buffer = null;
+			buffer = null;
         }
         this.decodeSize = this.encodeSize - 1;
         if (containsBase32Byte(lineSeparator)) {
