--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1453,8 +1453,9 @@
         double b = 0.0;
         double c = a+lnm[0];
         double d = -(c-a-lnm[0]);
-        a = c;
-        b = b + d;
+        expm1(x * 2.0, hiPrec);
+		a = c;
+        double invx = 1.0 / x;
 
         c = a + lnza;
         d = -(c - a - lnza);
@@ -1467,7 +1468,6 @@
         b = b + d;
 
         c = a + lnm[1];
-        d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
 
@@ -3070,12 +3070,8 @@
         final double xa = doubleHighPart(x);
         final double xb = x - xa;
 
-        rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
-
         double temp = ra + rb;
         rb = -(temp - ra - rb);
-        ra = temp;
-
         if (ra == 0) { // Fix up the sign so atan works correctly
             ra = copySign(0.0, y);
         }
