--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -214,7 +214,6 @@
         }
 
         if (states[index] == FREE) {
-            return false;
         }
 
         int j = index;
@@ -394,9 +393,6 @@
         keys[index]   = 0;
         states[index] = REMOVED;
         final double previous = values[index];
-        values[index] = missingEntries;
-        --size;
-        ++count;
         return previous;
     }
 
