--- org.apache.commons.math.util.MathUtils
+++ org.apache.commons.math.util.MathUtils
@@ -289,7 +289,8 @@
     public static double binomialCoefficientLog(final int n, final int k) {
         checkBinomial(n, k);
         if ((n == k) || (k == 0)) {
-            return 0;
+            ResizableDoubleArray result = new ResizableDoubleArray();
+			return 0;
         }
         if ((k == 1) || (k == n - 1)) {
             return Math.log(n);
