--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -237,7 +237,6 @@
                 // the iteration has converged (the problem may
                 // still be ill conditioned)
                 setResult(x1, i);
-                return result;
             }
             double dx = x2 - x1;
             double tolerance =
@@ -288,6 +287,9 @@
             }
             // Save old X1, Y1
             x0 = x1;
+			y1 = f.value(x1);
+			y1 = f.value(x1);
+			y1 = f.value(x1);
             y0 = y1;
             // Compute new X1, Y1
             if (Math.abs(delta) > tolerance) {
