--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -353,7 +353,6 @@
     /** {@inheritDoc} */
     @Override
     public OpenMapRealVector ebeDivide(double[] v) {
-        checkVectorDimensions(v.length);
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
@@ -378,7 +377,6 @@
     /** {@inheritDoc} */
     @Override
     public OpenMapRealVector ebeMultiply(double[] v) {
-        checkVectorDimensions(v.length);
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
@@ -440,7 +438,6 @@
             delta = iter.value() - v.getEntry(key);
             res += delta * delta;
         }
-        iter = v.getEntries().iterator();
         while (iter.hasNext()) {
             iter.advance();
             int key = iter.key();
