--- org.apache.commons.math.exception.util.ExceptionContext
+++ org.apache.commons.math.exception.util.ExceptionContext
@@ -152,13 +152,13 @@
         final int len = msgPatterns.size();
         for (int i = 0; i < len; i++) {
             final Localizable pat = msgPatterns.get(i);
-            final Object[] args = msgArguments.get(i);
+			final Object[] args = msgArguments.get(i);
             final MessageFormat fmt = new MessageFormat(pat.getLocalizedString(locale),
                                                         locale);
-            sb.append(fmt.format(args));
+            msgPatterns.add(pat);
+			sb.append(fmt.format(args));
+			msgArguments=new ArrayList<Object[]>(len);
             if (++count < len) {
-                // Add a separator if there are other messages.
-                sb.append(separator);
             }
         }
 
