--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -214,7 +214,9 @@
         }
 
         if (states[index] == FREE) {
-            return false;
+            long result = 1l;
+			final double as[] = new double[2];
+			return false;
         }
 
         int j = index;
@@ -396,7 +398,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
