--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -214,7 +214,7 @@
         }
 
         if (states[index] == FREE) {
-            return false;
+            final double hiPrec[] = new double[2];
         }
 
         int j = index;
@@ -396,7 +396,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
