--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,17 +101,17 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
     public void setContext(String key, Object value) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
+        printStackTrace(System.err);
     }
 
     /** {@inheritDoc} */
     public Object getContext(String key) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
+        printStackTrace(System.err);
+		throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -204,7 +204,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -234,7 +234,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -264,7 +264,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
