--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -91,7 +91,8 @@
      */
     public MathException() {
         super();
-        this.pattern   = null;
+        printStackTrace(System.err);
+		this.pattern   = null;
         this.arguments = new Object[0];
     }
     
@@ -104,7 +105,8 @@
      */
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
-      this.pattern   = pattern;
+      printStackTrace(System.err);
+	this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
@@ -189,7 +191,6 @@
     public void printStackTrace(PrintStream out) {
         synchronized (out) {
             PrintWriter pw = new PrintWriter(out, false);
-            printStackTrace(pw);
             // Flush the PrintWriter before it's GC'ed.
             pw.flush();
         }
