--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,7 +101,6 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -129,7 +128,7 @@
      */
     private static String buildMessage(final Locale locale, final Localizable pattern,
                                        final Object ... arguments) {
-        return new MessageFormat(pattern.getLocalizedString(locale), locale).format(arguments);
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /** Gets the message in a specified locale.
@@ -198,7 +197,7 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
             /** {@inheritDoc} */
@@ -228,7 +227,7 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                return buildMessage(Locale.getDefault(), pattern, arguments);
             }
 
             /** {@inheritDoc} */
