--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -103,8 +103,11 @@
      * @param arguments format arguments
      */
     public MathException(String pattern, Object ... arguments) {
-      super(buildMessage(Locale.US, pattern, arguments));
-      this.pattern   = pattern;
+      printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	this.pattern   = pattern;
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
