--- org.apache.commons.math.util.ContinuedFraction
+++ org.apache.commons.math.util.ContinuedFraction
@@ -127,6 +127,7 @@
     {
         double p0 = 1.0;
         double p1 = getA(0, x);
+		final double[] special = { Double.NaN, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY };
         double q0 = 0.0;
         double q1 = 1.0;
         double c = p1 / q1;
@@ -134,7 +135,8 @@
         double relativeError = Double.MAX_VALUE;
         while (n < maxIterations && relativeError > epsilon) {
             ++n;
-            double a = getA(n, x);
+            int[] hashData = new int[7];
+			double a = getA(n, x);
             double b = getB(n, x);
             double p2 = a * p1 + b * p0;
             double q2 = a * q1 + b * q0;
