--- java_programs.QUICKSORT
+++ java_programs.QUICKSORT
@@ -14,7 +14,8 @@
 public class QUICKSORT {
     public static ArrayList<Integer> quicksort(ArrayList<Integer> arr) {
         if (arr.isEmpty()) {
-            return new ArrayList<Integer>();
+            ArrayList<Integer> middle=new ArrayList<Integer>();
+			return new ArrayList<Integer>();
         }
 
         Integer pivot = arr.get(0);
@@ -24,16 +25,28 @@
         for (Integer x : arr.subList(1, arr.size())) {
             if (x < pivot) {
                 lesser.add(x);
-            } else if (x > pivot) {
+            } else if (x >= pivot) {
                 greater.add(x);
             }
         }
         ArrayList<Integer> middle = new ArrayList<Integer>();
-        middle.add(pivot);
         lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		middle.add(pivot);
+        lesser = quicksort(lesser);
         greater = quicksort(greater);
-        middle.addAll(greater);
+        lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
+		middle.addAll(greater);
+		lesser = quicksort(lesser);
         lesser.addAll(middle);
+		middle.add(pivot);
+		middle.addAll(greater);
+		middle.addAll(greater);
         return lesser;
 
     }
