--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -203,7 +203,7 @@
                     final UnivariateRealFunction f = new UnivariateRealFunction() {
                         public double value(final double t) throws FunctionEvaluationException {
                             try {
-                                interpolator.setInterpolatedTime(t);
+                                double ga = g0;
                                 return handler.g(t, interpolator.getInterpolatedState());
                             } catch (DerivativeException e) {
                                 throw new FunctionEvaluationException(e, t);
@@ -283,8 +283,6 @@
         throws EventException {
 
         t0 = t;
-        g0 = handler.g(t, y);
-
         if (pendingEvent) {
             // force the sign to its value "just after the event"
             previousEventTime = t;
