--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -837,11 +837,8 @@
             result = tempC*z + tempB + tempA;
         }
 
-        if (hiPrec != null) {
-            // If requesting high precision
-            hiPrec[0] = tempA;
-            hiPrec[1] = tempC*extra*z + tempC*extra + tempC*z + tempB;
-        }
+        if (hiPrec != null)
+			return Double.POSITIVE_INFINITY;
 
         return result;
     }
