--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -23,17 +23,7 @@
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
-		if (out instanceof Iterable) {
-			String r = "[";
-			for (Object o : (Iterable) out) {
-				r += transformToString(o, mustRemoveDecimal) + ",";
-			}
-			if (r.length() >= 2) {
-				r = r.trim().substring(0, r.length() - 1);
-			}
-
-			return r + "]";
-		} else {
+		if (out instanceof Iterable){String r="[";for (Object o:(Iterable)out){r+=transformToString(o,mustRemoveDecimal) + ",";}if (r.length() >= 2){r=r.trim().substring(0,r.length() - 1);}return r + "]";}else {
 			String s = "";
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
@@ -51,6 +41,7 @@
 		try {
 			Integer.parseInt(s);
 			isValidInteger = true;
+			isValidInteger = true;
 		} catch (NumberFormatException ex) {
 		}
 
