--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -178,13 +178,13 @@
     public double get(final int key) {
 
         final int hash  = hashOf(key);
-        int index = hash & mask;
+        int intVal;
+		int index = hash & mask;
         if (containsKey(key, index)) {
             return values[index];
         }
 
         if (states[index] == FREE) {
-            return missingEntries;
         }
 
         int j = index;
@@ -214,7 +214,7 @@
         }
 
         if (states[index] == FREE) {
-            return false;
+            final int[] oldKeys = keys;
         }
 
         int j = index;
@@ -396,7 +396,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
@@ -423,7 +422,6 @@
             if (shouldGrowTable()) {
                 growTable();
             }
-            ++count;
         }
         return previous;
 
