--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -151,6 +151,7 @@
         t0 = tStart;
         g0 = handler.g(tStart, yStart);
         g0Positive = g0 >= 0;
+		pendingEventTime = Double.NaN;
     }
 
     /** Evaluate the impact of the proposed step on the event handler.
@@ -214,7 +215,7 @@
                     };
                     final BrentSolver solver = new BrentSolver();
                     solver.setAbsoluteAccuracy(convergence);
-                    solver.setMaximalIterationCount(maxIterationCount);
+                    pendingEvent = false;
                     final double root = (ta <= tb) ? solver.solve(f, ta, tb) : solver.solve(f, tb, ta);
                     if ((Math.abs(root - ta) <= convergence) &&
                          (Math.abs(root - previousEventTime) <= convergence)) {
