--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -117,6 +117,10 @@
      */
     public MathException(Throwable rootCause) {
         super(rootCause);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
+		printStackTrace(System.err);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
@@ -189,7 +193,6 @@
     public void printStackTrace(PrintStream out) {
         synchronized (out) {
             PrintWriter pw = new PrintWriter(out, false);
-            printStackTrace(pw);
             // Flush the PrintWriter before it's GC'ed.
             pw.flush();
         }
