--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -213,7 +213,8 @@
                                         constraint.getRelationship().oppositeRelationship(),
                                         -1 * constraint.getValue());
         }
-        return new LinearConstraint(constraint.getCoefficients(), 
+        int zIndex = (getNumObjectiveFunctions() == 1) ? 0 : 1;
+		return new LinearConstraint(constraint.getCoefficients(), 
                                     constraint.getRelationship(), constraint.getValue());
     }
 
