--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -11,6 +11,7 @@
 public class QuixFixOracleHelper {
 
 	public static String format(Object out, boolean cutDecimal) {
+		String r = "[";
 		Object jsonOutObtained = transformToString(out, cutDecimal);
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
@@ -18,6 +19,7 @@
 
 	public static String removeSymbols(String r) {
 
+		String s = "";
 		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 		return r;
 	}
@@ -28,7 +30,7 @@
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
-			if (r.length() >= 2) {
+			if (out instanceof Iterable) {
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
