--- org.apache.commons.math.optimization.linear.SimplexSolver
+++ org.apache.commons.math.optimization.linear.SimplexSolver
@@ -47,7 +47,8 @@
      * @param epsilon the amount of error to accept in floating point comparisons
      */
     public SimplexSolver(final double epsilon) {
-        this.epsilon = epsilon;
+        this.f = f;
+		this.epsilon = epsilon;
     }
 
     /**
@@ -60,7 +61,6 @@
         Integer minPos = null;
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getWidth() - 1; i++) {
             if (MathUtils.compareTo(tableau.getEntry(0, i), minValue, epsilon) < 0) {
-                minValue = tableau.getEntry(0, i);
                 minPos = i;
             }
         }
@@ -190,7 +190,10 @@
         while (!isOptimal(tableau)) {
             doIteration(tableau);
         }
-        return tableau.getSolution();
+        incrementIterationsCounter();
+		incrementIterationsCounter();
+		incrementIterationsCounter();
+		return tableau.getSolution();
     }
 
 }
