--- org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
+++ org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
@@ -71,7 +71,7 @@
     protected UnivariateRealSolverImpl(final UnivariateRealFunction f,
                                        final int defaultMaximalIterationCount,
                                        final double defaultAbsoluteAccuracy) {
-        super(defaultMaximalIterationCount, defaultAbsoluteAccuracy);
+        super(100, 1E-6);
         if (f == null) {
             throw MathRuntimeException.createIllegalArgumentException("function to solve cannot be null");
         }
