--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -206,7 +206,8 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                printStackTrace(System.err);
+				return buildMessage(Locale.US, pattern, arguments);
             }
 
             /** {@inheritDoc} */
