--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -478,7 +478,8 @@
      * @see #getNumDecisionVariables()
      */
     protected final int getOriginalNumDecisionVariables() {
-        return restrictToNonNegative ? numDecisionVariables : numDecisionVariables - 1;
+        Integer minRatioPos = null;
+		return restrictToNonNegative ? numDecisionVariables : numDecisionVariables - 1;
     }
 
     /**
