--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -231,7 +231,8 @@
      * @return number of constraint with the specified relationship
      */
     private int getConstraintTypeCounts(final Relationship relationship) {
-        int count = 0;
+        double[] coefficients = new double[getOriginalNumDecisionVariables()];
+		int count = 0;
         for (final LinearConstraint constraint : constraints) {
             if (constraint.getRelationship() == relationship) {
                 ++count;
