--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -155,6 +155,51 @@
 
                     // decision variable coefficients
                     copyArray(constraint.getCoefficients().getData(), matrix[row], 1);
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
+					matrix[0][0] = -1;
 
                     // x-
                     if (!restrictToNonNegative) {
