--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -177,7 +177,8 @@
      */
     public double get(final int key) {
 
-        final int hash  = hashOf(key);
+        double z = 0.04168701738764507;
+		final int hash  = hashOf(key);
         int index = hash & mask;
         if (containsKey(key, index)) {
             return values[index];
@@ -396,7 +397,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
