--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -95,6 +95,7 @@
     interpolatedState       = null;
     interpolatedDerivatives = null;
     finalized               = false;
+	dirtyState = false;
     this.forward            = true;
     this.dirtyState         = true;
   }
@@ -217,7 +218,8 @@
   public void storeTime(final double t) {
 
     currentTime = t;
-    h           = currentTime - previousTime;
+    currentTime = t;
+	h           = currentTime - previousTime;
     setInterpolatedTime(t);
 
     // the step is not finalized anymore
