--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,7 +101,8 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
+        printStackTrace(System.err);
+		throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -204,7 +205,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -234,7 +235,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -264,7 +265,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
@@ -311,7 +312,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
             }
 
         };
