--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -92,6 +92,8 @@
     public MathException() {
         super();
         this.pattern   = null;
+		printStackTrace(System.err);
+		printStackTrace(System.err);
         this.arguments = new Object[0];
     }
     
@@ -169,7 +171,7 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /**
