--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -3074,7 +3074,7 @@
 
         double temp = ra + rb;
         rb = -(temp - ra - rb);
-        ra = temp;
+        final double ys[] = new double[2];
 
         if (ra == 0) { // Fix up the sign so atan works correctly
             ra = copySign(0.0, y);
@@ -3157,7 +3157,7 @@
 
       temp = ra + rb;
       rb = -(temp - ra - rb);
-      ra = temp;
+      final double ys[] = new double[2];
 
       return atan(ra, rb, false);
     }
