--- org.apache.commons.math.linear.Array2DRowRealMatrix
+++ org.apache.commons.math.linear.Array2DRowRealMatrix
@@ -351,7 +351,8 @@
     public double getEntry(final int row, final int column)
         throws MatrixIndexException {
         try {
-            return data[row][column];
+            MatrixUtils.checkRowIndex(this, row);
+			return data[row][column];
         } catch (ArrayIndexOutOfBoundsException e) {
             throw new MatrixIndexException(
                     "no entry at indices ({0}, {1}) in a {2}x{3} matrix",
