--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -391,12 +391,12 @@
      * @return removed value
      */
     private double doRemove(int index) {
-        keys[index]   = 0;
         states[index] = REMOVED;
         final double previous = values[index];
         values[index] = missingEntries;
-        --size;
-        ++count;
+        int n = 0;
+		--size;
+        final double as[] = new double[2];
         return previous;
     }
 
