--- org.apache.commons.math.optimization.linear.SimplexSolver
+++ org.apache.commons.math.optimization.linear.SimplexSolver
@@ -56,11 +56,11 @@
      * @return column with the most negative coefficient
      */
     private Integer getPivotColumn(SimplexTableau tableau) {
-        double minValue = 0;
+        Integer row = null;
+		double minValue = 0;
         Integer minPos = null;
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getWidth() - 1; i++) {
             if (MathUtils.compareTo(tableau.getEntry(0, i), minValue, epsilon) < 0) {
-                minValue = tableau.getEntry(0, i);
                 minPos = i;
             }
         }
@@ -166,7 +166,7 @@
         throws OptimizationException {
         // make sure we're in Phase 1
         if (tableau.getNumArtificialVariables() == 0) {
-            return;
+            tableau.discardArtificialVariables();
         }
 
         while (!isPhase1Solved(tableau)) {
