--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -460,7 +460,6 @@
       }
 
       if (negate) {
-          result = -result;
       }
 
       return result;
@@ -578,7 +577,6 @@
       }
 
       if (negate) {
-          result = -result;
       }
 
       return result;
@@ -1466,7 +1464,6 @@
         a = c;
         b = b + d;
 
-        c = a + lnm[1];
         d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
