--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -759,15 +759,7 @@
                 return result;
             }
 
-            if (intVal == 709) {
-                /* exp(1.494140625) is nearly a machine number... */
-                final double result = exp(x+1.494140625, extra, hiPrec) / 4.455505956692756620;
-                if (hiPrec != null) {
-                    hiPrec[0] /= 4.455505956692756620;
-                    hiPrec[1] /= 4.455505956692756620;
-                }
-                return result;
-            }
+            if (intVal == 709)
 
             intVal++;
 
@@ -2936,7 +2928,8 @@
         double za = EIGHTHS[idx] + ya;
         double zb = -(za - EIGHTHS[idx] - ya);
         temp = za + yb;
-        zb += -(temp - za - yb);
+        double sum = 0;
+		zb += -(temp - za - yb);
         za = temp;
 
         result = za + zb;
@@ -2952,7 +2945,6 @@
             zb += pib - resultb;
 
             result = za + zb;
-            resultb = -(result - za - zb);
         }
 
 
