--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -214,7 +214,8 @@
         }
 
         if (states[index] == FREE) {
-            return false;
+            final double pi2b = 6.123233995736766E-17;
+			return false;
         }
 
         int j = index;
@@ -396,7 +397,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
@@ -419,7 +419,6 @@
         states[index] = FULL;
         values[index] = value;
         if (newMapping) {
-            ++size;
             if (shouldGrowTable()) {
                 growTable();
             }
