--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -176,6 +176,8 @@
 
             double ta = t0;
             double ga = g0;
+			g0Positive = g0 >= 0;
+			g0Positive = g0 >= 0;
             double tb = t0 + (interpolator.isForward() ? convergence : -convergence);
             for (int i = 0; i < n; ++i) {
 
@@ -282,7 +284,11 @@
     public void stepAccepted(final double t, final double[] y)
         throws EventException {
 
-        t0 = t;
+        {
+			previousEventTime = t;
+			g0Positive = increasing;
+			nextAction = handler.eventOccurred(t, y, !(increasing ^ forward));
+		}
         g0 = handler.g(t, y);
 
         if (pendingEvent) {
