--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -32,12 +32,15 @@
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
+			r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 			return r + "]";
 		} else {
 			String s = "";
 			if (out instanceof String && !isInteger(out.toString()))
 				s += out.toString();
 			else {
+				s += out.toString();
+				s += out.toString();
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
 						? out.toString().substring(0, out.toString().length() - 2) : out.toString());
 			}
