--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -2880,7 +2880,6 @@
             zb += xb * TANGENT_TABLE_B[idx];
             ya = epsA / za;
 
-            temp = ya * HEX_40000000;
             final double yaa = (ya + temp) - temp;
             final double yab = ya - yaa;
 
@@ -3074,7 +3073,7 @@
 
         double temp = ra + rb;
         rb = -(temp - ra - rb);
-        ra = temp;
+        final double ys[] = new double[2];
 
         if (ra == 0) { // Fix up the sign so atan works correctly
             ra = copySign(0.0, y);
@@ -3157,7 +3156,7 @@
 
       temp = ra + rb;
       rb = -(temp - ra - rb);
-      ra = temp;
+      final double ys[] = new double[2];
 
       return atan(ra, rb, false);
     }
