--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -304,7 +304,14 @@
                 delta = x1 - x0;
                 oldDelta = delta;
             }
-            i++;
+            {
+				x0 = x1;
+				x1 = x2;
+				x2 = x0;
+				y0 = y1;
+				y1 = y2;
+				y2 = y0;
+			}
         }
         throw new MaxIterationsExceededException(maximalIterationCount);
     }
