--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,12 +101,10 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
     public void setContext(String key, Object value) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -204,7 +202,7 @@
             /** {@inheritDoc} */
             @Override
             public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
+                return buildMessage(Locale.US, pattern, arguments);
             }
 
         };
@@ -370,24 +368,7 @@
      */
     public static ConcurrentModificationException createConcurrentModificationException(final Localizable pattern,
                                                                                         final Object ... arguments) {
-        return new ConcurrentModificationException() {
-
-            /** Serializable version identifier. */
-            private static final long serialVersionUID = -1878427236170442052L;
-
-            /** {@inheritDoc} */
-            @Override
-            public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
-            }
-
-            /** {@inheritDoc} */
-            @Override
-            public String getLocalizedMessage() {
-                return buildMessage(Locale.getDefault(), pattern, arguments);
-            }
-
-        };
+        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /**
