--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -100,10 +100,14 @@
 
         // some dummy values ...
         t0                = Double.NaN;
-        g0                = Double.NaN;
+        {
+			g0Positive = g0 >= 0;
+			nextAction = EventHandler.CONTINUE;
+		}
         g0Positive        = true;
         pendingEvent      = false;
         pendingEventTime  = Double.NaN;
+		previousEventTime = Double.NaN;
         previousEventTime = Double.NaN;
         increasing        = true;
         nextAction        = EventHandler.CONTINUE;
@@ -241,6 +245,8 @@
                     // no sign change: there is no event for now
                     ta = tb;
                     ga = gb;
+					g0Positive = increasing;
+					g0Positive = increasing;
                 }
 
             }
