--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -101,12 +101,10 @@
     /** {@inheritDoc} */
     public void addMessage(Localizable pat,
                            Object ... args) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
     public void setContext(String key, Object value) {
-        throw new UnsupportedOperationException(DEPRECATION_MESSAGE);
     }
 
     /** {@inheritDoc} */
@@ -129,7 +127,7 @@
      */
     private static String buildMessage(final Locale locale, final Localizable pattern,
                                        final Object ... arguments) {
-        return new MessageFormat(pattern.getLocalizedString(locale), locale).format(arguments);
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /** Gets the message in a specified locale.
