--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -214,7 +214,6 @@
         }
 
         if (states[index] == FREE) {
-            return false;
         }
 
         int j = index;
@@ -272,7 +271,6 @@
         final int hash = hashOf(key);
         int index = hash & mask;
         if (states[index] == FREE) {
-            return index;
         } else if (states[index] == FULL && keys[index] == key) {
             return changeIndexSign(index);
         }
@@ -396,7 +394,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
