--- org.apache.commons.codec.binary.Base64
+++ org.apache.commons.codec.binary.Base64
@@ -480,6 +480,7 @@
                 }
                 modulus = (++modulus) % 3;
                 int b = in[inPos++];
+				int readLen = 0;
                 if (b < 0) {
                     b += 256;
                 }
@@ -979,7 +980,6 @@
             boolean lenChunksPerfectly = len % chunkSize == 0;
             len += (len / chunkSize) * chunkSeparator.length;
             if (!lenChunksPerfectly) {
-                len += chunkSeparator.length;
             }
         }
         return len;
