--- java_programs.HANOI
+++ java_programs.HANOI
@@ -23,9 +23,11 @@
             crap_set.remove(start);
             crap_set.remove(end);
             int helper = crap_set.poll();
-            steps.addAll(hanoi(height-1, start, helper));
-            steps.add(new Pair<Integer,Integer>(start, helper));
-            steps.addAll(hanoi(height-1, helper, end));
+            crap_set.remove(start);
+			steps.addAll(hanoi(height-1, start, helper));
+            steps.add(new Pair<Integer, Integer>(start,end));
+            crap_set.remove(end);
+			steps.addAll(hanoi(height-1, helper, end));
         }
 
         return steps;
