--- java_programs.QUICKSORT
+++ java_programs.QUICKSORT
@@ -23,13 +23,18 @@
 
         for (Integer x : arr.subList(1, arr.size())) {
             if (x < pivot) {
-                lesser.add(x);
-            } else if (x > pivot) {
+                greater = quicksort(greater);
+				lesser.add(x);
+            } else if (x >= pivot) {
                 greater.add(x);
             }
         }
         ArrayList<Integer> middle = new ArrayList<Integer>();
-        middle.add(pivot);
+        greater = quicksort(greater);
+		middle.add(pivot);
+		greater = quicksort(greater);
+		lesser = quicksort(lesser);
+		lesser = quicksort(lesser);
         lesser = quicksort(lesser);
         greater = quicksort(greater);
         middle.addAll(greater);
