--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -271,6 +271,27 @@
       // lazy evaluation of the state
       if (dirtyState) {
           final double oneMinusThetaH = currentTime - interpolatedTime;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
+		h = -h;
           final double theta = (h == 0) ? 0 : (h - oneMinusThetaH) / h;
           computeInterpolatedStateAndDerivatives(theta, oneMinusThetaH);
           dirtyState = false;
