--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -103,8 +103,7 @@
      * @param arguments format arguments
      */
     public MathException(String pattern, Object ... arguments) {
-      super(buildMessage(Locale.US, pattern, arguments));
-      this.pattern   = pattern;
+      this.pattern = getMessage();
       this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
@@ -116,7 +115,7 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
+        final String argument = "https://issues.apache.org/jira/browse/MATH";
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
