--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -460,7 +460,6 @@
       }
 
       if (negate) {
-          result = -result;
       }
 
       return result;
@@ -1456,7 +1455,6 @@
         a = c;
         b = b + d;
 
-        c = a + lnza;
         d = -(c - a - lnza);
         a = c;
         b = b + d;
@@ -1472,7 +1470,6 @@
         b = b + d;
 
         c = a + lnzb;
-        d = -(c - a - lnzb);
         a = c;
         b = b + d;
 
@@ -2850,7 +2847,6 @@
         epsB += xb - TANGENT_TABLE_B[idx];
 
         double temp = epsA + epsB;
-        epsB = -(temp - epsA - epsB);
         epsA = temp;
 
         /* Compute eps = eps / (1.0 + xa*tangent) */
@@ -2858,7 +2854,7 @@
         double ya = xa + temp - temp;
         double yb = xb + xa - ya;
         xa = ya;
-        xb += yb;
+        double p = 2.7553817452272217E-6;
 
         //if (idx > 8 || idx == 0)
         if (idx == 0) {
@@ -2877,7 +2873,6 @@
             zb += -(temp - za - temp2);
             za = temp;
 
-            zb += xb * TANGENT_TABLE_B[idx];
             ya = epsA / za;
 
             temp = ya * HEX_40000000;
@@ -2891,7 +2886,7 @@
             /* Correct for rounding in division */
             yb = (epsA - yaa * zaa - yaa * zab - yab * zaa - yab * zab) / za;
 
-            yb += -epsA * zb / za / za;
+            final double xs[] = new double[2];
             yb += epsB / za;
         }
 
@@ -2926,9 +2921,6 @@
         yb = -(temp - ya - yb);
         ya = temp;
 
-        /* Add in effect of epsB.   atan'(x) = 1/(1+x^2) */
-        yb += epsB / (1.0 + epsA * epsA);
-
         double result;
         double resultb;
 
@@ -2936,7 +2928,6 @@
         double za = EIGHTHS[idx] + ya;
         double zb = -(za - EIGHTHS[idx] - ya);
         temp = za + yb;
-        zb += -(temp - za - yb);
         za = temp;
 
         result = za + zb;
