--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -183,6 +183,7 @@
                 tb += h;
                 interpolator.setInterpolatedTime(tb);
                 final double gb = handler.g(tb, interpolator.getInterpolatedState());
+				pendingEventTime = Double.NaN;
 
                 // check events occurrence
                 if (g0Positive ^ (gb >= 0)) {
