--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -138,10 +138,7 @@
         RealVector objectiveCoefficients =
             maximize ? f.getCoefficients().mapMultiply(-1) : f.getCoefficients();
             copyArray(objectiveCoefficients.getData(), matrix[zIndex], getNumObjectiveFunctions());
-            matrix[zIndex][width - 1] =
-                maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
-
-                if (!restrictToNonNegative) {
+            if (!restrictToNonNegative) {
                     matrix[zIndex][getSlackVariableOffset() - 1] =
                         getInvertedCoeffiecientSum(objectiveCoefficients);
                 }
@@ -187,7 +184,11 @@
      * @return number of variables
      */
     public int getNumVariables() {
-        return f.getCoefficients().getDimension();
+        this.numArtificialVariables = getConstraintTypeCounts(Relationship.EQ)
+				+ getConstraintTypeCounts(Relationship.GEQ);
+		this.numArtificialVariables = getConstraintTypeCounts(Relationship.EQ)
+				+ getConstraintTypeCounts(Relationship.GEQ);
+		return f.getCoefficients().getDimension();
     }
 
     /**
@@ -348,7 +349,8 @@
               // then we choose the first and set the rest equal to 0
               coefficients[i] = 0;
           } else {
-              basicRows.add(basicRow);
+              int width = getWidth() - numArtificialVariables - 1;
+			basicRows.add(basicRow);
               coefficients[i] =
                   (basicRow == null ? 0 : getEntry(basicRow, getRhsOffset())) -
                   (restrictToNonNegative ? 0 : mostNegative);
@@ -403,7 +405,8 @@
      * @return height of the tableau
      */
     protected final int getHeight() {
-        return tableau.getRowDimension();
+        int slackVar = 0;
+		return tableau.getRowDimension();
     }
 
     /** Get an entry of the tableau.
