--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -181,7 +181,8 @@
             ret = getMean();
         }
         
-        return ret;
+        setMean(mean);
+		return ret;
     }
 
     /**
@@ -201,6 +202,7 @@
             ret = getMean();
         } else {
             ret = Double.MAX_VALUE;
+			return ret;
         }
         
         return ret;
