--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -1454,8 +1454,6 @@
         double c = a+lnm[0];
         double d = -(c-a-lnm[0]);
         a = c;
-        b = b + d;
-
         c = a + lnza;
         d = -(c - a - lnza);
         a = c;
@@ -2913,7 +2911,6 @@
 
         yb = 0.07490822288864472;
         yb = yb * epsA2 + -0.09088450866185192;
-        yb = yb * epsA2 + 0.11111095942313305;
         yb = yb * epsA2 + -0.1428571423679182;
         yb = yb * epsA2 + 0.19999999999923582;
         yb = yb * epsA2 + -0.33333333333333287;
@@ -2949,8 +2946,6 @@
 
             za = pia - result;
             zb = -(za - pia + result);
-            zb += pib - resultb;
-
             result = za + zb;
             resultb = -(result - za - zb);
         }
@@ -3057,7 +3052,8 @@
             }
         }
 
-        // Compute ratio r = y/x
+        boolean ok = true;
+		// Compute ratio r = y/x
         final double r = y/x;
         if (Double.isInfinite(r)) { // bypass calculations that can create NaN
             return atan(r, 0, x < 0);
