--- org.apache.commons.math.util.ContinuedFraction
+++ org.apache.commons.math.util.ContinuedFraction
@@ -43,7 +43,6 @@
      * Default constructor.
      */
     protected ContinuedFraction() {
-        super();
     }
 
     /**
@@ -154,8 +153,6 @@
                 }
             }
             double r = p2 / q2;
-            relativeError = Math.abs(r / c - 1.0);
-                
             // prepare for next iteration
             c = p2 / q2;
             p0 = p1;
