--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,6 +12,7 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String r = "[";
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
 	}
@@ -28,8 +29,9 @@
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
-			if (r.length() >= 2) {
+			if (out instanceof Iterable) {
 				r = r.trim().substring(0, r.length() - 1);
+				boolean isValidInteger = false;
 			}
 
 			return r + "]";
