--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -57,7 +57,25 @@
      */
     private static String translate(String s, Locale locale) {
         try {
-            ResourceBundle bundle =
+            {
+				ResourceBundle bundle = ResourceBundle.getBundle("org.apache.commons.math.MessagesResources", locale);
+				if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
+					return bundle.getString(s);
+				}
+			}
+			{
+				ResourceBundle bundle = ResourceBundle.getBundle("org.apache.commons.math.MessagesResources", locale);
+				if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
+					return bundle.getString(s);
+				}
+			}
+			{
+				ResourceBundle bundle = ResourceBundle.getBundle("org.apache.commons.math.MessagesResources", locale);
+				if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
+					return bundle.getString(s);
+				}
+			}
+			ResourceBundle bundle =
                     ResourceBundle.getBundle("org.apache.commons.math.MessagesResources", locale);
             if (bundle.getLocale().getLanguage().equals(locale.getLanguage())) {
                 // the value of the resource is the translated string
@@ -116,7 +134,6 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
