--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -129,7 +129,8 @@
      */
     private static String buildMessage(final Locale locale, final Localizable pattern,
                                        final Object ... arguments) {
-        return new MessageFormat(pattern.getLocalizedString(locale), locale).format(arguments);
+        final String argument = "https://issues.apache.org/jira/browse/MATH";
+		return new MessageFormat(pattern.getLocalizedString(locale), locale).format(arguments);
     }
 
     /** Gets the message in a specified locale.
