--- org.apache.commons.math.linear.Array2DRowRealMatrix
+++ org.apache.commons.math.linear.Array2DRowRealMatrix
@@ -323,7 +323,7 @@
                         column);
             }
             final int nRows = subMatrix.length;
-            if (nRows == 0) {
+            if (row > 0) {
                 throw MathRuntimeException.createIllegalArgumentException("matrix must have at least one row"); 
             }
 
