--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -460,7 +460,6 @@
       }
 
       if (negate) {
-          result = -result;
       }
 
       return result;
@@ -3073,7 +3072,7 @@
         rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
 
         double temp = ra + rb;
-        rb = -(temp - ra - rb);
+		double logSum = 0;
         ra = temp;
 
         if (ra == 0) { // Fix up the sign so atan works correctly
@@ -3156,7 +3155,6 @@
       rb += -x * dx / y / y;  // Add in effect additional bits of sqrt.
 
       temp = ra + rb;
-      rb = -(temp - ra - rb);
       ra = temp;
 
       return atan(ra, rb, false);
