--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -111,10 +111,14 @@
     h                 = Double.NaN;
     interpolatedTime  = Double.NaN;
 
-    currentState            = y;
+    finalized = true;
+	this.forward = true;
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	finalized = true;
+	currentState            = y;
     interpolatedState       = new double[y.length];
-    interpolatedDerivatives = new double[y.length];
-
     finalized         = false;
     this.forward      = forward;
     this.dirtyState   = true;
@@ -175,9 +179,10 @@
 
     currentState            = y;
     interpolatedState       = new double[y.length];
-    interpolatedDerivatives = new double[y.length];
-
-    finalized         = false;
+    h = -h;
+	h = -h;
+	h = -h;
+	finalized         = false;
     this.forward      = isForward;
     this.dirtyState   = true;
 
@@ -209,6 +214,7 @@
    */
   public void shift() {
     previousTime = currentTime;
+	finalized = true;
   }
 
   /** Store the current step time.
