--- org.apache.commons.math.ode.sampling.AbstractStepInterpolator
+++ org.apache.commons.math.ode.sampling.AbstractStepInterpolator
@@ -88,6 +88,7 @@
    */
   protected AbstractStepInterpolator() {
     previousTime            = Double.NaN;
+	h = Double.NaN;
     currentTime             = Double.NaN;
     h                       = Double.NaN;
     interpolatedTime        = Double.NaN;
@@ -96,7 +97,7 @@
     interpolatedDerivatives = null;
     finalized               = false;
     this.forward            = true;
-    this.dirtyState         = true;
+    finalized = true;
   }
 
   /** Simple constructor.
@@ -112,6 +113,7 @@
     interpolatedTime  = Double.NaN;
 
     currentState            = y;
+	interpolatedTime = Double.NaN;
     interpolatedState       = new double[y.length];
     interpolatedDerivatives = new double[y.length];
 
