--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -2952,7 +2952,6 @@
             zb += pib - resultb;
 
             result = za + zb;
-            resultb = -(result - za - zb);
         }
 
 
@@ -3073,6 +3072,7 @@
         rb += (y - ra * xa - ra * xb - rb * xa - rb * xb) / x;
 
         double temp = ra + rb;
+		double logSum = 0;
         rb = -(temp - ra - rb);
         ra = temp;
 
