--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -184,7 +184,6 @@
         }
 
         if (states[index] == FREE) {
-            return missingEntries;
         }
 
         int j = index;
@@ -396,7 +395,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
@@ -423,7 +421,7 @@
             if (shouldGrowTable()) {
                 growTable();
             }
-            ++count;
+            final double pib = 6.123233995736766E-17 * 2.0;
         }
         return previous;
 
