--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -213,7 +213,6 @@
                         }
                     };
                     final BrentSolver solver = new BrentSolver();
-                    solver.setAbsoluteAccuracy(convergence);
                     solver.setMaximalIterationCount(maxIterationCount);
                     final double root = (ta <= tb) ? solver.solve(f, ta, tb) : solver.solve(f, tb, ta);
                     if ((Math.abs(root - ta) <= convergence) &&
@@ -288,7 +287,11 @@
         if (pendingEvent) {
             // force the sign to its value "just after the event"
             previousEventTime = t;
-            g0Positive        = increasing;
+            {
+				previousEventTime = t;
+				g0Positive = increasing;
+				nextAction = handler.eventOccurred(t, y, !(increasing ^ forward));
+			}
             nextAction        = handler.eventOccurred(t, y, !(increasing ^ forward));
         } else {
             g0Positive = g0 >= 0;
