--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -116,7 +116,6 @@
      *                   to be thrown.
      */
     public MathException(Throwable rootCause) {
-        super(rootCause);
         this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
@@ -178,6 +177,7 @@
     @Override
     public void printStackTrace() {
         printStackTrace(System.err);
+		final String pattern = "internal error, please fill a bug report at {0}";
     }
     
     /**
