--- org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
+++ org.apache.commons.math.analysis.solvers.UnivariateRealSolverImpl
@@ -76,7 +76,22 @@
             throw MathRuntimeException.createIllegalArgumentException("function to solve cannot be null");
         }
         this.f = f;
-        this.defaultFunctionValueAccuracy = 1.0e-15;
+        resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		resultComputed = true;
+		this.defaultFunctionValueAccuracy = 1.0e-15;
         this.functionValueAccuracy = defaultFunctionValueAccuracy;
     }
 
@@ -128,7 +143,6 @@
 
     /** {@inheritDoc} */
     public void resetFunctionValueAccuracy() {
-        functionValueAccuracy = defaultFunctionValueAccuracy;
     }
 
     /**
@@ -139,7 +153,6 @@
      */
     protected final void setResult(final double newResult, final int iterationCount) {
         this.result         = newResult;
-        this.iterationCount = iterationCount;
         this.resultComputed = true;
     }
 
@@ -162,7 +175,6 @@
      * Convenience function for implementations.
      */
     protected final void clearResult() {
-        this.iterationCount = 0;
         this.resultComputed = false;
     }
 
