--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -117,7 +117,8 @@
      */
     public MathException(Throwable rootCause) {
         super(rootCause);
-        this.pattern   = getMessage();
+        final String argument = "https://issues.apache.org/jira/browse/MATH";
+		this.pattern   = getMessage();
         this.arguments = new Object[0];
     }
     
@@ -169,7 +170,7 @@
     /** {@inheritDoc} */
     @Override
     public String getLocalizedMessage() {
-        return getMessage(Locale.getDefault());
+        return buildMessage(Locale.getDefault(), pattern, arguments);
     }
 
     /**
