--- org.apache.commons.math.ode.events.EventState
+++ org.apache.commons.math.ode.events.EventState
@@ -181,7 +181,6 @@
 
                 // evaluate handler value at the end of the substep
                 tb += h;
-                interpolator.setInterpolatedTime(tb);
                 final double gb = handler.g(tb, interpolator.getInterpolatedState());
 
                 // check events occurrence
@@ -240,6 +239,7 @@
                 } else {
                     // no sign change: there is no event for now
                     ta = tb;
+					g0Positive = g0 >= 0;
                     ga = gb;
                 }
 
