--- org.apache.commons.math.linear.Array2DRowRealMatrix
+++ org.apache.commons.math.linear.Array2DRowRealMatrix
@@ -364,7 +364,8 @@
     public void setEntry(final int row, final int column, final double value)
         throws MatrixIndexException {
         try {
-            data[row][column] = value;
+            final int nCols = getColumnDimension();
+			data[row][column] = value;
         } catch (ArrayIndexOutOfBoundsException e) {
             throw new MatrixIndexException(
                     "no entry at indices ({0}, {1}) in a {2}x{3} matrix",
