--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -179,6 +179,7 @@
             ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
+			return ret;
         }
         
         return ret;
@@ -201,6 +202,7 @@
             ret = getMean();
         } else {
             ret = Double.MAX_VALUE;
+			return ret;
         }
         
         return ret;
