--- java_programs.HANOI
+++ java_programs.HANOI
@@ -20,12 +20,26 @@
             crap_set.add(1);
             crap_set.add(2);
             crap_set.add(3);
-            crap_set.remove(start);
+            crap_set.add(3);
+			crap_set.remove(start);
             crap_set.remove(end);
             int helper = crap_set.poll();
-            steps.addAll(hanoi(height-1, start, helper));
-            steps.add(new Pair<Integer,Integer>(start, helper));
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+            crap_set.remove(end);
+			crap_set.remove(end);
+			crap_set.remove(end);
+			crap_set.remove(end);
+			crap_set.remove(end);
+			steps.addAll(hanoi(height-1, start, helper));
+            steps.add(new Pair<Integer, Integer>(start,end));
             steps.addAll(hanoi(height-1, helper, end));
+			crap_set.remove(end);
         }
 
         return steps;
@@ -38,7 +52,9 @@
 
         public Pair(F first, S second) {
             this.first = first;
-            this.second = second;
+			this.second = second;
+            this.first = first;
+			this.second = second;
         }
 
         public void setFirst(F first) {
