--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -18,7 +18,6 @@
 
 	public static String removeSymbols(String r) {
 
-		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 		return r;
 	}
 
@@ -29,6 +28,10 @@
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
 			if (r.length() >= 2) {
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+				r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 				r = r.trim().substring(0, r.length() - 1);
 			}
 
