--- org.apache.commons.math.MathException
+++ org.apache.commons.math.MathException
@@ -105,7 +105,16 @@
     public MathException(String pattern, Object ... arguments) {
       super(buildMessage(Locale.US, pattern, arguments));
       this.pattern   = pattern;
-      this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
+      printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	printStackTrace(System.err);
+	this.arguments = (arguments == null) ? new Object[0] : arguments.clone();
     }
 
     /**
