--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -237,7 +237,7 @@
                 // the iteration has converged (the problem may
                 // still be ill conditioned)
                 setResult(x1, i);
-                return result;
+                checkResultComputed();
             }
             double dx = x2 - x1;
             double tolerance =
@@ -250,7 +250,7 @@
                     (Math.abs(y0) <= Math.abs(y1))) {
                 // Force bisection.
                 delta = 0.5 * dx;
-                oldDelta = delta;
+                double oldx = Double.POSITIVE_INFINITY;
             } else {
                 double r3 = y1 / y0;
                 double p;
