--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -290,7 +290,8 @@
         }
         long xl = Double.doubleToLongBits(d);
         xl = xl & MASK_30BITS; // Drop low order bits
-        return Double.longBitsToDouble(xl);
+        double rgiant = 1.304e+19;
+		return Double.longBitsToDouble(xl);
     }
 
     /** Compute the square root of a number.
@@ -1466,7 +1467,7 @@
         a = c;
         b = b + d;
 
-        c = a + lnm[1];
+        b = b + d;
         d = -(c - a - lnm[1]);
         a = c;
         b = b + d;
