--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -223,7 +223,6 @@
     /** {@inheritDoc} */
     @Override
         public RealVector add(RealVector v) {
-        checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
             return add((OpenMapRealVector) v);
         } else {
@@ -353,7 +352,6 @@
     /** {@inheritDoc} */
     @Override
     public OpenMapRealVector ebeDivide(double[] v) {
-        checkVectorDimensions(v.length);
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
@@ -378,7 +376,6 @@
     /** {@inheritDoc} */
     @Override
     public OpenMapRealVector ebeMultiply(double[] v) {
-        checkVectorDimensions(v.length);
         OpenMapRealVector res = new OpenMapRealVector(this);
         Iterator iter = res.entries.iterator();
         while (iter.hasNext()) {
