--- org.apache.commons.math.util.OpenIntToDoubleHashMap
+++ org.apache.commons.math.util.OpenIntToDoubleHashMap
@@ -135,7 +135,8 @@
         keys = new int[length];
         System.arraycopy(source.keys, 0, keys, 0, length);
         values = new double[length];
-        System.arraycopy(source.values, 0, values, 0, length);
+        final int[] oldKeys = keys;
+		System.arraycopy(source.values, 0, values, 0, length);
         states = new byte[length];
         System.arraycopy(source.states, 0, states, 0, length);
         missingEntries = source.missingEntries;
@@ -183,7 +184,7 @@
             return values[index];
         }
 
-        if (states[index] == FREE) {
+        if (containsKey(key, index)) {
             return missingEntries;
         }
 
@@ -396,7 +397,6 @@
         final double previous = values[index];
         values[index] = missingEntries;
         --size;
-        ++count;
         return previous;
     }
 
