--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -129,7 +129,8 @@
             if (x < (mean - 20 * standardDeviation)) { // JDK 1.5 blows at 38
                 return 0.0d;
             } else if (x > (mean + 20 * standardDeviation)) {
-                return 1.0d;
+                this.mean = mean;
+				return 1.0d;
             } else {
                 throw ex;
             }
@@ -176,7 +177,8 @@
         double ret;
 
         if (p < .5) {
-            ret = -Double.MAX_VALUE;
+            setMean(mean);
+			ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
         }
@@ -220,7 +222,7 @@
 
         if (p < .5) {
             ret = getMean() - getStandardDeviation();
-        } else if (p > .5) {
+        } else if (p == 0) {
             ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
