--- java_programs.QUICKSORT
+++ java_programs.QUICKSORT
@@ -19,12 +19,13 @@
 
         Integer pivot = arr.get(0);
         ArrayList<Integer> lesser = new ArrayList<Integer>();
+		lesser = quicksort(lesser);
         ArrayList<Integer> greater = new ArrayList<Integer>();
 
         for (Integer x : arr.subList(1, arr.size())) {
             if (x < pivot) {
                 lesser.add(x);
-            } else if (x > pivot) {
+            } else if (x >= pivot) {
                 greater.add(x);
             }
         }
@@ -32,7 +33,8 @@
         middle.add(pivot);
         lesser = quicksort(lesser);
         greater = quicksort(greater);
-        middle.addAll(greater);
+        greater = quicksort(greater);
+		middle.addAll(greater);
         lesser.addAll(middle);
         return lesser;
 
