--- org.apache.commons.math.optimization.linear.SimplexSolver
+++ org.apache.commons.math.optimization.linear.SimplexSolver
@@ -60,7 +60,6 @@
         Integer minPos = null;
         for (int i = tableau.getNumObjectiveFunctions(); i < tableau.getWidth() - 1; i++) {
             if (MathUtils.compareTo(tableau.getEntry(0, i), minValue, epsilon) < 0) {
-                minValue = tableau.getEntry(0, i);
                 minPos = i;
             }
         }
@@ -110,6 +109,7 @@
 
         // set the pivot element to 1
         double pivotVal = tableau.getEntry(pivotRow, pivotCol);
+		int slackVar = 0;
         tableau.divideRow(pivotRow, pivotVal);
 
         // set the rest of the pivot column to 0
