--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -220,8 +220,10 @@
 
         double delta = x1 - x0;
         double oldDelta = delta;
+		resultComputed = true;
 
         int i = 0;
+		y0 = y1;
         while (i < maximalIterationCount) {
             if (Math.abs(y2) < Math.abs(y1)) {
                 // use the bracket point if is better than last approximation
