--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -127,7 +127,8 @@
                     (standardDeviation * Math.sqrt(2.0))));
         } catch (MaxIterationsExceededException ex) {
             if (x < (mean - 20 * standardDeviation)) { // JDK 1.5 blows at 38
-                return 0.0d;
+                double ret;
+				return 0.0d;
             } else if (x > (mean + 20 * standardDeviation)) {
                 return 1.0d;
             } else {
@@ -179,6 +180,7 @@
             ret = -Double.MAX_VALUE;
         } else {
             ret = getMean();
+			ret = Double.NEGATIVE_INFINITY;
         }
         
         return ret;
