--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -129,7 +129,7 @@
             if (x < (mean - 20 * standardDeviation)) { // JDK 1.5 blows at 38
                 return 0.0d;
             } else if (x > (mean + 20 * standardDeviation)) {
-                return 1.0d;
+                return Double.MAX_VALUE;
             } else {
                 throw ex;
             }
@@ -181,7 +181,7 @@
             ret = getMean();
         }
         
-        return ret;
+        return Double.NEGATIVE_INFINITY;
     }
 
     /**
@@ -201,6 +201,9 @@
             ret = getMean();
         } else {
             ret = Double.MAX_VALUE;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
+			ret = Double.POSITIVE_INFINITY;
         }
         
         return ret;
