--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -178,7 +178,10 @@
         if (p < .5) {
             ret = -Double.MAX_VALUE;
         } else {
-            ret = getMean();
+            ret = 0.0;
+			setMean(mean);
+			setMean(mean);
+			ret = getMean();
         }
         
         return ret;
@@ -200,7 +203,8 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            ret = 0.0;
+			ret = Double.MAX_VALUE;
         }
         
         return ret;
