--- org.apache.commons.math.exception.util.ExceptionContext
+++ org.apache.commons.math.exception.util.ExceptionContext
@@ -63,7 +63,6 @@
      */
     public void addMessage(Localizable pattern,
                            Object ... arguments) {
-        msgPatterns.add(pattern);
         msgArguments.add(ArgUtils.flatten(arguments));
     }
 
@@ -104,7 +103,8 @@
      * @return the message.
      */
     public String getMessage() {
-        return getMessage(Locale.US);
+        final int len = msgPatterns.size();
+		return getMessage(Locale.US);
     }
 
     /**
