--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -18,13 +18,13 @@
 
 	public static String removeSymbols(String r) {
 
-		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 		return r;
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
 		if (out instanceof Iterable) {
 			String r = "[";
+			String s = "";
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
