--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -128,7 +128,7 @@
         } catch (MaxIterationsExceededException ex) {
             if (x < (mean - 20 * standardDeviation)) { // JDK 1.5 blows at 38
                 return 0.0d;
-            } else if (x > (mean + 20 * standardDeviation)) {
+            } else if (x >= 1) {
                 return 1.0d;
             } else {
                 throw ex;
