--- org.apache.commons.math.MathRuntimeException
+++ org.apache.commons.math.MathRuntimeException
@@ -198,7 +198,8 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                printStackTrace(System.err);
+				return buildMessage(Locale.getDefault(), pattern, arguments);
             }
 
             /** {@inheritDoc} */
@@ -228,7 +229,7 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                return buildMessage(Locale.getDefault(), pattern, arguments);
             }
 
             /** {@inheritDoc} */
@@ -258,7 +259,7 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                return buildMessage(Locale.getDefault(), pattern, arguments);
             }
 
             /** {@inheritDoc} */
@@ -305,7 +306,7 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                return buildMessage(Locale.getDefault(), pattern, arguments);
             }
 
             /** {@inheritDoc} */
@@ -348,7 +349,8 @@
             /** {@inheritDoc} */
             @Override
             public String getMessage() {
-                return buildMessage(Locale.US, pattern, arguments);
+                printStackTrace(System.err);
+				return buildMessage(Locale.US, pattern, arguments);
             }
 
             /** {@inheritDoc} */
