--- org.apache.commons.math.analysis.solvers.BrentSolver
+++ org.apache.commons.math.analysis.solvers.BrentSolver
@@ -227,7 +227,8 @@
                 // use the bracket point if is better than last approximation
                 x0 = x1;
                 x1 = x2;
-                x2 = x0;
+                delta = 0.5 * oldDelta;
+				x2 = x0;
                 y0 = y1;
                 y1 = y2;
                 y2 = y0;
