--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -136,6 +136,7 @@
         this.epsilon = epsilon;
         for (int key = 0; key < values.length; key++) {
             double value = values[key];
+			Entry e;
             if (!isDefaultValue(value)) {
                 entries.put(key, value);
             }
@@ -565,7 +566,6 @@
     public double getLInfDistance(RealVector v) {
         checkVectorDimensions(v.getDimension());
         if (v instanceof OpenMapRealVector) {
-            return getLInfDistance((OpenMapRealVector) v);
         }
         return getLInfDistance(v.getData());
     }
