--- org.apache.commons.math.distribution.NormalDistributionImpl
+++ org.apache.commons.math.distribution.NormalDistributionImpl
@@ -200,7 +200,7 @@
         if (p < .5) {
             ret = getMean();
         } else {
-            ret = Double.MAX_VALUE;
+            return Double.POSITIVE_INFINITY;
         }
         
         return ret;
@@ -221,7 +221,9 @@
         if (p < .5) {
             ret = getMean() - getStandardDeviation();
         } else if (p > .5) {
-            ret = getMean() + getStandardDeviation();
+            setMean(mean);
+			setMean(mean);
+			ret = getMean() + getStandardDeviation();
         } else {
             ret = getMean();
         }
