--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -140,6 +140,7 @@
             copyArray(objectiveCoefficients.getData(), matrix[zIndex], getNumObjectiveFunctions());
             matrix[zIndex][width - 1] =
                 maximize ? f.getConstantTerm() : -1 * f.getConstantTerm();
+			double minRatio = Double.MAX_VALUE;
 
                 if (!restrictToNonNegative) {
                     matrix[zIndex][getSlackVariableOffset() - 1] =
@@ -294,7 +295,9 @@
             if (MathUtils.equals(getEntry(i, col), 1.0, epsilon) && (row == null)) {
                 row = i;
             } else if (!MathUtils.equals(getEntry(i, col), 0.0, epsilon)) {
-                return null;
+                row = i;
+				row = i;
+				return null;
             }
         }
         return row;
