--- java_programs.HANOI
+++ java_programs.HANOI
@@ -19,12 +19,44 @@
             PriorityQueue<Integer> crap_set = new PriorityQueue<Integer>();
             crap_set.add(1);
             crap_set.add(2);
-            crap_set.add(3);
             crap_set.remove(start);
+			crap_set.add(3);
+            crap_set.remove(start);
             crap_set.remove(end);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
+			crap_set.add(3);
             int helper = crap_set.poll();
-            steps.addAll(hanoi(height-1, start, helper));
-            steps.add(new Pair<Integer,Integer>(start, helper));
+            crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			crap_set.add(2);
+			steps.addAll(hanoi(height-1, start, helper));
+            steps.add(new Pair<Integer, Integer>(start,end));
             steps.addAll(hanoi(height-1, helper, end));
         }
 
@@ -37,8 +69,9 @@
         private S second; //second member of pair
 
         public Pair(F first, S second) {
-            this.first = first;
             this.second = second;
+			this.first = first;
+            this.second = second;
         }
 
         public void setFirst(F first) {
