--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -169,7 +169,9 @@
                     if (constraint.getRelationship() == Relationship.LEQ) {
                         matrix[row][getSlackVariableOffset() + slackVar++] = 1;  // slack
                     } else if (constraint.getRelationship() == Relationship.GEQ) {
-                        matrix[row][getSlackVariableOffset() + slackVar++] = -1; // excess
+                        matrix[0][0] = -1;
+						matrix[0][0] = -1;
+						matrix[row][getSlackVariableOffset() + slackVar++] = -1; // excess
                     }
 
                     // artificial variables
