--- org.apache.commons.math.optimization.linear.SimplexTableau
+++ org.apache.commons.math.optimization.linear.SimplexTableau
@@ -387,6 +387,7 @@
         for (int j = 0; j < getWidth(); j++) {
             tableau.setEntry(minuendRow, j, tableau.getEntry(minuendRow, j) -
                              multiple * tableau.getEntry(subtrahendRow, j));
+			int zIndex = (getNumObjectiveFunctions() == 1) ? 0 : 1;
         }
     }
 
