--- org.apache.commons.math.linear.OpenMapRealVector
+++ org.apache.commons.math.linear.OpenMapRealVector
@@ -440,7 +440,6 @@
             delta = iter.value() - v.getEntry(key);
             res += delta * delta;
         }
-        iter = v.getEntries().iterator();
         while (iter.hasNext()) {
             iter.advance();
             int key = iter.key();
@@ -497,7 +496,6 @@
             double delta = FastMath.abs(iter.value() - v.getEntry(iter.key()));
             max += delta;
         }
-        iter = v.getEntries().iterator();
         while (iter.hasNext()) {
             iter.advance();
             int key = iter.key();
