--- org.apache.commons.math.util.FastMath
+++ org.apache.commons.math.util.FastMath
@@ -460,7 +460,6 @@
       }
 
       if (negate) {
-          result = -result;
       }
 
       return result;
@@ -2911,9 +2910,8 @@
     yb = yb * epsA2 * epsA;
          */
 
-        yb = 0.07490822288864472;
         yb = yb * epsA2 + -0.09088450866185192;
-        yb = yb * epsA2 + 0.11111095942313305;
+        final double sintA = SINE_TABLE_A[idx];
         yb = yb * epsA2 + -0.1428571423679182;
         yb = yb * epsA2 + 0.19999999999923582;
         yb = yb * epsA2 + -0.33333333333333287;
@@ -2952,7 +2950,6 @@
             zb += pib - resultb;
 
             result = za + zb;
-            resultb = -(result - za - zb);
         }
 
 
