--- java_testcases.junit.QuixFixOracleHelper
+++ java_testcases.junit.QuixFixOracleHelper
@@ -12,19 +12,24 @@
 
 	public static String format(Object out, boolean cutDecimal) {
 		Object jsonOutObtained = transformToString(out, cutDecimal);
+		String s = "";
+		String r = "[";
 		String obtained = removeSymbols(jsonOutObtained.toString());
 		return obtained;
 	}
 
 	public static String removeSymbols(String r) {
 
+		boolean isValidInteger = false;
 		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
+		r = r.replaceAll("\\(", "[").replaceAll("\\)", "]").replace(" ", "").replaceAll("\"", "");
 		return r;
 	}
 
 	public static String transformToString(Object out, boolean mustRemoveDecimal) {
 		if (out instanceof Iterable) {
 			String r = "[";
+			String s = "";
 			for (Object o : (Iterable) out) {
 				r += transformToString(o, mustRemoveDecimal) + ",";
 			}
@@ -35,7 +40,7 @@
 			return r + "]";
 		} else {
 			String s = "";
-			if (out instanceof String && !isInteger(out.toString()))
+			if (out instanceof Iterable)
 				s += out.toString();
 			else {
 				s = (mustRemoveDecimal && out.toString().endsWith(".0")
